<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Answers to practical exercises | Practical exercises for Methods in Macroecology and Macroevolution module</title>
  <meta name="description" content="Chapter 12 Answers to practical exercises | Practical exercises for Methods in Macroecology and Macroevolution module" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Answers to practical exercises | Practical exercises for Methods in Macroecology and Macroevolution module" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Answers to practical exercises | Practical exercises for Methods in Macroecology and Macroevolution module" />
  
  
  

<meta name="author" content="Natalie Cooper" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="critical-thinking-about-methods-and-analyses.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">PCM Primer</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#datasets-and-scripts"><i class="fa fa-check"></i>Datasets and scripts</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-to-use-these-materials"><i class="fa fa-check"></i>How to use these materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#boxes"><i class="fa fa-check"></i>Boxes</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#example-datasets"><i class="fa fa-check"></i>Example datasets</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#frog-eye-size-evolution"><i class="fa fa-check"></i>Frog eye size evolution</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#natricine-snake-head-shape-evolution"><i class="fa fa-check"></i>Natricine snake head shape evolution</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#diversification-in-dragonflies"><i class="fa fa-check"></i>Diversification in dragonflies</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citing-r-and-r-packages"><i class="fa fa-check"></i>Citing R and R packages</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><i class="fa fa-check"></i><b>1</b> What you need to be able to do in R before you start</a><ul>
<li class="chapter" data-level="1.1" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>1.1</b> Installing R (and RStudio)</a></li>
<li class="chapter" data-level="1.2" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#downloading-the-data"><i class="fa fa-check"></i><b>1.2</b> Downloading the data</a></li>
<li class="chapter" data-level="1.3" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#r-projects"><i class="fa fa-check"></i><b>1.3</b> R Projects</a></li>
<li class="chapter" data-level="1.4" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#using-a-script"><i class="fa fa-check"></i><b>1.4</b> Using a script</a></li>
<li class="chapter" data-level="1.5" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#installing-and-loading-extra-packages-in-r"><i class="fa fa-check"></i><b>1.5</b> Installing and loading extra packages in R</a><ul>
<li class="chapter" data-level="1.5.1" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#loading-packages-into-r"><i class="fa fa-check"></i><b>1.5.1</b> Loading packages into R</a></li>
<li class="chapter" data-level="1.5.2" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#ggtree-a-special-case-for-package-installation"><i class="fa fa-check"></i><b>1.5.2</b> <code>ggtree</code>, a special case for package installation</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#loading-and-viewing-your-data-in-r"><i class="fa fa-check"></i><b>1.6</b> Loading and viewing your data in R</a></li>
<li class="chapter" data-level="1.7" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#errors-and-warnings"><i class="fa fa-check"></i><b>1.7</b> Errors and warnings</a><ul>
<li class="chapter" data-level="1.7.1" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#removing-the"><i class="fa fa-check"></i><b>1.7.1</b> Removing the <code>+</code></a></li>
</ul></li>
<li class="chapter" data-level="1.8" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#summary"><i class="fa fa-check"></i><b>1.8</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html"><i class="fa fa-check"></i><b>2</b> Diversity Indices in R</a><ul>
<li class="chapter" data-level="2.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#before-you-start"><i class="fa fa-check"></i><b>2.1</b> Before you start</a></li>
<li class="chapter" data-level="2.2" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#preparation"><i class="fa fa-check"></i><b>2.2</b> Preparation</a></li>
<li class="chapter" data-level="2.3" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#what-are-diversity-indices"><i class="fa fa-check"></i><b>2.3</b> What are diversity indices?</a><ul>
<li class="chapter" data-level="2.3.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#alpha-beta-and-gamma-diversity"><i class="fa fa-check"></i><b>2.3.1</b> <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\gamma\)</span> diversity</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#practical-example-using-pokemon"><i class="fa fa-check"></i><b>2.4</b> Practical example using Pokemon…</a></li>
<li class="chapter" data-level="2.5" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#species-diversity-indices"><i class="fa fa-check"></i><b>2.5</b> Species diversity indices</a><ul>
<li class="chapter" data-level="2.5.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#alpha-diversity"><i class="fa fa-check"></i><b>2.5.1</b> <span class="math inline">\(\alpha\)</span> diversity</a></li>
<li class="chapter" data-level="2.5.2" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#beta-diversity"><i class="fa fa-check"></i><b>2.5.2</b> <span class="math inline">\(\beta\)</span> diversity</a></li>
<li class="chapter" data-level="2.5.3" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#gamma-diversity"><i class="fa fa-check"></i><b>2.5.3</b> <span class="math inline">\(\gamma\)</span> diversity</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#species-accumulation-curves-colwell-coddington-1994"><i class="fa fa-check"></i><b>2.6</b> Species accumulation curves (Colwell &amp; Coddington 1994)</a><ul>
<li class="chapter" data-level="2.6.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#pokemon-species-accumulation-curves"><i class="fa fa-check"></i><b>2.6.1</b> Pokemon species accumulation curves</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#practical-exercise"><i class="fa fa-check"></i><b>2.7</b> Practical exercise</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html"><i class="fa fa-check"></i><b>3</b> Understanding, plotting and manipulating phylogenies in R</a><ul>
<li class="chapter" data-level="3.1" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#before-you-start-1"><i class="fa fa-check"></i><b>3.1</b> Before you start</a></li>
<li class="chapter" data-level="3.2" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#what-do-phylogeny-files-look-like"><i class="fa fa-check"></i><b>3.2</b> What do phylogeny files look like?</a><ul>
<li class="chapter" data-level="3.2.1" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#newick-trees"><i class="fa fa-check"></i><b>3.2.1</b> Newick trees</a></li>
<li class="chapter" data-level="3.2.2" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#nexus-trees"><i class="fa fa-check"></i><b>3.2.2</b> NEXUS trees</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#how-are-phylogenies-formatted-in-r"><i class="fa fa-check"></i><b>3.3</b> How are phylogenies formatted in R?</a></li>
<li class="chapter" data-level="3.4" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#plotting-phylogenies"><i class="fa fa-check"></i><b>3.4</b> Plotting phylogenies</a><ul>
<li class="chapter" data-level="3.4.1" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#basic-phylogeny-plotting-with-ape"><i class="fa fa-check"></i><b>3.4.1</b> Basic phylogeny plotting with APE</a></li>
<li class="chapter" data-level="3.4.2" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#basic-phylogeny-plotting-with-ggtree"><i class="fa fa-check"></i><b>3.4.2</b> Basic phylogeny plotting with <code>ggtree</code></a></li>
<li class="chapter" data-level="3.4.3" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#back-to-ggtree"><i class="fa fa-check"></i><b>3.4.3</b> Back to <code>ggtree</code></a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#phylogeny-manipulation"><i class="fa fa-check"></i><b>3.5</b> Phylogeny manipulation</a><ul>
<li class="chapter" data-level="3.5.1" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#removing-polytomies"><i class="fa fa-check"></i><b>3.5.1</b> Removing polytomies</a></li>
<li class="chapter" data-level="3.5.2" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#rooting-your-phylogeny"><i class="fa fa-check"></i><b>3.5.2</b> Rooting your phylogeny</a></li>
<li class="chapter" data-level="3.5.3" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#manipulating-the-species-in-your-phlyogeny"><i class="fa fa-check"></i><b>3.5.3</b> Manipulating the species in your phlyogeny</a></li>
<li class="chapter" data-level="3.5.4" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#writing-phylogenies-to-file"><i class="fa fa-check"></i><b>3.5.4</b> Writing phylogenies to file</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#summary-1"><i class="fa fa-check"></i><b>3.6</b> Summary</a></li>
<li class="chapter" data-level="3.7" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#practical-exercise-1"><i class="fa fa-check"></i><b>3.7</b> Practical exercise</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html"><i class="fa fa-check"></i><b>4</b> Preparing your tree and data for PCMs in R</a><ul>
<li class="chapter" data-level="4.1" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#before-you-start-2"><i class="fa fa-check"></i><b>4.1</b> Before you start</a></li>
<li class="chapter" data-level="4.2" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#preparation-1"><i class="fa fa-check"></i><b>4.2</b> Preparation</a></li>
<li class="chapter" data-level="4.3" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#reading-and-checking-your-phylogeny-in-r"><i class="fa fa-check"></i><b>4.3</b> Reading and checking your phylogeny in R</a></li>
<li class="chapter" data-level="4.4" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#reading-the-data-into-r"><i class="fa fa-check"></i><b>4.4</b> Reading the data into R</a></li>
<li class="chapter" data-level="4.5" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#matching-your-data-to-your-phylogeny"><i class="fa fa-check"></i><b>4.5</b> Matching your data to your phylogeny</a><ul>
<li class="chapter" data-level="4.5.1" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#species-names-with-spaces"><i class="fa fa-check"></i><b>4.5.1</b> Species names with spaces</a></li>
<li class="chapter" data-level="4.5.2" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#mismatches-between-species-in-your-data-and-phylogeny"><i class="fa fa-check"></i><b>4.5.2</b> Mismatches between species in your data and phylogeny</a></li>
<li class="chapter" data-level="4.5.3" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#matching-the-tree-and-the-data"><i class="fa fa-check"></i><b>4.5.3</b> Matching the tree and the data</a></li>
<li class="chapter" data-level="4.5.4" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#subsetting-your-tree-and-data"><i class="fa fa-check"></i><b>4.5.4</b> Subsetting your tree and data</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#quick-template-code"><i class="fa fa-check"></i><b>4.6</b> Quick template code</a></li>
<li class="chapter" data-level="4.7" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#summary-2"><i class="fa fa-check"></i><b>4.7</b> Summary</a></li>
<li class="chapter" data-level="4.8" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#practical-exercise-2"><i class="fa fa-check"></i><b>4.8</b> Practical exercise</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Signal in R</a><ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#before-you-start-3"><i class="fa fa-check"></i><b>5.1</b> Before you start</a></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#preparation-2"><i class="fa fa-check"></i><b>5.2</b> Preparation</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#estimating-phylogenetic-signal-for-continuous-variables"><i class="fa fa-check"></i><b>5.3</b> Estimating phylogenetic signal for continuous variables</a><ul>
<li class="chapter" data-level="5.3.1" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#pagels-lambda"><i class="fa fa-check"></i><b>5.3.1</b> Pagel’s <span class="math inline">\(\lambda\)</span></a></li>
<li class="chapter" data-level="5.3.2" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#blombergs-k-blomberg-et-al-2003"><i class="fa fa-check"></i><b>5.3.2</b> Blomberg’s <em>K</em> (Blomberg et al 2003)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#estimating-phylogenetic-signal-for-non-continuous-variables"><i class="fa fa-check"></i><b>5.4</b> Estimating phylogenetic signal for non-continuous variables</a><ul>
<li class="chapter" data-level="5.4.1" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#d"><i class="fa fa-check"></i><b>5.4.1</b> <em>D</em></a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#summary-3"><i class="fa fa-check"></i><b>5.5</b> Summary</a></li>
<li class="chapter" data-level="5.6" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#practical-exercises"><i class="fa fa-check"></i><b>5.6</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html"><i class="fa fa-check"></i><b>6</b> Phylogenetic Generalised Least Squares (PGLS) in R</a><ul>
<li class="chapter" data-level="6.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#before-you-start-4"><i class="fa fa-check"></i><b>6.1</b> Before you start</a></li>
<li class="chapter" data-level="6.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#preparation-3"><i class="fa fa-check"></i><b>6.2</b> Preparation</a></li>
<li class="chapter" data-level="6.3" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#phylogenetic-generalized-least-squares-models-pgls"><i class="fa fa-check"></i><b>6.3</b> Phylogenetic generalized least squares models (PGLS)</a><ul>
<li class="chapter" data-level="6.3.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#fitting-pgls-models-in-r"><i class="fa fa-check"></i><b>6.3.1</b> Fitting PGLS models in R</a></li>
<li class="chapter" data-level="6.3.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#model-diagnostics-for-pgls-models"><i class="fa fa-check"></i><b>6.3.2</b> Model diagnostics for PGLS models</a></li>
<li class="chapter" data-level="6.3.3" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#likelihood-profiles-for-lambda-in-pgls-models"><i class="fa fa-check"></i><b>6.3.3</b> Likelihood profiles for <span class="math inline">\(\lambda\)</span> in PGLS models</a></li>
<li class="chapter" data-level="6.3.4" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#a-pgls-example-with-more-than-one-predictor-variable"><i class="fa fa-check"></i><b>6.3.4</b> A PGLS example with more than one predictor variable</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#common-issues-when-running-pgls-models-in-r"><i class="fa fa-check"></i><b>6.4</b> Common issues when running PGLS models in R</a><ul>
<li class="chapter" data-level="6.4.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#optim-errors"><i class="fa fa-check"></i><b>6.4.1</b> Optim errors</a></li>
<li class="chapter" data-level="6.4.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#big-problems-with-small-datasets"><i class="fa fa-check"></i><b>6.4.2</b> Big problems with small datasets</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#summary-4"><i class="fa fa-check"></i><b>6.5</b> Summary</a></li>
<li class="chapter" data-level="6.6" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#practical-exercises-1"><i class="fa fa-check"></i><b>6.6</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html"><i class="fa fa-check"></i><b>7</b> Macroevolutionary models for continuous traits</a><ul>
<li class="chapter" data-level="7.1" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#before-you-start-5"><i class="fa fa-check"></i><b>7.1</b> Before you start</a></li>
<li class="chapter" data-level="7.2" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#preparation-4"><i class="fa fa-check"></i><b>7.2</b> Preparation</a></li>
<li class="chapter" data-level="7.3" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#models-of-evolution-for-continuous-traits"><i class="fa fa-check"></i><b>7.3</b> Models of evolution for continuous traits</a><ul>
<li class="chapter" data-level="7.3.1" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#loook-at-the-data"><i class="fa fa-check"></i><b>7.3.1</b> loook at the data????</a></li>
<li class="chapter" data-level="7.3.2" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#fitting-the-brownian-motion-and-ou-models-of-evolution-using-fitcontinuous"><i class="fa fa-check"></i><b>7.3.2</b> Fitting the Brownian motion and OU models of evolution using <code>fitContinuous</code></a></li>
<li class="chapter" data-level="7.3.3" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#comparing-models-of-evolution-using-aic"><i class="fa fa-check"></i><b>7.3.3</b> Comparing models of evolution using AIC</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#summary-5"><i class="fa fa-check"></i><b>7.4</b> Summary</a></li>
<li class="chapter" data-level="7.5" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#practical-exercises-2"><i class="fa fa-check"></i><b>7.5</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html"><i class="fa fa-check"></i><b>8</b> Macroevolutionary models for discrete traits</a><ul>
<li class="chapter" data-level="8.1" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#before-you-start-6"><i class="fa fa-check"></i><b>8.1</b> Before you start</a></li>
<li class="chapter" data-level="8.2" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#preparation-5"><i class="fa fa-check"></i><b>8.2</b> Preparation</a></li>
<li class="chapter" data-level="8.3" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#models-of-evolution-for-discrete-traits"><i class="fa fa-check"></i><b>8.3</b> Models of evolution for discrete traits</a><ul>
<li class="chapter" data-level="8.3.1" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#fitting-the-er-sym-and-ard-models-of-evolution-using-fitdiscrete"><i class="fa fa-check"></i><b>8.3.1</b> Fitting the ER, SYM and ARD models of evolution using <code>fitDiscrete</code></a></li>
<li class="chapter" data-level="8.3.2" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#comparing-models-of-evolution-using-aic-1"><i class="fa fa-check"></i><b>8.3.2</b> Comparing models of evolution using AIC</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#ancestral-state-estimations"><i class="fa fa-check"></i><b>8.4</b> Ancestral state estimations</a></li>
<li class="chapter" data-level="8.5" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#summary-6"><i class="fa fa-check"></i><b>8.5</b> Summary</a></li>
<li class="chapter" data-level="8.6" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#practical-exercises-3"><i class="fa fa-check"></i><b>8.6</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><i class="fa fa-check"></i><b>9</b> Beyond simple macroevolutionary models of trait evolution</a><ul>
<li class="chapter" data-level="9.1" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#before-you-start-7"><i class="fa fa-check"></i><b>9.1</b> Before you start</a></li>
<li class="chapter" data-level="9.2" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#preparation-6"><i class="fa fa-check"></i><b>9.2</b> Preparation</a></li>
<li class="chapter" data-level="9.3" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#more-complex-models-of-trait-evolution"><i class="fa fa-check"></i><b>9.3</b> More complex models of trait evolution</a><ul>
<li class="chapter" data-level="9.3.1" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#preparing-the-tree-and-data-for-ouwie"><i class="fa fa-check"></i><b>9.3.1</b> Preparing the tree and data for OUwie</a></li>
<li class="chapter" data-level="9.3.2" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#running-models-using-ouwie"><i class="fa fa-check"></i><b>9.3.2</b> Running models using OUwie</a></li>
<li class="chapter" data-level="9.3.3" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#single-rate-bm-models"><i class="fa fa-check"></i><b>9.3.3</b> Single rate BM models</a></li>
<li class="chapter" data-level="9.3.4" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#single-optimum-ou-models"><i class="fa fa-check"></i><b>9.3.4</b> Single optimum OU models</a></li>
<li class="chapter" data-level="9.3.5" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#multi-rate-bm-bms-models"><i class="fa fa-check"></i><b>9.3.5</b> Multi-rate BM (BMS) models</a></li>
<li class="chapter" data-level="9.3.6" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#multi-optima-ou-oum-models"><i class="fa fa-check"></i><b>9.3.6</b> Multi-optima OU (OUM) models</a></li>
<li class="chapter" data-level="9.3.7" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#multi-rate-multi-optima-ou-motion-models-oumv"><i class="fa fa-check"></i><b>9.3.7</b> Multi-rate multi-optima OU motion models (OUMV)</a></li>
<li class="chapter" data-level="9.3.8" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#multi-alpha-multi-optima-ou-motion-models-ouma"><i class="fa fa-check"></i><b>9.3.8</b> Multi-alpha multi-optima OU motion models (OUMA)</a></li>
<li class="chapter" data-level="9.3.9" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#multi-rate-multi-optima-ou-motion-models-oumva"><i class="fa fa-check"></i><b>9.3.9</b> Multi-rate multi-optima OU motion models (OUMVA)</a></li>
<li class="chapter" data-level="9.3.10" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#comparing-models-of-evolution-using-aic-2"><i class="fa fa-check"></i><b>9.3.10</b> Comparing models of evolution using AIC</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#summary-7"><i class="fa fa-check"></i><b>9.4</b> Summary</a></li>
<li class="chapter" data-level="9.5" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#practical-exercises-4"><i class="fa fa-check"></i><b>9.5</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html"><i class="fa fa-check"></i><b>10</b> Diversification rate analyses</a><ul>
<li class="chapter" data-level="10.1" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#before-you-start-8"><i class="fa fa-check"></i><b>10.1</b> Before you start</a></li>
<li class="chapter" data-level="10.2" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#preparation-7"><i class="fa fa-check"></i><b>10.2</b> Preparation</a></li>
<li class="chapter" data-level="10.3" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#lineage-through-time-plots-and-the-gamma-statistic"><i class="fa fa-check"></i><b>10.3</b> Lineage through time plots and the <span class="math inline">\(\gamma\)</span> statistic</a></li>
<li class="chapter" data-level="10.4" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#bamm-bayesian-analysis-of-macroevolutionary-mixtures"><i class="fa fa-check"></i><b>10.4</b> BAMM: Bayesian Analysis of Macroevolutionary Mixtures</a></li>
<li class="chapter" data-level="10.5" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#running-bamm"><i class="fa fa-check"></i><b>10.5</b> Running BAMM</a></li>
<li class="chapter" data-level="10.6" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#looking-at-the-output-of-bamm-using-bammtools-in-r"><i class="fa fa-check"></i><b>10.6</b> Looking at the output of BAMM using BAMMtools in R</a><ul>
<li class="chapter" data-level="10.6.1" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#assessing-mcmc-convergence"><i class="fa fa-check"></i><b>10.6.1</b> Assessing MCMC convergence</a></li>
<li class="chapter" data-level="10.6.2" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#analysis-of-rate-shifts-in-the-dragonfly-phylogeny"><i class="fa fa-check"></i><b>10.6.2</b> Analysis of rate shifts in the dragonfly phylogeny</a></li>
<li class="chapter" data-level="10.6.3" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#mean-phylorate-plot"><i class="fa fa-check"></i><b>10.6.3</b> Mean phylorate plot</a></li>
<li class="chapter" data-level="10.6.4" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#best-single-distinct-shift-configuration"><i class="fa fa-check"></i><b>10.6.4</b> Best single distinct shift configuration</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#credible-sets-of-shifts"><i class="fa fa-check"></i><b>10.7</b> Credible sets of shifts</a></li>
<li class="chapter" data-level="10.8" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#clade-specific-evolutionary-rates"><i class="fa fa-check"></i><b>10.8</b> Clade specific evolutionary rates</a></li>
<li class="chapter" data-level="10.9" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#rate-through-time-analysis"><i class="fa fa-check"></i><b>10.9</b> Rate-through-time analysis</a></li>
<li class="chapter" data-level="10.10" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#macroevolutionary-cohort-analysis"><i class="fa fa-check"></i><b>10.10</b> Macroevolutionary cohort analysis</a></li>
<li class="chapter" data-level="10.11" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#assumptions-and-issues-with-bamm"><i class="fa fa-check"></i><b>10.11</b> Assumptions and issues with BAMM</a></li>
<li class="chapter" data-level="10.12" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#summary-8"><i class="fa fa-check"></i><b>10.12</b> Summary</a></li>
<li class="chapter" data-level="10.13" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#practical-exercises-5"><i class="fa fa-check"></i><b>10.13</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html"><i class="fa fa-check"></i><b>11</b> Critical thinking about methods and analyses</a><ul>
<li class="chapter" data-level="11.1" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#how-to-critically-evaluate-a-paper-that-uses-phylogenetic-comparative-methods"><i class="fa fa-check"></i><b>11.1</b> How to critically evaluate a paper that uses phylogenetic comparative methods</a><ul>
<li class="chapter" data-level="" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#logicinterpretation"><i class="fa fa-check"></i>Logic/interpretation</a></li>
<li class="chapter" data-level="" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#methods"><i class="fa fa-check"></i>Methods</a></li>
<li class="chapter" data-level="" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#moving-forwards"><i class="fa fa-check"></i>Moving forwards</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#further-reading"><i class="fa fa-check"></i><b>11.2</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html"><i class="fa fa-check"></i><b>12</b> Answers to practical exercises</a><ul>
<li class="chapter" data-level="12.1" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#diversity-indices"><i class="fa fa-check"></i><b>12.1</b> Diversity indices</a></li>
<li class="chapter" data-level="12.2" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#phylogenies"><i class="fa fa-check"></i><b>12.2</b> Phylogenies</a></li>
<li class="chapter" data-level="12.3" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#preparation-8"><i class="fa fa-check"></i><b>12.3</b> Preparation</a></li>
<li class="chapter" data-level="12.4" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#phylogenetic-signal"><i class="fa fa-check"></i><b>12.4</b> Phylogenetic signal</a></li>
<li class="chapter" data-level="12.5" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#pgls"><i class="fa fa-check"></i><b>12.5</b> PGLS</a></li>
<li class="chapter" data-level="12.6" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#models-of-evolution-with-continuous-traits"><i class="fa fa-check"></i><b>12.6</b> Models of evolution with continuous traits</a></li>
<li class="chapter" data-level="12.7" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#models-of-evolution-with-discrete-traits"><i class="fa fa-check"></i><b>12.7</b> Models of evolution with discrete traits</a></li>
<li class="chapter" data-level="12.8" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#beyond-simple-macroevolutionary-models-of-trait-evolution-1"><i class="fa fa-check"></i><b>12.8</b> Beyond simple macroevolutionary models of trait evolution</a></li>
<li class="chapter" data-level="12.9" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#diversification"><i class="fa fa-check"></i><b>12.9</b> Diversification</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Practical exercises for Methods in Macroecology and Macroevolution module</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="answers-to-practical-exercises" class="section level1">
<h1><span class="header-section-number">Chapter 12</span> Answers to practical exercises</h1>
<p>Below are all the questions and answers to the practical exercises at the end of each exercise.</p>
<div id="diversity-indices" class="section level2">
<h2><span class="header-section-number">12.1</span> Diversity indices</h2>
<p>In the folder there is another (invented) dataset using British bats called <code>bat-communities.csv</code>. Read in the data, manipulate it as required by <code>vegan</code>, then answer the following questions.</p>
<div class="sourceCode" id="cb474"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb474-1"><a href="answers-to-practical-exercises.html#cb474-1"></a><span class="co"># Load packages</span></span>
<span id="cb474-2"><a href="answers-to-practical-exercises.html#cb474-2"></a><span class="kw">library</span>(vegan)</span>
<span id="cb474-3"><a href="answers-to-practical-exercises.html#cb474-3"></a><span class="kw">library</span>(picante)</span>
<span id="cb474-4"><a href="answers-to-practical-exercises.html#cb474-4"></a></span>
<span id="cb474-5"><a href="answers-to-practical-exercises.html#cb474-5"></a><span class="co"># Read in the dataset</span></span>
<span id="cb474-6"><a href="answers-to-practical-exercises.html#cb474-6"></a>bat &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/bat-communities.csv&quot;</span>)</span>
<span id="cb474-7"><a href="answers-to-practical-exercises.html#cb474-7"></a><span class="co"># Create a matrix we can use with vegan</span></span>
<span id="cb474-8"><a href="answers-to-practical-exercises.html#cb474-8"></a>bat.matrix &lt;-<span class="st"> </span><span class="kw">sample2matrix</span>(bat)</span></code></pre></div>
<ol style="list-style-type: decimal">
<li>Which site has the fewest species?</li>
</ol>
<div class="sourceCode" id="cb475"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb475-1"><a href="answers-to-practical-exercises.html#cb475-1"></a><span class="kw">specnumber</span>(bat.matrix)</span></code></pre></div>
<pre><code>##  A  B  C  D  E  F  G  H  I  J  K 
##  8  6  7  9  8 11 11 13 12 12  7</code></pre>
<p><em>Site B</em></p>
<ol start="2" style="list-style-type: decimal">
<li>How many different species are there in total?</li>
</ol>
<div class="sourceCode" id="cb477"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb477-1"><a href="answers-to-practical-exercises.html#cb477-1"></a><span class="co"># How many unique species are there?</span></span>
<span id="cb477-2"><a href="answers-to-practical-exercises.html#cb477-2"></a><span class="kw">length</span>(<span class="kw">unique</span>(bat<span class="op">$</span>Species[bat<span class="op">$</span>Abundance <span class="op">&gt;</span><span class="dv">0</span>]))</span></code></pre></div>
<pre><code>## [1] 18</code></pre>
<p><em>18</em></p>
<ol start="3" style="list-style-type: decimal">
<li>What is Simpson’s diversity index for Site J?</li>
</ol>
<div class="sourceCode" id="cb479"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb479-1"><a href="answers-to-practical-exercises.html#cb479-1"></a><span class="co"># Simpson&#39;s index</span></span>
<span id="cb479-2"><a href="answers-to-practical-exercises.html#cb479-2"></a><span class="kw">diversity</span>(bat.matrix, <span class="dt">index =</span> <span class="st">&quot;simpson&quot;</span>)</span></code></pre></div>
<pre><code>##         A         B         C         D         E         F         G 
## 0.8430769 0.7947061 0.7375510 0.8097758 0.8350592 0.8820984 0.8733518 
##         H         I         J         K 
## 0.8922723 0.8825614 0.8198242 0.8394970</code></pre>
<p><em>0.8198242</em></p>
<ol start="4" style="list-style-type: decimal">
<li>Draw a species accumulation curve for the bats and estimate the total number of species. If you round up numbers with decimal places, what is the maximum number of species estimated by any metric?</li>
</ol>
<div class="sourceCode" id="cb481"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb481-1"><a href="answers-to-practical-exercises.html#cb481-1"></a><span class="co"># Fit species accumulation curve</span></span>
<span id="cb481-2"><a href="answers-to-practical-exercises.html#cb481-2"></a>bat.curve &lt;-<span class="st"> </span><span class="kw">specaccum</span>(bat.matrix, <span class="dt">method =</span> <span class="st">&quot;random&quot;</span>, <span class="dt">permutations =</span> <span class="dv">1000</span>)</span>
<span id="cb481-3"><a href="answers-to-practical-exercises.html#cb481-3"></a></span>
<span id="cb481-4"><a href="answers-to-practical-exercises.html#cb481-4"></a><span class="co"># Plot the curve</span></span>
<span id="cb481-5"><a href="answers-to-practical-exercises.html#cb481-5"></a><span class="kw">plot</span>(bat.curve, <span class="dt">ci.type =</span> <span class="st">&quot;poly&quot;</span>, <span class="dt">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="dt">ci.col =</span> <span class="st">&quot;lightblue&quot;</span>, </span>
<span id="cb481-6"><a href="answers-to-practical-exercises.html#cb481-6"></a>     <span class="dt">lwd =</span> <span class="dv">2</span>, <span class="dt">ci.lty =</span> <span class="dv">0</span>, <span class="dt">xlab =</span> <span class="st">&quot;number of sites&quot;</span>, </span>
<span id="cb481-7"><a href="answers-to-practical-exercises.html#cb481-7"></a>     <span class="dt">ylab =</span> <span class="st">&quot;cumulative number of bat species&quot;</span>)</span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-328-1.png" width="672" /></p>
<div class="sourceCode" id="cb482"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb482-1"><a href="answers-to-practical-exercises.html#cb482-1"></a><span class="co"># Estimate diversity</span></span>
<span id="cb482-2"><a href="answers-to-practical-exercises.html#cb482-2"></a><span class="kw">specpool</span>(bat.matrix)</span></code></pre></div>
<pre><code>##     Species chao chao.se jack1 jack1.se    jack2     boot   boot.se  n
## All      18   18       0    18        0 16.52727 18.26721 0.4762132 11</code></pre>
<p><em>18.267 + 0.476, i.e. 19 species in total</em></p>
</div>
<div id="phylogenies" class="section level2">
<h2><span class="header-section-number">12.2</span> Phylogenies</h2>
<p>Read in the <code>frog-tree.nex</code> phylogeny from the folder. This comes from <span class="citation">Feng et al. (<a href="#ref-feng2017phylogenomics" role="doc-biblioref">2017</a>)</span>. Then do the following:</p>
<p>Read the tree into R.</p>
<div class="sourceCode" id="cb484"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb484-1"><a href="answers-to-practical-exercises.html#cb484-1"></a><span class="co"># Load packages</span></span>
<span id="cb484-2"><a href="answers-to-practical-exercises.html#cb484-2"></a><span class="kw">library</span>(ape)</span>
<span id="cb484-3"><a href="answers-to-practical-exercises.html#cb484-3"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb484-4"><a href="answers-to-practical-exercises.html#cb484-4"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb484-5"><a href="answers-to-practical-exercises.html#cb484-5"></a></span>
<span id="cb484-6"><a href="answers-to-practical-exercises.html#cb484-6"></a><span class="co"># Read in the tree</span></span>
<span id="cb484-7"><a href="answers-to-practical-exercises.html#cb484-7"></a>tree &lt;-<span class="st"> </span><span class="kw">read.nexus</span>(<span class="st">&quot;data/frog-tree.nex&quot;</span>)</span>
<span id="cb484-8"><a href="answers-to-practical-exercises.html#cb484-8"></a></span>
<span id="cb484-9"><a href="answers-to-practical-exercises.html#cb484-9"></a><span class="co"># Look at it</span></span>
<span id="cb484-10"><a href="answers-to-practical-exercises.html#cb484-10"></a>tree</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 214 tips and 213 internal nodes.
## 
## Tip labels:
##   Ascaphus_truei_Ascaphidae, Leiopelma_hochstetteri_Leiopelmatidae, Alytes_obstetricans_Alytidae, Discoglossus_pictus_Alytidae, Barbourula_busuangensis_Bombinatoridae, Bombina_orientalis_Bombinatoridae, ...
## 
## Rooted; includes branch lengths.</code></pre>
<ol style="list-style-type: decimal">
<li>Use R functions to determine:</li>
</ol>
<ol style="list-style-type: lower-alpha">
<li>how many species are in the tree?</li>
</ol>
<div class="sourceCode" id="cb486"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb486-1"><a href="answers-to-practical-exercises.html#cb486-1"></a><span class="co"># Look at the tree structure</span></span>
<span id="cb486-2"><a href="answers-to-practical-exercises.html#cb486-2"></a><span class="kw">str</span>(tree)</span></code></pre></div>
<pre><code>## List of 4
##  $ edge       : int [1:426, 1:2] 215 216 217 218 219 220 221 222 223 224 ...
##  $ edge.length: num [1:426] 0.166 0.114 0.102 0.4 0.133 ...
##  $ Nnode      : int 213
##  $ tip.label  : chr [1:214] &quot;Ascaphus_truei_Ascaphidae&quot; &quot;Leiopelma_hochstetteri_Leiopelmatidae&quot; &quot;Alytes_obstetricans_Alytidae&quot; &quot;Discoglossus_pictus_Alytidae&quot; ...
##  - attr(*, &quot;class&quot;)= chr &quot;phylo&quot;
##  - attr(*, &quot;order&quot;)= chr &quot;cladewise&quot;</code></pre>
<div class="sourceCode" id="cb488"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb488-1"><a href="answers-to-practical-exercises.html#cb488-1"></a><span class="co"># Or the whole tree</span></span>
<span id="cb488-2"><a href="answers-to-practical-exercises.html#cb488-2"></a>tree</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 214 tips and 213 internal nodes.
## 
## Tip labels:
##   Ascaphus_truei_Ascaphidae, Leiopelma_hochstetteri_Leiopelmatidae, Alytes_obstetricans_Alytidae, Discoglossus_pictus_Alytidae, Barbourula_busuangensis_Bombinatoridae, Bombina_orientalis_Bombinatoridae, ...
## 
## Rooted; includes branch lengths.</code></pre>
<p><em>There are 214 tips</em></p>
<ol start="2" style="list-style-type: lower-alpha">
<li>is the tree fully resolved?</li>
</ol>
<div class="sourceCode" id="cb490"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb490-1"><a href="answers-to-practical-exercises.html#cb490-1"></a><span class="co"># Check whether the tree is binary/fully resolved</span></span>
<span id="cb490-2"><a href="answers-to-practical-exercises.html#cb490-2"></a><span class="kw">is.binary.tree</span>(tree)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p><em>Yes</em></p>
<ol start="3" style="list-style-type: lower-alpha">
<li>is the tree rooted?</li>
</ol>
<div class="sourceCode" id="cb492"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb492-1"><a href="answers-to-practical-exercises.html#cb492-1"></a><span class="co"># Check whether the tree is rooted</span></span>
<span id="cb492-2"><a href="answers-to-practical-exercises.html#cb492-2"></a><span class="kw">is.rooted</span>(tree)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p><em>Yes</em></p>
<ol start="2" style="list-style-type: decimal">
<li>Use <code>plot.phylo</code> to plot the tree.
<ol style="list-style-type: lower-alpha">
<li>Can you change the size of the tip labels?</li>
<li>Can you make a fan-shaped plot?</li>
<li>Can you change the colour of the tips and branches?</li>
</ol></li>
</ol>
<div class="sourceCode" id="cb494"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb494-1"><a href="answers-to-practical-exercises.html#cb494-1"></a><span class="co"># Plot the tree</span></span>
<span id="cb494-2"><a href="answers-to-practical-exercises.html#cb494-2"></a><span class="kw">plot</span>(tree, </span>
<span id="cb494-3"><a href="answers-to-practical-exercises.html#cb494-3"></a>     <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="co"># to make sure it fills the whole space</span></span>
<span id="cb494-4"><a href="answers-to-practical-exercises.html#cb494-4"></a>     <span class="dt">type =</span> <span class="st">&quot;fan&quot;</span>, <span class="co"># to make a fan tree</span></span>
<span id="cb494-5"><a href="answers-to-practical-exercises.html#cb494-5"></a>     <span class="dt">cex =</span> <span class="fl">0.6</span>, <span class="co"># to reduce the tip label size (default = 1)</span></span>
<span id="cb494-6"><a href="answers-to-practical-exercises.html#cb494-6"></a>     <span class="dt">tip.color =</span> <span class="st">&quot;deeppink&quot;</span>, <span class="co"># to change the tip colour</span></span>
<span id="cb494-7"><a href="answers-to-practical-exercises.html#cb494-7"></a>     <span class="dt">edge.color =</span> <span class="st">&quot;springgreen&quot;</span>) <span class="co"># to change the branch colour</span></span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-333-1.png" width="672" /></p>
<div class="sourceCode" id="cb495"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb495-1"><a href="answers-to-practical-exercises.html#cb495-1"></a><span class="co"># yuk!</span></span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Save the tree to file as “mysuperdoopertree.nex”</li>
</ol>
<div class="sourceCode" id="cb496"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb496-1"><a href="answers-to-practical-exercises.html#cb496-1"></a><span class="co"># Save the tree</span></span>
<span id="cb496-2"><a href="answers-to-practical-exercises.html#cb496-2"></a><span class="kw">write.nexus</span>(tree, <span class="dt">file =</span> <span class="st">&quot;data/mysuperdoopertree.nex&quot;</span>)</span></code></pre></div>
</div>
<div id="preparation-8" class="section level2">
<h2><span class="header-section-number">12.3</span> Preparation</h2>
<p>In the data folder there is another tree (<code>primate-tree.nex</code>) and dataset (<code>primate-data.csv</code>) for investigating the evolution of primate life-history variables. These data come from the PanTHERIA database <span class="citation">(Jones et al. <a href="#ref-jones2009pantheria" role="doc-biblioref">2009</a>)</span> and 10kTrees <span class="citation">(Arnold, Matthews, and Nunn <a href="#ref-arnold201010ktrees" role="doc-biblioref">2010</a>)</span>.</p>
<p>Read in the tree and data then prepare them for a PCM analysis.</p>
<div class="sourceCode" id="cb497"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb497-1"><a href="answers-to-practical-exercises.html#cb497-1"></a><span class="co"># Load packages</span></span>
<span id="cb497-2"><a href="answers-to-practical-exercises.html#cb497-2"></a><span class="kw">library</span>(ape)</span>
<span id="cb497-3"><a href="answers-to-practical-exercises.html#cb497-3"></a><span class="kw">library</span>(geiger)</span>
<span id="cb497-4"><a href="answers-to-practical-exercises.html#cb497-4"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb497-5"><a href="answers-to-practical-exercises.html#cb497-5"></a><span class="kw">library</span>(treeplyr)</span>
<span id="cb497-6"><a href="answers-to-practical-exercises.html#cb497-6"></a></span>
<span id="cb497-7"><a href="answers-to-practical-exercises.html#cb497-7"></a><span class="co"># Read in the tree</span></span>
<span id="cb497-8"><a href="answers-to-practical-exercises.html#cb497-8"></a>primatetree &lt;-<span class="st"> </span><span class="kw">read.nexus</span>(<span class="st">&quot;data/primate-tree.nex&quot;</span>)</span>
<span id="cb497-9"><a href="answers-to-practical-exercises.html#cb497-9"></a><span class="co"># Look at the tree summary</span></span>
<span id="cb497-10"><a href="answers-to-practical-exercises.html#cb497-10"></a><span class="kw">str</span>(primatetree)</span></code></pre></div>
<pre><code>## List of 4
##  $ edge       : int [1:446, 1:2] 227 228 229 230 231 232 233 234 234 235 ...
##  $ edge.length: num [1:446] 4.95 17.69 19.65 8.12 4.82 ...
##  $ Nnode      : int 221
##  $ tip.label  : chr [1:226] &quot;Allenopithecus_nigroviridis&quot; &quot;Cercopithecus_ascanius&quot; &quot;Cercopithecus_cephus&quot; &quot;Cercopithecus_cephus_cephus&quot; ...
##  - attr(*, &quot;class&quot;)= chr &quot;phylo&quot;
##  - attr(*, &quot;order&quot;)= chr &quot;cladewise&quot;</code></pre>
<div class="sourceCode" id="cb499"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb499-1"><a href="answers-to-practical-exercises.html#cb499-1"></a><span class="co"># Plot the tree as a circular/fan phylogeny with small labels</span></span>
<span id="cb499-2"><a href="answers-to-practical-exercises.html#cb499-2"></a><span class="kw">plot</span>(primatetree, <span class="dt">cex =</span> <span class="fl">0.2</span>, <span class="dt">typ =</span> <span class="st">&quot;fan&quot;</span>, <span class="dt">no.margin =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-335-1.png" width="672" /></p>
<div class="sourceCode" id="cb500"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb500-1"><a href="answers-to-practical-exercises.html#cb500-1"></a><span class="co"># Check whether the tree is binary</span></span>
<span id="cb500-2"><a href="answers-to-practical-exercises.html#cb500-2"></a><span class="co"># We want this to be TRUE</span></span>
<span id="cb500-3"><a href="answers-to-practical-exercises.html#cb500-3"></a><span class="kw">is.binary.tree</span>(primatetree) </span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<div class="sourceCode" id="cb502"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb502-1"><a href="answers-to-practical-exercises.html#cb502-1"></a><span class="co"># It is FALSE so make the tree binary:</span></span>
<span id="cb502-2"><a href="answers-to-practical-exercises.html#cb502-2"></a>primatetree &lt;-<span class="st"> </span><span class="kw">multi2di</span>(primatetree)</span>
<span id="cb502-3"><a href="answers-to-practical-exercises.html#cb502-3"></a></span>
<span id="cb502-4"><a href="answers-to-practical-exercises.html#cb502-4"></a><span class="co"># Check whether the tree is binary now...</span></span>
<span id="cb502-5"><a href="answers-to-practical-exercises.html#cb502-5"></a><span class="kw">is.binary.tree</span>(primatetree) </span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb504"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb504-1"><a href="answers-to-practical-exercises.html#cb504-1"></a><span class="co"># Check whether the tree is rooted</span></span>
<span id="cb504-2"><a href="answers-to-practical-exercises.html#cb504-2"></a><span class="co"># We want this to be TRUE</span></span>
<span id="cb504-3"><a href="answers-to-practical-exercises.html#cb504-3"></a><span class="kw">is.rooted</span>(primatetree)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb506"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb506-1"><a href="answers-to-practical-exercises.html#cb506-1"></a><span class="co"># Check whether the tree is ultrametric</span></span>
<span id="cb506-2"><a href="answers-to-practical-exercises.html#cb506-2"></a><span class="co"># We want this to be TRUE, but it is FALSE</span></span>
<span id="cb506-3"><a href="answers-to-practical-exercises.html#cb506-3"></a><span class="kw">is.ultrametric</span>(primatetree)</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<div class="sourceCode" id="cb508"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb508-1"><a href="answers-to-practical-exercises.html#cb508-1"></a><span class="co"># Force ultrametric</span></span>
<span id="cb508-2"><a href="answers-to-practical-exercises.html#cb508-2"></a>primatetree &lt;-<span class="st"> </span><span class="kw">force.ultrametric</span>(primatetree)</span>
<span id="cb508-3"><a href="answers-to-practical-exercises.html#cb508-3"></a></span>
<span id="cb508-4"><a href="answers-to-practical-exercises.html#cb508-4"></a><span class="co"># Read in the data</span></span>
<span id="cb508-5"><a href="answers-to-practical-exercises.html#cb508-5"></a>primatedata &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="st">&quot;data/primate-data.csv&quot;</span>)</span></code></pre></div>
<pre><code>## 
## ── Column specification ───────────────────────────────────────────────────
## cols(
##   Order = col_character(),
##   Family = col_character(),
##   Binomial = col_character(),
##   AdultBodyMass_g = col_double(),
##   GestationLen_d = col_double(),
##   HomeRange_km2 = col_double(),
##   MaxLongevity_m = col_double(),
##   SocialGroupSize = col_double(),
##   SocialStatus = col_double()
## )</code></pre>
<div class="sourceCode" id="cb510"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb510-1"><a href="answers-to-practical-exercises.html#cb510-1"></a><span class="co"># Look at the data</span></span>
<span id="cb510-2"><a href="answers-to-practical-exercises.html#cb510-2"></a><span class="kw">glimpse</span>(primatedata)</span></code></pre></div>
<pre><code>## Rows: 77
## Columns: 9
## $ Order           &lt;chr&gt; &quot;Primates&quot;, &quot;Primates&quot;, &quot;Primates&quot;, &quot;Primates&quot;, …
## $ Family          &lt;chr&gt; &quot;Atelidae&quot;, &quot;Atelidae&quot;, &quot;Atelidae&quot;, &quot;Pitheciidae…
## $ Binomial        &lt;chr&gt; &quot;Ateles belzebuth&quot;, &quot;Ateles geoffroyi&quot;, &quot;Ateles …
## $ AdultBodyMass_g &lt;dbl&gt; 6692.42, 7582.40, 8697.25, 958.13, 558.00, 290.2…
## $ GestationLen_d  &lt;dbl&gt; 138.20, 226.37, 228.18, 164.00, 153.99, 144.00, …
## $ HomeRange_km2   &lt;dbl&gt; 2.28000, 0.73000, 1.36000, 0.02000, 0.32000, 0.0…
## $ MaxLongevity_m  &lt;dbl&gt; 336.0, 327.6, 453.6, 303.6, 214.8, 201.6, 181.2,…
## $ SocialGroupSize &lt;dbl&gt; 14.50, 42.00, 20.00, 2.95, 6.85, 8.55, 6.00, 20.…
## $ SocialStatus    &lt;dbl&gt; 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, …</code></pre>
<div class="sourceCode" id="cb512"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb512-1"><a href="answers-to-practical-exercises.html#cb512-1"></a><span class="co"># Replace spaces with underscores in species names</span></span>
<span id="cb512-2"><a href="answers-to-practical-exercises.html#cb512-2"></a>primatedata &lt;-<span class="st"> </span></span>
<span id="cb512-3"><a href="answers-to-practical-exercises.html#cb512-3"></a><span class="st">  </span>primatedata <span class="op">%&gt;%</span></span>
<span id="cb512-4"><a href="answers-to-practical-exercises.html#cb512-4"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Binomial =</span> <span class="kw">str_replace</span>(<span class="dt">string =</span> Binomial, <span class="dt">pattern =</span> <span class="st">&quot; &quot;</span>, <span class="dt">replacement =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb512-5"><a href="answers-to-practical-exercises.html#cb512-5"></a><span class="co"># Check it worked</span></span>
<span id="cb512-6"><a href="answers-to-practical-exercises.html#cb512-6"></a><span class="kw">glimpse</span>(primatedata)</span></code></pre></div>
<pre><code>## Rows: 77
## Columns: 9
## $ Order           &lt;chr&gt; &quot;Primates&quot;, &quot;Primates&quot;, &quot;Primates&quot;, &quot;Primates&quot;, …
## $ Family          &lt;chr&gt; &quot;Atelidae&quot;, &quot;Atelidae&quot;, &quot;Atelidae&quot;, &quot;Pitheciidae…
## $ Binomial        &lt;chr&gt; &quot;Ateles_belzebuth&quot;, &quot;Ateles_geoffroyi&quot;, &quot;Ateles_…
## $ AdultBodyMass_g &lt;dbl&gt; 6692.42, 7582.40, 8697.25, 958.13, 558.00, 290.2…
## $ GestationLen_d  &lt;dbl&gt; 138.20, 226.37, 228.18, 164.00, 153.99, 144.00, …
## $ HomeRange_km2   &lt;dbl&gt; 2.28000, 0.73000, 1.36000, 0.02000, 0.32000, 0.0…
## $ MaxLongevity_m  &lt;dbl&gt; 336.0, 327.6, 453.6, 303.6, 214.8, 201.6, 181.2,…
## $ SocialGroupSize &lt;dbl&gt; 14.50, 42.00, 20.00, 2.95, 6.85, 8.55, 6.00, 20.…
## $ SocialStatus    &lt;dbl&gt; 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, …</code></pre>
<div class="sourceCode" id="cb514"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb514-1"><a href="answers-to-practical-exercises.html#cb514-1"></a><span class="co"># Check whether the names match in the data and the tree</span></span>
<span id="cb514-2"><a href="answers-to-practical-exercises.html#cb514-2"></a>check &lt;-<span class="st"> </span><span class="kw">name.check</span>(<span class="dt">phy =</span> primatetree, <span class="dt">data =</span> primatedata, </span>
<span id="cb514-3"><a href="answers-to-practical-exercises.html#cb514-3"></a>                    <span class="dt">data.names =</span> primatedata<span class="op">$</span>Binomial)</span>
<span id="cb514-4"><a href="answers-to-practical-exercises.html#cb514-4"></a><span class="co"># Look at check</span></span>
<span id="cb514-5"><a href="answers-to-practical-exercises.html#cb514-5"></a>check</span></code></pre></div>
<pre><code>## $tree_not_data
##   [1] &quot;Allenopithecus_nigroviridis&quot;                  
##   [2] &quot;Allocebus_trichotis&quot;                          
##   [3] &quot;Alouatta_caraya&quot;                              
##   [4] &quot;Alouatta_sara&quot;                                
##   [5] &quot;Aotus_azarae&quot;                                 
##   [6] &quot;Aotus_azarae_infulatus&quot;                       
##   [7] &quot;Aotus_lemurinus_griseimembra&quot;                 
##   [8] &quot;Aotus_nancymaae&quot;                              
##   [9] &quot;Arctocebus_aureus&quot;                            
##  [10] &quot;Arctocebus_calabarensis&quot;                      
##  [11] &quot;Ateles_fusciceps&quot;                             
##  [12] &quot;Ateles_geoffroyi_ornatus&quot;                     
##  [13] &quot;Ateles_geoffroyi_vellerosus&quot;                  
##  [14] &quot;Avahi_laniger&quot;                                
##  [15] &quot;Avahi_occidentalis&quot;                           
##  [16] &quot;Brachyteles_arachnoides&quot;                      
##  [17] &quot;Bunopithecus_hoolock&quot;                         
##  [18] &quot;Callicebus_donacophilus&quot;                      
##  [19] &quot;Callithrix_(Mico)_emiliae&quot;                    
##  [20] &quot;Callithrix_argentata&quot;                         
##  [21] &quot;Callithrix_aurita&quot;                            
##  [22] &quot;Callithrix_geoffroyi&quot;                         
##  [23] &quot;Callithrix_humeralifera&quot;                      
##  [24] &quot;Callithrix_kuhli&quot;                             
##  [25] &quot;Callithrix_penicillata&quot;                       
##  [26] &quot;Cercocebus_agilis&quot;                            
##  [27] &quot;Cercocebus_atys&quot;                              
##  [28] &quot;Cercocebus_torquatus&quot;                         
##  [29] &quot;Cercopithecus_cephus_cephus&quot;                  
##  [30] &quot;Cercopithecus_cephus_ngottoensis&quot;             
##  [31] &quot;Cercopithecus_diana&quot;                          
##  [32] &quot;Cercopithecus_erythrogaster_erythrogaster&quot;    
##  [33] &quot;Cercopithecus_erythrotis&quot;                     
##  [34] &quot;Cercopithecus_hamlyni&quot;                        
##  [35] &quot;Cercopithecus_lhoesti&quot;                        
##  [36] &quot;Cercopithecus_lowei&quot;                          
##  [37] &quot;Cercopithecus_mona&quot;                           
##  [38] &quot;Cercopithecus_petaurista&quot;                     
##  [39] &quot;Cercopithecus_preussi&quot;                        
##  [40] &quot;Cercopithecus_solatus&quot;                        
##  [41] &quot;Cercopithecus_wolfi&quot;                          
##  [42] &quot;Cheirogaleus_crossleyi&quot;                       
##  [43] &quot;Chlorocebus_aethiops&quot;                         
##  [44] &quot;Chlorocebus_pygerythrus&quot;                      
##  [45] &quot;Chlorocebus_sabaeus&quot;                          
##  [46] &quot;Chlorocebus_tantalus&quot;                         
##  [47] &quot;Colobus_angolensis&quot;                           
##  [48] &quot;Eulemur_albifrons&quot;                            
##  [49] &quot;Eulemur_albocollaris&quot;                         
##  [50] &quot;Eulemur_collaris&quot;                             
##  [51] &quot;Eulemur_macaco_flavifrons&quot;                    
##  [52] &quot;Eulemur_macaco_macaco&quot;                        
##  [53] &quot;Eulemur_rubriventer&quot;                          
##  [54] &quot;Eulemur_rufus&quot;                                
##  [55] &quot;Eulemur_sanfordi&quot;                             
##  [56] &quot;Euoticus_elegantulus&quot;                         
##  [57] &quot;Galago_gallarum&quot;                              
##  [58] &quot;Galago_zanzibaricus&quot;                          
##  [59] &quot;Gorilla_gorilla_gorilla&quot;                      
##  [60] &quot;Hapalemur_alaotrensis&quot;                        
##  [61] &quot;Hapalemur_aureus&quot;                             
##  [62] &quot;Hapalemur_griseus_griseus&quot;                    
##  [63] &quot;Hapalemur_griseus_meridionalis&quot;               
##  [64] &quot;Hapalemur_occidentalis&quot;                       
##  [65] &quot;Homo_sapiens&quot;                                 
##  [66] &quot;Hylobates_agilis&quot;                             
##  [67] &quot;Hylobates_klossii&quot;                            
##  [68] &quot;Hylobates_moloch&quot;                             
##  [69] &quot;Hylobates_muelleri&quot;                           
##  [70] &quot;Indri_indri&quot;                                  
##  [71] &quot;Leontopithecus_chrysomelas&quot;                   
##  [72] &quot;Leontopithecus_chrysopygus&quot;                   
##  [73] &quot;Lepilemur_aeeclis&quot;                            
##  [74] &quot;Lepilemur_ankaranensis&quot;                       
##  [75] &quot;Lepilemur_dorsalis&quot;                           
##  [76] &quot;Lepilemur_edwardsi&quot;                           
##  [77] &quot;Lepilemur_microdon&quot;                           
##  [78] &quot;Lepilemur_mitsinjoensis&quot;                      
##  [79] &quot;Lepilemur_randrianasoli&quot;                      
##  [80] &quot;Lepilemur_ruficaudatus&quot;                       
##  [81] &quot;Lepilemur_sahamalazensis&quot;                     
##  [82] &quot;Lepilemur_seali&quot;                              
##  [83] &quot;Lepilemur_septentrionalis&quot;                    
##  [84] &quot;Lophocebus_aterrimus&quot;                         
##  [85] &quot;Loris_lydekkerianus_grandis&quot;                  
##  [86] &quot;Loris_lydekkerianus_malabaricus&quot;              
##  [87] &quot;Macaca_arctoides&quot;                             
##  [88] &quot;Macaca_assamensis&quot;                            
##  [89] &quot;Macaca_cyclopis&quot;                              
##  [90] &quot;Macaca_hecki&quot;                                 
##  [91] &quot;Macaca_leonina&quot;                               
##  [92] &quot;Macaca_maura&quot;                                 
##  [93] &quot;Macaca_nigra&quot;                                 
##  [94] &quot;Macaca_nigrescens&quot;                            
##  [95] &quot;Macaca_ochreata&quot;                              
##  [96] &quot;Macaca_ochreata_brunnescens&quot;                  
##  [97] &quot;Macaca_pagensis&quot;                              
##  [98] &quot;Macaca_siberu&quot;                                
##  [99] &quot;Macaca_thibetana&quot;                             
## [100] &quot;Macaca_tonkeana&quot;                              
## [101] &quot;Mandrillus_leucophaeus&quot;                       
## [102] &quot;Microcebus_berthae&quot;                           
## [103] &quot;Microcebus_bongolavensis&quot;                     
## [104] &quot;Microcebus_danfossi&quot;                          
## [105] &quot;Microcebus_griseorufus&quot;                       
## [106] &quot;Microcebus_jollyae&quot;                           
## [107] &quot;Microcebus_lehilahytsara&quot;                     
## [108] &quot;Microcebus_lokobensis&quot;                        
## [109] &quot;Microcebus_mittermeieri&quot;                      
## [110] &quot;Microcebus_myoxinus&quot;                          
## [111] &quot;Microcebus_ravelobensis&quot;                      
## [112] &quot;Microcebus_sambiranensis&quot;                     
## [113] &quot;Microcebus_simmonsi&quot;                          
## [114] &quot;Microcebus_tavaratra&quot;                         
## [115] &quot;Nomascus_gabriellae&quot;                          
## [116] &quot;Nomascus_leucogenys&quot;                          
## [117] &quot;Nycticebus_coucang&quot;                           
## [118] &quot;Nycticebus_pygmaeus&quot;                          
## [119] &quot;Pan_troglodytes_schweinfurthii&quot;               
## [120] &quot;Pan_troglodytes_troglodytes&quot;                  
## [121] &quot;Pan_troglodytes_verus&quot;                        
## [122] &quot;Papio_papio&quot;                                  
## [123] &quot;Piliocolobus_badius&quot;                          
## [124] &quot;Pongo_abelii&quot;                                 
## [125] &quot;Pongo_pygmaeus_pygmaeus&quot;                      
## [126] &quot;Presbytis_melalophos&quot;                         
## [127] &quot;Propithecus_coquereli&quot;                        
## [128] &quot;Propithecus_diadema&quot;                          
## [129] &quot;Propithecus_edwardsi&quot;                         
## [130] &quot;Propithecus_tattersalli&quot;                      
## [131] &quot;Pygathrix_nemaeus&quot;                            
## [132] &quot;Rhinopithecus_avunculus&quot;                      
## [133] &quot;Rhinopithecus_bieti&quot;                          
## [134] &quot;Rhinopithecus_brelichi&quot;                       
## [135] &quot;Rhinopithecus_roxellana&quot;                      
## [136] &quot;Rungwecebus_kipunji&quot;                          
## [137] &quot;Saguinus_geoffroyi&quot;                           
## [138] &quot;Saguinus_imperator&quot;                           
## [139] &quot;Saimiri_boliviensis_boliviensis&quot;              
## [140] &quot;Saimiri_oerstedii&quot;                            
## [141] &quot;Trachypithecus_(Trachypithecus)_auratus&quot;      
## [142] &quot;Trachypithecus_(Trachypithecus)_poliocephalus&quot;
## [143] &quot;Trachypithecus_cristatus&quot;                     
## [144] &quot;Trachypithecus_francoisi&quot;                     
## [145] &quot;Trachypithecus_johnii&quot;                        
## [146] &quot;Trachypithecus_phayrei&quot;                       
## [147] &quot;Trachypithecus_pileatus&quot;                      
## [148] &quot;Varecia_rubra&quot;                                
## [149] &quot;Varecia_variegata_variegata&quot;                  
## 
## $data_not_tree
## character(0)</code></pre>
<div class="sourceCode" id="cb516"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb516-1"><a href="answers-to-practical-exercises.html#cb516-1"></a><span class="co"># 149 species are in the tree but not the data</span></span>
<span id="cb516-2"><a href="answers-to-practical-exercises.html#cb516-2"></a><span class="co"># 0 species are in the data but not the tree</span></span>
<span id="cb516-3"><a href="answers-to-practical-exercises.html#cb516-3"></a>  </span>
<span id="cb516-4"><a href="answers-to-practical-exercises.html#cb516-4"></a><span class="co"># Combine and match the tree and data</span></span>
<span id="cb516-5"><a href="answers-to-practical-exercises.html#cb516-5"></a>primatestuff &lt;-<span class="st"> </span><span class="kw">make.treedata</span>(<span class="dt">tree =</span> primatetree,  <span class="dt">data =</span> primatedata, </span>
<span id="cb516-6"><a href="answers-to-practical-exercises.html#cb516-6"></a>                              <span class="dt">name_column =</span> <span class="st">&quot;Binomial&quot;</span>)</span>
<span id="cb516-7"><a href="answers-to-practical-exercises.html#cb516-7"></a></span>
<span id="cb516-8"><a href="answers-to-practical-exercises.html#cb516-8"></a><span class="co"># Look at the tree summary</span></span>
<span id="cb516-9"><a href="answers-to-practical-exercises.html#cb516-9"></a>primatestuff<span class="op">$</span>phy</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 77 tips and 76 internal nodes.
## 
## Tip labels:
##   Cercopithecus_ascanius, Cercopithecus_cephus, Cercopithecus_mitis, Cercopithecus_neglectus, Cercopithecus_nictitans, Erythrocebus_patas, ...
## 
## Rooted; includes branch lengths.</code></pre>
<div class="sourceCode" id="cb518"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb518-1"><a href="answers-to-practical-exercises.html#cb518-1"></a><span class="co"># Look at the data</span></span>
<span id="cb518-2"><a href="answers-to-practical-exercises.html#cb518-2"></a><span class="kw">glimpse</span>(primatestuff<span class="op">$</span>dat)</span></code></pre></div>
<pre><code>## Rows: 77
## Columns: 8
## $ Order           &lt;fct&gt; Primates, Primates, Primates, Primates, Primates…
## $ Family          &lt;fct&gt; Cercopithecidae, Cercopithecidae, Cercopithecida…
## $ AdultBodyMass_g &lt;dbl&gt; 3540.24, 3444.88, 5041.29, 5324.52, 5256.91, 796…
## $ GestationLen_d  &lt;dbl&gt; 148.50, 169.51, 138.39, 172.07, 169.51, 167.20, …
## $ HomeRange_km2   &lt;dbl&gt; 0.16000, 0.24000, 0.10000, 0.06000, 1.15000, 28.…
## $ MaxLongevity_m  &lt;dbl&gt; 339.6, 276.0, 325.2, 315.6, 276.0, 286.8, 370.8,…
## $ SocialGroupSize &lt;dbl&gt; 26.30, 11.00, 16.00, 4.50, 16.00, 28.00, 91.20, …
## $ SocialStatus    &lt;int&gt; 2, 2, 2, 2, 2, 2, 2, 1, 1, 1, 2, 2, 2, 2, 1, 1, …</code></pre>
<div class="sourceCode" id="cb520"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb520-1"><a href="answers-to-practical-exercises.html#cb520-1"></a><span class="co">## There are 77 species in the tree and the data</span></span>
<span id="cb520-2"><a href="answers-to-practical-exercises.html#cb520-2"></a></span>
<span id="cb520-3"><a href="answers-to-practical-exercises.html#cb520-3"></a><span class="co"># Make a new column called tiplabel with the tip labels in it</span></span>
<span id="cb520-4"><a href="answers-to-practical-exercises.html#cb520-4"></a>primatestuff<span class="op">$</span>dat<span class="op">$</span>tiplabel &lt;-<span class="st"> </span>primatestuff<span class="op">$</span>phy<span class="op">$</span>tip.label</span>
<span id="cb520-5"><a href="answers-to-practical-exercises.html#cb520-5"></a></span>
<span id="cb520-6"><a href="answers-to-practical-exercises.html#cb520-6"></a><span class="co"># Save tree as mytree</span></span>
<span id="cb520-7"><a href="answers-to-practical-exercises.html#cb520-7"></a>mytree &lt;-<span class="st"> </span>primatestuff<span class="op">$</span>phy</span>
<span id="cb520-8"><a href="answers-to-practical-exercises.html#cb520-8"></a><span class="co"># Save data as mydata</span></span>
<span id="cb520-9"><a href="answers-to-practical-exercises.html#cb520-9"></a>mydata &lt;-<span class="st"> </span>primatestuff<span class="op">$</span>dat</span>
<span id="cb520-10"><a href="answers-to-practical-exercises.html#cb520-10"></a></span>
<span id="cb520-11"><a href="answers-to-practical-exercises.html#cb520-11"></a><span class="co"># Force mydata to be a data frame</span></span>
<span id="cb520-12"><a href="answers-to-practical-exercises.html#cb520-12"></a>mydata &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(mydata)</span>
<span id="cb520-13"><a href="answers-to-practical-exercises.html#cb520-13"></a><span class="co"># Check mydata is now a dataframe</span></span>
<span id="cb520-14"><a href="answers-to-practical-exercises.html#cb520-14"></a><span class="kw">str</span>(mydata)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    77 obs. of  9 variables:
##  $ Order          : Factor w/ 1 level &quot;Primates&quot;: 1 1 1 1 1 1 1 1 1 1 ...
##  $ Family         : Factor w/ 15 levels &quot;Aotidae&quot;,&quot;Atelidae&quot;,..: 4 4 4 4 4 4 4 5 5 6 ...
##  $ AdultBodyMass_g: num  3540 3445 5041 5325 5257 ...
##  $ GestationLen_d : num  148 170 138 172 170 ...
##  $ HomeRange_km2  : num  0.16 0.24 0.1 0.06 1.15 ...
##  $ MaxLongevity_m : num  340 276 325 316 276 ...
##  $ SocialGroupSize: num  26.3 11 16 4.5 16 28 91.2 1 1 1 ...
##  $ SocialStatus   : int  2 2 2 2 2 2 2 1 1 1 ...
##  $ tiplabel       : chr  &quot;Cercopithecus_ascanius&quot; &quot;Cercopithecus_cephus&quot; &quot;Cercopithecus_mitis&quot; &quot;Cercopithecus_neglectus&quot; ...</code></pre>
<div class="sourceCode" id="cb522"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb522-1"><a href="answers-to-practical-exercises.html#cb522-1"></a><span class="co">## OPTIONAL</span></span>
<span id="cb522-2"><a href="answers-to-practical-exercises.html#cb522-2"></a><span class="co"># Make any required subsets of the tree/data</span></span>
<span id="cb522-3"><a href="answers-to-practical-exercises.html#cb522-3"></a></span>
<span id="cb522-4"><a href="answers-to-practical-exercises.html#cb522-4"></a><span class="co">## OPTIONAL</span></span>
<span id="cb522-5"><a href="answers-to-practical-exercises.html#cb522-5"></a><span class="co"># Write the cleaned data to a new file</span></span>
<span id="cb522-6"><a href="answers-to-practical-exercises.html#cb522-6"></a><span class="co"># write_csv(mydata, path = &quot;data/clean-primate-data.csv&quot;)</span></span>
<span id="cb522-7"><a href="answers-to-practical-exercises.html#cb522-7"></a></span>
<span id="cb522-8"><a href="answers-to-practical-exercises.html#cb522-8"></a><span class="co"># Write the cleaned tree to a new file</span></span>
<span id="cb522-9"><a href="answers-to-practical-exercises.html#cb522-9"></a><span class="co"># write.nexus(mytree, file = &quot;data/clean-primate-tree.nex&quot;)</span></span></code></pre></div>
</div>
<div id="phylogenetic-signal" class="section level2">
<h2><span class="header-section-number">12.4</span> Phylogenetic signal</h2>
<p>In the data folder there is another tree (<code>primate-tree.nex</code>) and dataset (<code>primate-data.csv</code>) for investigating the evolution of primate life-history variables. These data come from the PanTHERIA database <span class="citation">(Jones et al. <a href="#ref-jones2009pantheria" role="doc-biblioref">2009</a>)</span> and 10kTrees <span class="citation">(Arnold, Matthews, and Nunn <a href="#ref-arnold201010ktrees" role="doc-biblioref">2010</a>)</span>.</p>
<p>Read in the tree and data, then prepare them for a PCM analysis (you may have already done this in the previous exercise which should save you some time). Then use <code>phylosig</code> to answer the following questions:</p>
<div class="sourceCode" id="cb523"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb523-1"><a href="answers-to-practical-exercises.html#cb523-1"></a><span class="co"># Load the packages</span></span>
<span id="cb523-2"><a href="answers-to-practical-exercises.html#cb523-2"></a><span class="kw">library</span>(ape)</span>
<span id="cb523-3"><a href="answers-to-practical-exercises.html#cb523-3"></a><span class="kw">library</span>(geiger)</span>
<span id="cb523-4"><a href="answers-to-practical-exercises.html#cb523-4"></a><span class="kw">library</span>(treeplyr)</span>
<span id="cb523-5"><a href="answers-to-practical-exercises.html#cb523-5"></a><span class="kw">library</span>(caper)</span>
<span id="cb523-6"><a href="answers-to-practical-exercises.html#cb523-6"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb523-7"><a href="answers-to-practical-exercises.html#cb523-7"></a><span class="kw">library</span>(phytools)</span>
<span id="cb523-8"><a href="answers-to-practical-exercises.html#cb523-8"></a></span>
<span id="cb523-9"><a href="answers-to-practical-exercises.html#cb523-9"></a><span class="co">### Prepare the tree and the data as shown in the Preparation answer above </span><span class="al">###</span></span>
<span id="cb523-10"><a href="answers-to-practical-exercises.html#cb523-10"></a><span class="co">### This should mean you now have mydata and mytree ready to work with </span><span class="al">###</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>What is <span class="math inline">\(\lambda\)</span> for log gestation length?</li>
</ol>
<div class="sourceCode" id="cb524"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb524-1"><a href="answers-to-practical-exercises.html#cb524-1"></a><span class="co"># First create logGestation containing just log gestation length values</span></span>
<span id="cb524-2"><a href="answers-to-practical-exercises.html#cb524-2"></a>logGestation &lt;-<span class="st"> </span><span class="kw">log</span>(<span class="kw">pull</span>(mydata, GestationLen_d))</span>
<span id="cb524-3"><a href="answers-to-practical-exercises.html#cb524-3"></a><span class="co"># Give log Eye names = species names at the tips of the phylogeny</span></span>
<span id="cb524-4"><a href="answers-to-practical-exercises.html#cb524-4"></a><span class="kw">names</span>(logGestation) &lt;-<span class="st"> </span>mydata<span class="op">$</span>tiplabel</span>
<span id="cb524-5"><a href="answers-to-practical-exercises.html#cb524-5"></a><span class="co"># Look at the first few rows</span></span>
<span id="cb524-6"><a href="answers-to-practical-exercises.html#cb524-6"></a><span class="kw">head</span>(logGestation)</span></code></pre></div>
<pre><code>##  Cercopithecus_ascanius    Cercopithecus_cephus     Cercopithecus_mitis 
##                5.000585                5.132912                4.930076 
## Cercopithecus_neglectus Cercopithecus_nictitans      Erythrocebus_patas 
##                5.147901                5.132912                5.119191</code></pre>
<div class="sourceCode" id="cb526"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb526-1"><a href="answers-to-practical-exercises.html#cb526-1"></a><span class="co"># Estimate lambda</span></span>
<span id="cb526-2"><a href="answers-to-practical-exercises.html#cb526-2"></a>lambdaGestation &lt;-<span class="st"> </span><span class="kw">phylosig</span>(mytree, logGestation, <span class="dt">method =</span> <span class="st">&quot;lambda&quot;</span>, <span class="dt">test =</span> <span class="ot">TRUE</span>)</span>
<span id="cb526-3"><a href="answers-to-practical-exercises.html#cb526-3"></a><span class="co"># Look at the results</span></span>
<span id="cb526-4"><a href="answers-to-practical-exercises.html#cb526-4"></a>lambdaGestation</span></code></pre></div>
<pre><code>## 
## Phylogenetic signal lambda : 0.948326 
## logL(lambda) : 34.9204 
## LR(lambda=0) : 89.026 
## P-value (based on LR test) : 3.89649e-21</code></pre>
<p><em>0.948</em></p>
<ol start="2" style="list-style-type: decimal">
<li>What is <em>K</em> for log gestation length?</li>
</ol>
<div class="sourceCode" id="cb528"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb528-1"><a href="answers-to-practical-exercises.html#cb528-1"></a><span class="co"># Estimate Blomberg’s *K*</span></span>
<span id="cb528-2"><a href="answers-to-practical-exercises.html#cb528-2"></a>KGestation &lt;-<span class="st"> </span><span class="kw">phylosig</span>(mytree, logGestation, <span class="dt">method =</span> <span class="st">&quot;K&quot;</span>, <span class="dt">test =</span> <span class="ot">TRUE</span>, <span class="dt">nsim =</span> <span class="dv">1000</span>)</span>
<span id="cb528-3"><a href="answers-to-practical-exercises.html#cb528-3"></a><span class="co"># Look at the results</span></span>
<span id="cb528-4"><a href="answers-to-practical-exercises.html#cb528-4"></a>KGestation</span></code></pre></div>
<pre><code>## 
## Phylogenetic signal K : 0.775777 
## P-value (based on 1000 randomizations) : 0.001</code></pre>
<p><em>0.776</em></p>
<ol start="3" style="list-style-type: decimal">
<li>What is D for social status?</li>
</ol>
<div class="sourceCode" id="cb530"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb530-1"><a href="answers-to-practical-exercises.html#cb530-1"></a><span class="co"># Prepare the data for use in caper</span></span>
<span id="cb530-2"><a href="answers-to-practical-exercises.html#cb530-2"></a>primate &lt;-<span class="st"> </span><span class="kw">comparative.data</span>(<span class="dt">phy =</span> mytree, <span class="dt">data =</span> mydata, </span>
<span id="cb530-3"><a href="answers-to-practical-exercises.html#cb530-3"></a>                            <span class="dt">names.col =</span> tiplabel, <span class="dt">vcv =</span> <span class="ot">TRUE</span>, </span>
<span id="cb530-4"><a href="answers-to-practical-exercises.html#cb530-4"></a>                            <span class="dt">na.omit =</span> <span class="ot">FALSE</span>, <span class="dt">warn.dropped =</span> <span class="ot">TRUE</span>)</span>
<span id="cb530-5"><a href="answers-to-practical-exercises.html#cb530-5"></a></span>
<span id="cb530-6"><a href="answers-to-practical-exercises.html#cb530-6"></a><span class="co"># Force the phylogeny to have polytomies again nrather than zero length branches</span></span>
<span id="cb530-7"><a href="answers-to-practical-exercises.html#cb530-7"></a>primate<span class="op">$</span>phy &lt;-<span class="st"> </span><span class="kw">di2multi</span>(primate<span class="op">$</span>phy)</span>
<span id="cb530-8"><a href="answers-to-practical-exercises.html#cb530-8"></a></span>
<span id="cb530-9"><a href="answers-to-practical-exercises.html#cb530-9"></a><span class="co"># Estimate D</span></span>
<span id="cb530-10"><a href="answers-to-practical-exercises.html#cb530-10"></a>Dsocial &lt;-<span class="st"> </span><span class="kw">phylo.d</span>(<span class="dt">data =</span> primate, <span class="dt">names.col =</span> tiplabel, <span class="dt">binvar =</span> SocialStatus, </span>
<span id="cb530-11"><a href="answers-to-practical-exercises.html#cb530-11"></a>                  <span class="dt">permut =</span> <span class="dv">1000</span>)</span>
<span id="cb530-12"><a href="answers-to-practical-exercises.html#cb530-12"></a></span>
<span id="cb530-13"><a href="answers-to-practical-exercises.html#cb530-13"></a><span class="co"># Look at the output</span></span>
<span id="cb530-14"><a href="answers-to-practical-exercises.html#cb530-14"></a>Dsocial</span></code></pre></div>
<pre><code>## 
## Calculation of D statistic for the phylogenetic structure of a binary variable
## 
##   Data :  mydata
##   Binary variable :  SocialStatus
##   Counts of states:  1 = 14
##                      2 = 63
##   Phylogeny :  mytree
##   Number of permutations :  1000
## 
## Estimated D :  -0.7228336
## Probability of E(D) resulting from no (random) phylogenetic structure :  0
## Probability of E(D) resulting from Brownian phylogenetic structure    :  0.969</code></pre>
<p><em>~ -0.72, this is a randomisation test so may vary</em></p>
</div>
<div id="pgls" class="section level2">
<h2><span class="header-section-number">12.5</span> PGLS</h2>
<p>In the data folder there is another tree (<code>primate-tree.nex</code>) and dataset (<code>primate-data.csv</code>) for investigating the evolution of primate life-history variables. These data come from the PanTHERIA database <span class="citation">(Jones et al. <a href="#ref-jones2009pantheria" role="doc-biblioref">2009</a>)</span> and 10kTrees <span class="citation">(Arnold, Matthews, and Nunn <a href="#ref-arnold201010ktrees" role="doc-biblioref">2010</a>)</span>.</p>
<p>Let’s investigate the relationship between gestation length in Primates and their body size. Theory leads us to believe that bigger mammals have longer gestation lengths (compare mice to elephants), but is this true in Primates?</p>
<p>Read in the tree and data, then prepare them for a PCM analysis (you may have already done this in a previous exercise which should save you some time). Fit a PGLS model to investigate the relationship between log gestation length (<code>y = log(GestationLen_d)</code>) and log body size (<code>x = log(AdultBodyMass_g)</code>) in Primates. Don’t forget to look at the model diagnostics!</p>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="answers-to-practical-exercises.html#cb532-1"></a><span class="co"># Load the packages</span></span>
<span id="cb532-2"><a href="answers-to-practical-exercises.html#cb532-2"></a><span class="kw">library</span>(ape)</span>
<span id="cb532-3"><a href="answers-to-practical-exercises.html#cb532-3"></a><span class="kw">library</span>(geiger)</span>
<span id="cb532-4"><a href="answers-to-practical-exercises.html#cb532-4"></a><span class="kw">library</span>(treeplyr)</span>
<span id="cb532-5"><a href="answers-to-practical-exercises.html#cb532-5"></a><span class="kw">library</span>(caper)</span>
<span id="cb532-6"><a href="answers-to-practical-exercises.html#cb532-6"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb532-7"><a href="answers-to-practical-exercises.html#cb532-7"></a></span>
<span id="cb532-8"><a href="answers-to-practical-exercises.html#cb532-8"></a><span class="co">### Prepare the tree and the data as shown in the Preparation answer above </span><span class="al">###</span></span>
<span id="cb532-9"><a href="answers-to-practical-exercises.html#cb532-9"></a><span class="co">### This should mean you now have mydata and mytree ready to work with </span><span class="al">###</span></span>
<span id="cb532-10"><a href="answers-to-practical-exercises.html#cb532-10"></a></span>
<span id="cb532-11"><a href="answers-to-practical-exercises.html#cb532-11"></a><span class="co"># Prepare the data for use in caper</span></span>
<span id="cb532-12"><a href="answers-to-practical-exercises.html#cb532-12"></a>primate &lt;-<span class="st"> </span><span class="kw">comparative.data</span>(<span class="dt">phy =</span> mytree, <span class="dt">data =</span> mydata, </span>
<span id="cb532-13"><a href="answers-to-practical-exercises.html#cb532-13"></a>                            <span class="dt">names.col =</span> tiplabel, <span class="dt">vcv =</span> <span class="ot">TRUE</span>, </span>
<span id="cb532-14"><a href="answers-to-practical-exercises.html#cb532-14"></a>                            <span class="dt">na.omit =</span> <span class="ot">FALSE</span>, <span class="dt">warn.dropped =</span> <span class="ot">TRUE</span>)</span>
<span id="cb532-15"><a href="answers-to-practical-exercises.html#cb532-15"></a></span>
<span id="cb532-16"><a href="answers-to-practical-exercises.html#cb532-16"></a><span class="co"># Check the list of dropped species is what you expected</span></span>
<span id="cb532-17"><a href="answers-to-practical-exercises.html#cb532-17"></a>primate<span class="op">$</span>dropped<span class="op">$</span>tips</span></code></pre></div>
<pre><code>## character(0)</code></pre>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="answers-to-practical-exercises.html#cb534-1"></a>primate<span class="op">$</span>dropped<span class="op">$</span>unmatched.rows</span></code></pre></div>
<pre><code>## character(0)</code></pre>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="answers-to-practical-exercises.html#cb536-1"></a><span class="co"># Fit a PGLS model</span></span>
<span id="cb536-2"><a href="answers-to-practical-exercises.html#cb536-2"></a>model.pgls &lt;-<span class="st"> </span><span class="kw">pgls</span>(<span class="kw">log</span>(GestationLen_d) <span class="op">~</span><span class="st"> </span><span class="kw">log</span>(AdultBodyMass_g), </span>
<span id="cb536-3"><a href="answers-to-practical-exercises.html#cb536-3"></a>                   <span class="dt">data =</span> primate, <span class="dt">lambda =</span> <span class="st">&quot;ML&quot;</span>, <span class="dt">bounds =</span> <span class="kw">list</span>(<span class="dt">lambda =</span> <span class="kw">c</span>(<span class="fl">0.01</span>, <span class="dv">1</span>)))</span>
<span id="cb536-4"><a href="answers-to-practical-exercises.html#cb536-4"></a></span>
<span id="cb536-5"><a href="answers-to-practical-exercises.html#cb536-5"></a><span class="co">### Model diagnostics</span></span>
<span id="cb536-6"><a href="answers-to-practical-exercises.html#cb536-6"></a><span class="co"># Make a plotting window with 4 panes as there are four plots</span></span>
<span id="cb536-7"><a href="answers-to-practical-exercises.html#cb536-7"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb536-8"><a href="answers-to-practical-exercises.html#cb536-8"></a><span class="co"># Now plot the model diagnostics</span></span>
<span id="cb536-9"><a href="answers-to-practical-exercises.html#cb536-9"></a><span class="kw">plot</span>(model.pgls)</span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-340-1.png" width="672" /></p>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="answers-to-practical-exercises.html#cb537-1"></a><span class="co"># Return the plot window to one pane for later plotting</span></span>
<span id="cb537-2"><a href="answers-to-practical-exercises.html#cb537-2"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb537-3"><a href="answers-to-practical-exercises.html#cb537-3"></a></span>
<span id="cb537-4"><a href="answers-to-practical-exercises.html#cb537-4"></a><span class="co"># Look at the model outputs</span></span>
<span id="cb537-5"><a href="answers-to-practical-exercises.html#cb537-5"></a><span class="kw">summary</span>(model.pgls)</span></code></pre></div>
<pre><code>## 
## Call:
## pgls(formula = log(GestationLen_d) ~ log(AdultBodyMass_g), data = primate, 
##     lambda = &quot;ML&quot;, bounds = list(lambda = c(0.01, 1)))
## 
## Residuals:
##       Min        1Q    Median        3Q       Max 
## -0.097244 -0.019277 -0.001562  0.009037  0.050542 
## 
## Branch length transformations:
## 
## kappa  [Fix]  : 1.000
## lambda [ ML]  : 0.892
##    lower bound : 0.010, p = 3.3196e-14
##    upper bound : 1.000, p = 0.00046393
##    95.0% CI   : (0.753, 0.967)
## delta  [Fix]  : 1.000
## 
## Coefficients:
##                      Estimate Std. Error t value  Pr(&gt;|t|)    
## (Intercept)          4.290229   0.160355 26.7546 &lt; 2.2e-16 ***
## log(AdultBodyMass_g) 0.104864   0.019628  5.3426 9.479e-07 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.0261 on 75 degrees of freedom
## Multiple R-squared: 0.2757,  Adjusted R-squared: 0.266 
## F-statistic: 28.54 on 1 and 75 DF,  p-value: 9.479e-07</code></pre>
<div class="sourceCode" id="cb539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb539-1"><a href="answers-to-practical-exercises.html#cb539-1"></a><span class="co"># Plot the results</span></span>
<span id="cb539-2"><a href="answers-to-practical-exercises.html#cb539-2"></a><span class="kw">ggplot</span>(mydata, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">log</span>(AdultBodyMass_g), </span>
<span id="cb539-3"><a href="answers-to-practical-exercises.html#cb539-3"></a>                     <span class="dt">y =</span> <span class="kw">log</span>(GestationLen_d))) <span class="op">+</span></span>
<span id="cb539-4"><a href="answers-to-practical-exercises.html#cb539-4"></a><span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span></span>
<span id="cb539-5"><a href="answers-to-practical-exercises.html#cb539-5"></a><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">slope =</span> <span class="kw">coefficients</span>(model.pgls)[<span class="dv">2</span>], </span>
<span id="cb539-6"><a href="answers-to-practical-exercises.html#cb539-6"></a>              <span class="dt">intercept =</span> <span class="kw">coefficients</span>(model.pgls)[<span class="dv">1</span>]) <span class="op">+</span></span>
<span id="cb539-7"><a href="answers-to-practical-exercises.html#cb539-7"></a><span class="st">  </span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-340-2.png" width="672" /></p>
<p>Then answer the following questions.</p>
<ol style="list-style-type: decimal">
<li>What is the maximum likelihood estimate of <span class="math inline">\(\lambda\)</span> in the model?</li>
</ol>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="answers-to-practical-exercises.html#cb540-1"></a><span class="co"># Look at the model outputs. Lambda is near the middle</span></span>
<span id="cb540-2"><a href="answers-to-practical-exercises.html#cb540-2"></a><span class="kw">summary</span>(model.pgls)</span></code></pre></div>
<pre><code>## 
## Call:
## pgls(formula = log(GestationLen_d) ~ log(AdultBodyMass_g), data = primate, 
##     lambda = &quot;ML&quot;, bounds = list(lambda = c(0.01, 1)))
## 
## Residuals:
##       Min        1Q    Median        3Q       Max 
## -0.097244 -0.019277 -0.001562  0.009037  0.050542 
## 
## Branch length transformations:
## 
## kappa  [Fix]  : 1.000
## lambda [ ML]  : 0.892
##    lower bound : 0.010, p = 3.3196e-14
##    upper bound : 1.000, p = 0.00046393
##    95.0% CI   : (0.753, 0.967)
## delta  [Fix]  : 1.000
## 
## Coefficients:
##                      Estimate Std. Error t value  Pr(&gt;|t|)    
## (Intercept)          4.290229   0.160355 26.7546 &lt; 2.2e-16 ***
## log(AdultBodyMass_g) 0.104864   0.019628  5.3426 9.479e-07 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.0261 on 75 degrees of freedom
## Multiple R-squared: 0.2757,  Adjusted R-squared: 0.266 
## F-statistic: 28.54 on 1 and 75 DF,  p-value: 9.479e-07</code></pre>
<p><em>0.892</em></p>
<ol start="2" style="list-style-type: decimal">
<li>What are the confidence intervals on the maximum likelihood estimate of <span class="math inline">\(\lambda\)</span>. How would you interpret these?</li>
</ol>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="answers-to-practical-exercises.html#cb542-1"></a><span class="co"># Extract the confidence intervals on lambda</span></span>
<span id="cb542-2"><a href="answers-to-practical-exercises.html#cb542-2"></a><span class="kw">pgls.confint</span>(model.pgls, <span class="st">&quot;lambda&quot;</span>)<span class="op">$</span>ci.val</span></code></pre></div>
<pre><code>## [1] 0.753434 0.966543</code></pre>
<p><em>0.753434, 0.966543. We are 95% certain that lambda is between these two values.</em></p>
<ol start="3" style="list-style-type: decimal">
<li>Plot the <span class="math inline">\(\lambda\)</span> profile for the the maximum likelihood estimate of <span class="math inline">\(\lambda\)</span>. What does it look like?</li>
</ol>
<div class="sourceCode" id="cb544"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb544-1"><a href="answers-to-practical-exercises.html#cb544-1"></a><span class="co"># Create a likelihood profile of the lambda estimate</span></span>
<span id="cb544-2"><a href="answers-to-practical-exercises.html#cb544-2"></a>lambda.profile &lt;-<span class="st"> </span><span class="kw">pgls.profile</span>(model.pgls, <span class="st">&quot;lambda&quot;</span>)</span>
<span id="cb544-3"><a href="answers-to-practical-exercises.html#cb544-3"></a><span class="co"># Plot the likelihood profile</span></span>
<span id="cb544-4"><a href="answers-to-practical-exercises.html#cb544-4"></a><span class="kw">plot</span>(lambda.profile)</span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-343-1.png" width="672" /></p>
<ol start="4" style="list-style-type: decimal">
<li>Is there a significant relationship between log gestation length and log body size? What is the slope of this relationship?</li>
</ol>
<div class="sourceCode" id="cb545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb545-1"><a href="answers-to-practical-exercises.html#cb545-1"></a><span class="co"># Look at the model outputs. </span></span>
<span id="cb545-2"><a href="answers-to-practical-exercises.html#cb545-2"></a><span class="kw">summary</span>(model.pgls)</span></code></pre></div>
<pre><code>## 
## Call:
## pgls(formula = log(GestationLen_d) ~ log(AdultBodyMass_g), data = primate, 
##     lambda = &quot;ML&quot;, bounds = list(lambda = c(0.01, 1)))
## 
## Residuals:
##       Min        1Q    Median        3Q       Max 
## -0.097244 -0.019277 -0.001562  0.009037  0.050542 
## 
## Branch length transformations:
## 
## kappa  [Fix]  : 1.000
## lambda [ ML]  : 0.892
##    lower bound : 0.010, p = 3.3196e-14
##    upper bound : 1.000, p = 0.00046393
##    95.0% CI   : (0.753, 0.967)
## delta  [Fix]  : 1.000
## 
## Coefficients:
##                      Estimate Std. Error t value  Pr(&gt;|t|)    
## (Intercept)          4.290229   0.160355 26.7546 &lt; 2.2e-16 ***
## log(AdultBodyMass_g) 0.104864   0.019628  5.3426 9.479e-07 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.0261 on 75 degrees of freedom
## Multiple R-squared: 0.2757,  Adjusted R-squared: 0.266 
## F-statistic: 28.54 on 1 and 75 DF,  p-value: 9.479e-07</code></pre>
<p><em>Yes. p &lt; 0.001. Slope = 0.105 <span class="math inline">\(\pm\)</span> 0.019</em></p>
</div>
<div id="models-of-evolution-with-continuous-traits" class="section level2">
<h2><span class="header-section-number">12.6</span> Models of evolution with continuous traits</h2>
<p>In the data folder there is another tree (<code>primate-tree.nex</code>) and dataset (<code>primate-data.csv</code>) for investigating the evolution of primate life-history variables. These data come from the PanTHERIA database <span class="citation">(Jones et al. <a href="#ref-jones2009pantheria" role="doc-biblioref">2009</a>)</span> and 10kTrees <span class="citation">(Arnold, Matthews, and Nunn <a href="#ref-arnold201010ktrees" role="doc-biblioref">2010</a>)</span>.</p>
<p>Read in the tree and data, then prepare them for a PCM analysis (you may have already done this in a previous exercise which should save you some time). Fit Brownian and OU models to log adult body mass for Primates (<code>AdultBodyMass_g</code>). Then answer the following questions.</p>
<div class="sourceCode" id="cb547"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb547-1"><a href="answers-to-practical-exercises.html#cb547-1"></a><span class="co"># Load the packages</span></span>
<span id="cb547-2"><a href="answers-to-practical-exercises.html#cb547-2"></a><span class="kw">library</span>(ape)</span>
<span id="cb547-3"><a href="answers-to-practical-exercises.html#cb547-3"></a><span class="kw">library</span>(geiger)</span>
<span id="cb547-4"><a href="answers-to-practical-exercises.html#cb547-4"></a><span class="kw">library</span>(treeplyr)</span>
<span id="cb547-5"><a href="answers-to-practical-exercises.html#cb547-5"></a><span class="kw">library</span>(caper)</span>
<span id="cb547-6"><a href="answers-to-practical-exercises.html#cb547-6"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb547-7"><a href="answers-to-practical-exercises.html#cb547-7"></a></span>
<span id="cb547-8"><a href="answers-to-practical-exercises.html#cb547-8"></a><span class="co">### Prepare the tree and the data as shown in the Preparation answer above </span><span class="al">###</span></span>
<span id="cb547-9"><a href="answers-to-practical-exercises.html#cb547-9"></a><span class="co">### This should mean you now have mydata and mytree ready to work with </span><span class="al">###</span></span>
<span id="cb547-10"><a href="answers-to-practical-exercises.html#cb547-10"></a></span>
<span id="cb547-11"><a href="answers-to-practical-exercises.html#cb547-11"></a><span class="co"># Create logBodyMass containing just log body mass values</span></span>
<span id="cb547-12"><a href="answers-to-practical-exercises.html#cb547-12"></a>logBodyMass &lt;-<span class="st"> </span><span class="kw">log</span>(<span class="kw">pull</span>(mydata, AdultBodyMass_g))</span>
<span id="cb547-13"><a href="answers-to-practical-exercises.html#cb547-13"></a><span class="co"># Give log body mass names = species names at the tips of the phylogeny</span></span>
<span id="cb547-14"><a href="answers-to-practical-exercises.html#cb547-14"></a><span class="kw">names</span>(logBodyMass) &lt;-<span class="st"> </span>mydata<span class="op">$</span>tiplabel</span>
<span id="cb547-15"><a href="answers-to-practical-exercises.html#cb547-15"></a><span class="co"># Look at the first few rows</span></span>
<span id="cb547-16"><a href="answers-to-practical-exercises.html#cb547-16"></a><span class="kw">head</span>(logBodyMass)</span></code></pre></div>
<pre><code>##  Cercopithecus_ascanius    Cercopithecus_cephus     Cercopithecus_mitis 
##                8.171950                8.144644                8.525417 
## Cercopithecus_neglectus Cercopithecus_nictitans      Erythrocebus_patas 
##                8.580078                8.567299                8.982975</code></pre>
<div class="sourceCode" id="cb549"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb549-1"><a href="answers-to-practical-exercises.html#cb549-1"></a><span class="co"># Fit the Brownian model</span></span>
<span id="cb549-2"><a href="answers-to-practical-exercises.html#cb549-2"></a>BM &lt;-<span class="st"> </span><span class="kw">fitContinuous</span>(mytree, logBodyMass, <span class="dt">model =</span> <span class="kw">c</span>(<span class="st">&quot;BM&quot;</span>))</span>
<span id="cb549-3"><a href="answers-to-practical-exercises.html#cb549-3"></a><span class="co"># Fit the OU model</span></span>
<span id="cb549-4"><a href="answers-to-practical-exercises.html#cb549-4"></a>OU &lt;-<span class="st"> </span><span class="kw">fitContinuous</span>(mytree, logBodyMass, <span class="dt">model =</span> <span class="kw">c</span>(<span class="st">&quot;OU&quot;</span>))</span></code></pre></div>
<ol style="list-style-type: decimal">
<li>Which model fits best?</li>
</ol>
<div class="sourceCode" id="cb550"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb550-1"><a href="answers-to-practical-exercises.html#cb550-1"></a><span class="co"># Could look at AIC weights...</span></span>
<span id="cb550-2"><a href="answers-to-practical-exercises.html#cb550-2"></a>aic.scores &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">c</span>(BM<span class="op">$</span>opt<span class="op">$</span>aic, OU<span class="op">$</span>opt<span class="op">$</span>aic), <span class="kw">c</span>(<span class="st">&quot;BM&quot;</span>,<span class="st">&quot;OU&quot;</span>))</span>
<span id="cb550-3"><a href="answers-to-practical-exercises.html#cb550-3"></a><span class="kw">aicw</span>(aic.scores)</span></code></pre></div>
<pre><code>##         fit delta         w
## BM 160.1921     0 0.7310586
## OU 162.1921     2 0.2689414</code></pre>
<div class="sourceCode" id="cb552"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb552-1"><a href="answers-to-practical-exercises.html#cb552-1"></a><span class="co"># Or just the raw AIC values</span></span>
<span id="cb552-2"><a href="answers-to-practical-exercises.html#cb552-2"></a>BM<span class="op">$</span>opt<span class="op">$</span>aic</span></code></pre></div>
<pre><code>## [1] 160.1921</code></pre>
<div class="sourceCode" id="cb554"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb554-1"><a href="answers-to-practical-exercises.html#cb554-1"></a>OU<span class="op">$</span>opt<span class="op">$</span>aic</span></code></pre></div>
<pre><code>## [1] 162.1921</code></pre>
<p><em>BM but only by a small amount</em></p>
<ol start="2" style="list-style-type: decimal">
<li>What is the rate of evolution for the Brownian model?</li>
</ol>
<div class="sourceCode" id="cb556"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb556-1"><a href="answers-to-practical-exercises.html#cb556-1"></a><span class="co"># Look at the whole output</span></span>
<span id="cb556-2"><a href="answers-to-practical-exercises.html#cb556-2"></a>BM</span></code></pre></div>
<pre><code>## GEIGER-fitted comparative model of continuous data
##  fitted &#39;BM&#39; model parameters:
##  sigsq = 0.028655
##  z0 = 6.773956
## 
##  model summary:
##  log-likelihood = -78.096042
##  AIC = 160.192084
##  AICc = 160.354246
##  free parameters = 2
## 
## Convergence diagnostics:
##  optimization iterations = 100
##  failed iterations = 0
##  number of iterations with same best fit = 100
##  frequency of best fit = 1.00
## 
##  object summary:
##  &#39;lik&#39; -- likelihood function
##  &#39;bnd&#39; -- bounds for likelihood search
##  &#39;res&#39; -- optimization iteration summary
##  &#39;opt&#39; -- maximum likelihood parameter estimates</code></pre>
<div class="sourceCode" id="cb558"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb558-1"><a href="answers-to-practical-exercises.html#cb558-1"></a><span class="co"># Or just the one parameter</span></span>
<span id="cb558-2"><a href="answers-to-practical-exercises.html#cb558-2"></a>BM<span class="op">$</span>opt<span class="op">$</span>sigsq</span></code></pre></div>
<pre><code>## [1] 0.0286548</code></pre>
<p><em>0.028655</em></p>
<ol start="3" style="list-style-type: decimal">
<li>What is the strength of pull towards the long-term mean in the OU model? What does this value suggest?</li>
</ol>
<div class="sourceCode" id="cb560"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb560-1"><a href="answers-to-practical-exercises.html#cb560-1"></a><span class="co"># Look at the whole output</span></span>
<span id="cb560-2"><a href="answers-to-practical-exercises.html#cb560-2"></a>OU</span></code></pre></div>
<pre><code>## GEIGER-fitted comparative model of continuous data
##  fitted &#39;OU&#39; model parameters:
##  alpha = 0.000000
##  sigsq = 0.028655
##  z0 = 6.773956
## 
##  model summary:
##  log-likelihood = -78.096042
##  AIC = 162.192084
##  AICc = 162.520851
##  free parameters = 3
## 
## Convergence diagnostics:
##  optimization iterations = 100
##  failed iterations = 0
##  number of iterations with same best fit = 54
##  frequency of best fit = 0.54
## 
##  object summary:
##  &#39;lik&#39; -- likelihood function
##  &#39;bnd&#39; -- bounds for likelihood search
##  &#39;res&#39; -- optimization iteration summary
##  &#39;opt&#39; -- maximum likelihood parameter estimates</code></pre>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="answers-to-practical-exercises.html#cb562-1"></a><span class="co"># Or just the one parameter</span></span>
<span id="cb562-2"><a href="answers-to-practical-exercises.html#cb562-2"></a>OU<span class="op">$</span>opt<span class="op">$</span>alpha</span></code></pre></div>
<pre><code>## [1] 1.153003e-193</code></pre>
<p><em>8.080092e-49, i.e. very close to zero which makes it very close to a Brownian model</em></p>
</div>
<div id="models-of-evolution-with-discrete-traits" class="section level2">
<h2><span class="header-section-number">12.7</span> Models of evolution with discrete traits</h2>
<p>In the data folder there is another tree (<code>primate-tree.nex</code>) and dataset (<code>primate-data.csv</code>) for investigating the evolution of primate life-history variables. These data come from the PanTHERIA database <span class="citation">(Jones et al. <a href="#ref-jones2009pantheria" role="doc-biblioref">2009</a>)</span> and 10kTrees <span class="citation">(Arnold, Matthews, and Nunn <a href="#ref-arnold201010ktrees" role="doc-biblioref">2010</a>)</span>.</p>
<p>Read in the tree and data, then prepare them for a PCM analysis (you may have already done this in a previous exercise which should save you some time). Fit equal rates, symmetric rates and all rates different models to the social status variable for Primates (<code>SocialStatus</code>). Note that social status has two states: non-social (1) or social (2).</p>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="answers-to-practical-exercises.html#cb564-1"></a><span class="co"># Load the packages</span></span>
<span id="cb564-2"><a href="answers-to-practical-exercises.html#cb564-2"></a><span class="kw">library</span>(ape)</span>
<span id="cb564-3"><a href="answers-to-practical-exercises.html#cb564-3"></a><span class="kw">library</span>(geiger)</span>
<span id="cb564-4"><a href="answers-to-practical-exercises.html#cb564-4"></a><span class="kw">library</span>(treeplyr)</span>
<span id="cb564-5"><a href="answers-to-practical-exercises.html#cb564-5"></a><span class="kw">library</span>(caper)</span>
<span id="cb564-6"><a href="answers-to-practical-exercises.html#cb564-6"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb564-7"><a href="answers-to-practical-exercises.html#cb564-7"></a></span>
<span id="cb564-8"><a href="answers-to-practical-exercises.html#cb564-8"></a><span class="co">### Prepare the tree and the data as shown in the Preparation answer above </span><span class="al">###</span></span>
<span id="cb564-9"><a href="answers-to-practical-exercises.html#cb564-9"></a><span class="co">### This should mean you now have mydata and mytree ready to work with </span><span class="al">###</span></span>
<span id="cb564-10"><a href="answers-to-practical-exercises.html#cb564-10"></a></span>
<span id="cb564-11"><a href="answers-to-practical-exercises.html#cb564-11"></a><span class="co"># Create Social containing just social status values</span></span>
<span id="cb564-12"><a href="answers-to-practical-exercises.html#cb564-12"></a>Social &lt;-<span class="st"> </span><span class="kw">pull</span>(mydata, SocialStatus)</span>
<span id="cb564-13"><a href="answers-to-practical-exercises.html#cb564-13"></a><span class="co"># Give social status names = species names at the tips of the phylogeny</span></span>
<span id="cb564-14"><a href="answers-to-practical-exercises.html#cb564-14"></a><span class="kw">names</span>(Social) &lt;-<span class="st"> </span>mydata<span class="op">$</span>tiplabel</span>
<span id="cb564-15"><a href="answers-to-practical-exercises.html#cb564-15"></a><span class="co"># Look at the first few rows</span></span>
<span id="cb564-16"><a href="answers-to-practical-exercises.html#cb564-16"></a><span class="kw">head</span>(Social)</span></code></pre></div>
<pre><code>##  Cercopithecus_ascanius    Cercopithecus_cephus     Cercopithecus_mitis 
##                       2                       2                       2 
## Cercopithecus_neglectus Cercopithecus_nictitans      Erythrocebus_patas 
##                       2                       2                       2</code></pre>
<div class="sourceCode" id="cb566"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb566-1"><a href="answers-to-practical-exercises.html#cb566-1"></a><span class="co"># Fit the ER model</span></span>
<span id="cb566-2"><a href="answers-to-practical-exercises.html#cb566-2"></a>equal &lt;-<span class="st"> </span><span class="kw">fitDiscrete</span>(mytree, Social, <span class="dt">model =</span> <span class="st">&quot;ER&quot;</span>)</span>
<span id="cb566-3"><a href="answers-to-practical-exercises.html#cb566-3"></a><span class="co"># Fit the ARD model</span></span>
<span id="cb566-4"><a href="answers-to-practical-exercises.html#cb566-4"></a>ard &lt;-<span class="st"> </span><span class="kw">fitDiscrete</span>(mytree, Social, <span class="dt">model =</span> <span class="st">&quot;ARD&quot;</span>)</span></code></pre></div>
<p>Then answer the following questions.</p>
<ol style="list-style-type: decimal">
<li>Which model fits best?</li>
</ol>
<div class="sourceCode" id="cb567"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb567-1"><a href="answers-to-practical-exercises.html#cb567-1"></a><span class="co"># Using AIC weights</span></span>
<span id="cb567-2"><a href="answers-to-practical-exercises.html#cb567-2"></a>aic.scores &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">c</span>(equal<span class="op">$</span>opt<span class="op">$</span>aic, ard<span class="op">$</span>opt<span class="op">$</span>aic), </span>
<span id="cb567-3"><a href="answers-to-practical-exercises.html#cb567-3"></a>                       <span class="kw">c</span>(<span class="st">&quot;equal&quot;</span>, <span class="st">&quot;different&quot;</span>))</span>
<span id="cb567-4"><a href="answers-to-practical-exercises.html#cb567-4"></a><span class="kw">aicw</span>(aic.scores)</span></code></pre></div>
<pre><code>##                fit    delta          w
## equal     38.98307 5.313651 0.06556956
## different 33.66942 0.000000 0.93443044</code></pre>
<div class="sourceCode" id="cb569"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb569-1"><a href="answers-to-practical-exercises.html#cb569-1"></a><span class="co"># Or just the raw AIC values</span></span>
<span id="cb569-2"><a href="answers-to-practical-exercises.html#cb569-2"></a>equal<span class="op">$</span>opt<span class="op">$</span>aic</span></code></pre></div>
<pre><code>## [1] 38.98307</code></pre>
<div class="sourceCode" id="cb571"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb571-1"><a href="answers-to-practical-exercises.html#cb571-1"></a>ard<span class="op">$</span>opt<span class="op">$</span>aic</span></code></pre></div>
<pre><code>## [1] 33.66942</code></pre>
<p><em>All rates different</em></p>
<ol start="2" style="list-style-type: decimal">
<li>What is the transition rate from non-social (1) to social (2), and vice versa, in the all rates different model? What does this mean biologically?</li>
</ol>
<div class="sourceCode" id="cb573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb573-1"><a href="answers-to-practical-exercises.html#cb573-1"></a><span class="co"># Could look at the model output</span></span>
<span id="cb573-2"><a href="answers-to-practical-exercises.html#cb573-2"></a>ard</span></code></pre></div>
<pre><code>## GEIGER-fitted comparative model of discrete data
##  fitted Q matrix:
##                   1             2
##     1 -9.264914e-03  9.264914e-03
##     2  1.476682e-17 -1.476682e-17
## 
##  model summary:
##  log-likelihood = -14.834711
##  AIC = 33.669423
##  AICc = 33.831585
##  free parameters = 2
## 
## Convergence diagnostics:
##  optimization iterations = 100
##  failed iterations = 0
##  number of iterations with same best fit = 44
##  frequency of best fit = 0.44
## 
##  object summary:
##  &#39;lik&#39; -- likelihood function
##  &#39;bnd&#39; -- bounds for likelihood search
##  &#39;res&#39; -- optimization iteration summary
##  &#39;opt&#39; -- maximum likelihood parameter estimates</code></pre>
<div class="sourceCode" id="cb575"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb575-1"><a href="answers-to-practical-exercises.html#cb575-1"></a><span class="co"># Or plot the Q matrix as network</span></span>
<span id="cb575-2"><a href="answers-to-practical-exercises.html#cb575-2"></a><span class="kw">plot.gfit</span>(ard)</span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-351-1.png" width="672" /></p>
<p><em>1 to 2 is ~0.009, 2 to 1 is very close to zero. Transitions from social (2) to non-social (1) behaviour are extremely rare.</em></p>
<ol start="3" style="list-style-type: decimal">
<li>Plot ancestral state estimates of social status (from the ARD model) on the phylogeny. Is the root node estimated as being non-social or social?</li>
</ol>
<div class="sourceCode" id="cb576"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb576-1"><a href="answers-to-practical-exercises.html#cb576-1"></a><span class="co"># Ancestral state estimation (ignore the warnings)</span></span>
<span id="cb576-2"><a href="answers-to-practical-exercises.html#cb576-2"></a>ancestral_social &lt;-<span class="st"> </span><span class="kw">ace</span>(Social, mytree, <span class="dt">type =</span> <span class="st">&quot;discrete&quot;</span>, <span class="dt">model =</span> <span class="st">&quot;ARD&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in sqrt(diag(solve(h))): NaNs produced</code></pre>
<div class="sourceCode" id="cb578"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb578-1"><a href="answers-to-practical-exercises.html#cb578-1"></a><span class="co"># Choose colours</span></span>
<span id="cb578-2"><a href="answers-to-practical-exercises.html#cb578-2"></a>social_colours &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;deeppink&quot;</span>, <span class="st">&quot;goldenrod&quot;</span>)</span>
<span id="cb578-3"><a href="answers-to-practical-exercises.html#cb578-3"></a></span>
<span id="cb578-4"><a href="answers-to-practical-exercises.html#cb578-4"></a><span class="co"># Plot the tree with colours at the tips and nodes to represent socail status</span></span>
<span id="cb578-5"><a href="answers-to-practical-exercises.html#cb578-5"></a><span class="kw">plot</span>(mytree, <span class="dt">cex =</span> <span class="fl">0.5</span>, <span class="dt">adj =</span> <span class="kw">c</span>(<span class="fl">0.2</span>), <span class="dt">type =</span> <span class="st">&quot;fan&quot;</span>, </span>
<span id="cb578-6"><a href="answers-to-practical-exercises.html#cb578-6"></a>     <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">show.tip.label =</span> <span class="ot">TRUE</span>)</span>
<span id="cb578-7"><a href="answers-to-practical-exercises.html#cb578-7"></a><span class="kw">tiplabels</span>(<span class="dt">pch =</span> <span class="dv">16</span>, <span class="dt">col =</span> social_colours[mydata<span class="op">$</span>SocialStatus])</span>
<span id="cb578-8"><a href="answers-to-practical-exercises.html#cb578-8"></a><span class="kw">nodelabels</span>(<span class="dt">pie =</span> ancestral_social<span class="op">$</span>lik.anc, <span class="dt">piecol =</span> social_colours, <span class="dt">cex =</span> <span class="fl">0.5</span>)</span>
<span id="cb578-9"><a href="answers-to-practical-exercises.html#cb578-9"></a><span class="kw">legend</span>(<span class="st">&quot;bottomleft&quot;</span>, <span class="dt">pch =</span> <span class="dv">15</span>, <span class="dt">bty =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb578-10"><a href="answers-to-practical-exercises.html#cb578-10"></a>       <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;Non Social&quot;</span>, <span class="st">&quot;Social&quot;</span>), </span>
<span id="cb578-11"><a href="answers-to-practical-exercises.html#cb578-11"></a>       <span class="dt">col =</span> social_colours)</span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-352-1.png" width="672" /></p>
<p><em>Root is non-social</em></p>
<ol start="4" style="list-style-type: decimal">
<li>Why didn’t I ask you to fit the symmetric model?</li>
</ol>
<p><em>Because with only two states the symmetric model is equivalent to the equal rates model</em></p>
</div>
<div id="beyond-simple-macroevolutionary-models-of-trait-evolution-1" class="section level2">
<h2><span class="header-section-number">12.8</span> Beyond simple macroevolutionary models of trait evolution</h2>
<p>In the data folder there is another tree (<code>primate-tree.nex</code>) and dataset (<code>primate-data.csv</code>) for investigating the evolution of primate life-history variables. These data come from the PanTHERIA database <span class="citation">(Jones et al. <a href="#ref-jones2009pantheria" role="doc-biblioref">2009</a>)</span> and 10kTrees <span class="citation">(Arnold, Matthews, and Nunn <a href="#ref-arnold201010ktrees" role="doc-biblioref">2010</a>)</span>.</p>
<p>Read in the tree and data, then prepare them for a PCM analysis (you may have already done this in a previous exercise which should save you some time).</p>
<p>Using log body size (<code>AdultBodyMass_g</code>) as your continuous trait, and social status (<code>SocialStatus</code>) to define the selective regimes, fit (A) a multi-rate Brownian motion model and (B) an OU model with trait optima or long-term mean values for different evolutionary regimes but the same <span class="math inline">\(\sigma^2\)</span> and <span class="math inline">\(\alpha\)</span>. Note that social status has two states: non-social (1) or social (2). Assume it is evolving according to an all rates different model.</p>
<div class="sourceCode" id="cb579"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb579-1"><a href="answers-to-practical-exercises.html#cb579-1"></a><span class="co"># Load the packages</span></span>
<span id="cb579-2"><a href="answers-to-practical-exercises.html#cb579-2"></a><span class="kw">library</span>(ape)</span>
<span id="cb579-3"><a href="answers-to-practical-exercises.html#cb579-3"></a><span class="kw">library</span>(geiger)</span>
<span id="cb579-4"><a href="answers-to-practical-exercises.html#cb579-4"></a><span class="kw">library</span>(treeplyr)</span>
<span id="cb579-5"><a href="answers-to-practical-exercises.html#cb579-5"></a><span class="kw">library</span>(caper)</span>
<span id="cb579-6"><a href="answers-to-practical-exercises.html#cb579-6"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb579-7"><a href="answers-to-practical-exercises.html#cb579-7"></a><span class="kw">library</span>(OUwie)</span>
<span id="cb579-8"><a href="answers-to-practical-exercises.html#cb579-8"></a></span>
<span id="cb579-9"><a href="answers-to-practical-exercises.html#cb579-9"></a><span class="co">### Prepare the tree and the data as shown in the Preparation answer above </span><span class="al">###</span></span>
<span id="cb579-10"><a href="answers-to-practical-exercises.html#cb579-10"></a><span class="co">### This should mean you now have mydata and mytree ready to work with </span><span class="al">###</span></span>
<span id="cb579-11"><a href="answers-to-practical-exercises.html#cb579-11"></a></span>
<span id="cb579-12"><a href="answers-to-practical-exercises.html#cb579-12"></a><span class="co"># Create Social containing just social status values</span></span>
<span id="cb579-13"><a href="answers-to-practical-exercises.html#cb579-13"></a>Social &lt;-<span class="st"> </span><span class="kw">pull</span>(mydata, SocialStatus)</span>
<span id="cb579-14"><a href="answers-to-practical-exercises.html#cb579-14"></a><span class="co"># Give social status names = species names at the tips of the phylogeny</span></span>
<span id="cb579-15"><a href="answers-to-practical-exercises.html#cb579-15"></a><span class="kw">names</span>(Social) &lt;-<span class="st"> </span>mydata<span class="op">$</span>tiplabel</span>
<span id="cb579-16"><a href="answers-to-practical-exercises.html#cb579-16"></a></span>
<span id="cb579-17"><a href="answers-to-practical-exercises.html#cb579-17"></a><span class="co"># Simulate node and tip values for 500 trees using the ARD model</span></span>
<span id="cb579-18"><a href="answers-to-practical-exercises.html#cb579-18"></a>trees.ARD &lt;-<span class="st"> </span><span class="kw">make.simmap</span>(mytree, Social, <span class="dt">model =</span> <span class="st">&quot;ARD&quot;</span>, <span class="dt">nsim =</span> <span class="dv">500</span>)</span></code></pre></div>
<pre><code>## make.simmap is sampling character histories conditioned on
## the transition matrix
## 
## Q =
##              1           2
## 1 -0.009264911 0.009264911
## 2  0.000000000 0.000000000
## (estimated using likelihood);
## and (mean) root node prior probabilities
## pi =
##   1   2 
## 0.5 0.5</code></pre>
<div class="sourceCode" id="cb581"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb581-1"><a href="answers-to-practical-exercises.html#cb581-1"></a><span class="co"># Store the results of the summary of the 500 trees</span></span>
<span id="cb581-2"><a href="answers-to-practical-exercises.html#cb581-2"></a>summary.ARD &lt;-<span class="st"> </span><span class="kw">summary</span>(trees.ARD)</span>
<span id="cb581-3"><a href="answers-to-practical-exercises.html#cb581-3"></a></span>
<span id="cb581-4"><a href="answers-to-practical-exercises.html#cb581-4"></a><span class="co"># Get the max likelihood state at each node</span></span>
<span id="cb581-5"><a href="answers-to-practical-exercises.html#cb581-5"></a>best &lt;-<span class="st"> </span><span class="kw">apply</span>(summary.ARD<span class="op">$</span>ace, <span class="dv">1</span>, which.max)</span>
<span id="cb581-6"><a href="answers-to-practical-exercises.html#cb581-6"></a></span>
<span id="cb581-7"><a href="answers-to-practical-exercises.html#cb581-7"></a><span class="co"># Assign to the tree node labels</span></span>
<span id="cb581-8"><a href="answers-to-practical-exercises.html#cb581-8"></a>mytree<span class="op">$</span>node.label &lt;-<span class="st"> </span><span class="kw">levels</span>(<span class="kw">as.factor</span>(mydata<span class="op">$</span>SocialStatus))[best]</span>
<span id="cb581-9"><a href="answers-to-practical-exercises.html#cb581-9"></a></span>
<span id="cb581-10"><a href="answers-to-practical-exercises.html#cb581-10"></a><span class="co"># Make new dataset with just three columns for OUwie</span></span>
<span id="cb581-11"><a href="answers-to-practical-exercises.html#cb581-11"></a>bodymass &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">species =</span> mydata<span class="op">$</span>tiplabel, </span>
<span id="cb581-12"><a href="answers-to-practical-exercises.html#cb581-12"></a>                    <span class="dt">regime =</span> mydata<span class="op">$</span>SocialStatus, </span>
<span id="cb581-13"><a href="answers-to-practical-exercises.html#cb581-13"></a>                    <span class="dt">trait =</span> <span class="kw">log</span>(mydata<span class="op">$</span>AdultBodyMass_g))</span>
<span id="cb581-14"><a href="answers-to-practical-exercises.html#cb581-14"></a></span>
<span id="cb581-15"><a href="answers-to-practical-exercises.html#cb581-15"></a><span class="co"># Fit BMS model</span></span>
<span id="cb581-16"><a href="answers-to-practical-exercises.html#cb581-16"></a>BMS &lt;-<span class="st"> </span><span class="kw">OUwie</span>(mytree, bodymass, <span class="dt">model =</span> <span class="st">&quot;BMS&quot;</span>, <span class="dt">root.station =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<pre><code>## Warning: An algorithm was not specified. Defaulting to computing the
## determinant and inversion of the vcv.</code></pre>
<pre><code>## Initializing... 
## Finished. Begin thorough search... 
## Finished. Summarizing results.</code></pre>
<div class="sourceCode" id="cb584"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb584-1"><a href="answers-to-practical-exercises.html#cb584-1"></a><span class="co"># Fit OUM model</span></span>
<span id="cb584-2"><a href="answers-to-practical-exercises.html#cb584-2"></a>OUM &lt;-<span class="st"> </span><span class="kw">OUwie</span>(mytree, bodymass, <span class="dt">model =</span> <span class="st">&quot;OUM&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: An algorithm was not specified. Defaulting to computing the
## determinant and inversion of the vcv.</code></pre>
<pre><code>## Initializing... 
## Finished. Begin thorough search... 
## Finished. Summarizing results.</code></pre>
<p>Then answer the following questions:</p>
<ol style="list-style-type: decimal">
<li>Which is the best fitting model?</li>
</ol>
<div class="sourceCode" id="cb587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb587-1"><a href="answers-to-practical-exercises.html#cb587-1"></a><span class="co"># Extract AICc scores from the models</span></span>
<span id="cb587-2"><a href="answers-to-practical-exercises.html#cb587-2"></a>aic.scores &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">c</span>(BMS<span class="op">$</span>AICc, OUM<span class="op">$</span>AICc), </span>
<span id="cb587-3"><a href="answers-to-practical-exercises.html#cb587-3"></a>                           <span class="kw">c</span>(<span class="st">&quot;BMS&quot;</span>, <span class="st">&quot;OUM&quot;</span>))</span>
<span id="cb587-4"><a href="answers-to-practical-exercises.html#cb587-4"></a><span class="co"># Get aic weights</span></span>
<span id="cb587-5"><a href="answers-to-practical-exercises.html#cb587-5"></a><span class="kw">aicw</span>(aic.scores)</span></code></pre></div>
<pre><code>##          fit    delta        w
## BMS 161.6226 2.704939 0.205467
## OUM 158.9176 0.000000 0.794533</code></pre>
<p><em>OUM</em></p>
<ol start="2" style="list-style-type: decimal">
<li>If you were running these analyses for a publication, what additional models would you compare these models to?</li>
</ol>
<p><em>BM and single peak OU for sure. Possibly one or more of the more complex models if you had a good biological reason to</em></p>
<ol start="3" style="list-style-type: decimal">
<li>In the multi rate Brownian model, which social status state has the higher rate of log body mass evolution?</li>
</ol>
<div class="sourceCode" id="cb589"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb589-1"><a href="answers-to-practical-exercises.html#cb589-1"></a><span class="co"># Look at the model output</span></span>
<span id="cb589-2"><a href="answers-to-practical-exercises.html#cb589-2"></a>BMS</span></code></pre></div>
<pre><code>## 
## Fit
##        lnL      AIC     AICc      BIC model ntax
##  -77.64691 161.2938 161.6226 168.3252   BMS   77
## 
## Rates
##                   1         2
## alpha            NA        NA
## sigma.sq 0.04031313 0.0257608
## 
## Optima
##                 1        2
## estimate 6.944823 6.944823
## se       0.688323 0.688323
## 
## Arrived at a reliable solution</code></pre>
<p><em>Non-social (1) has the higher rate of body mass evolution 0.04031313</em></p>
<ol start="4" style="list-style-type: decimal">
<li>Look at the parameter estimates of the multi peak OU model. Do you see anything worrying there? It might help to look at a histogram showing the body sizes of species in the non-social and the social categories.</li>
</ol>
<div class="sourceCode" id="cb591"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb591-1"><a href="answers-to-practical-exercises.html#cb591-1"></a><span class="co"># Look at the output</span></span>
<span id="cb591-2"><a href="answers-to-practical-exercises.html#cb591-2"></a>OUM</span></code></pre></div>
<pre><code>## 
## Fit
##        lnL      AIC     AICc      BIC model ntax
##  -75.18104 158.3621 158.9176 167.7373   OUM   77
## 
## 
## Rates
##                     1            2
## alpha    2.061154e-09 2.061154e-09
## sigma.sq 2.657014e-02 2.657014e-02
## 
## Optima
##                 1        2
## estimate 5.918899 18607801
## se       0.679552  7561841
## 
## 
## Half life (another way of reporting alpha)
##         1         2 
## 336290887 336290887 
## 
## Arrived at a reliable solution</code></pre>
<div class="sourceCode" id="cb593"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb593-1"><a href="answers-to-practical-exercises.html#cb593-1"></a><span class="co"># Plot the body sizes for the different groups</span></span>
<span id="cb593-2"><a href="answers-to-practical-exercises.html#cb593-2"></a><span class="kw">ggplot</span>(mydata, <span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">log</span>(AdultBodyMass_g), <span class="dt">fill =</span> <span class="kw">as.factor</span>(SocialStatus))) <span class="op">+</span></span>
<span id="cb593-3"><a href="answers-to-practical-exercises.html#cb593-3"></a><span class="st">  </span><span class="kw">geom_histogram</span>(<span class="dt">bins =</span> <span class="dv">20</span>, <span class="dt">alpha =</span> <span class="fl">0.5</span>) <span class="op">+</span></span>
<span id="cb593-4"><a href="answers-to-practical-exercises.html#cb593-4"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> social_colours) <span class="op">+</span></span>
<span id="cb593-5"><a href="answers-to-practical-exercises.html#cb593-5"></a><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>SocialStatus, <span class="dt">ncol =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb593-6"><a href="answers-to-practical-exercises.html#cb593-6"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span></span>
<span id="cb593-7"><a href="answers-to-practical-exercises.html#cb593-7"></a><span class="st">  </span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-356-1.png" width="672" /></p>
<p><em>The optima estimates are way off. Non-social species have a long-term mean log body size estimate of 5.918899, which makes sense looking at the histograms of the data. But social species have a long-term mean log body size estimate of 18607801, which if you exponentiate it is infinity. Even King Kong was not infinitely large! Something has clearly gone wrong here, but if we had not looked at the model parameters we would have said this was the best model…It’s likely we just don’t have enough data to fit a model that is as complex as this, so the parameter optimisations are not reliable.</em></p>
</div>
<div id="diversification" class="section level2">
<h2><span class="header-section-number">12.9</span> Diversification</h2>
<p><code>BAMMtools</code> has a several inbuilt example datasets. For this practical exercise we will use the cetacean (<code>whales</code>; from <span class="citation">Steeman et al. (<a href="#ref-steeman2009radiation" role="doc-biblioref">2009</a>)</span>) and the ray-finned fishes (<code>fishes</code>; from <span class="citation">Rabosky et al. (<a href="#ref-rabosky2013rates" role="doc-biblioref">2013</a>)</span>) datasets.</p>
<p>Load the whales and fishes trees and BAMM outputs data into R (<strong>there is no <code>mcmc.fishes</code> file</strong>_).</p>
<div class="sourceCode" id="cb594"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb594-1"><a href="answers-to-practical-exercises.html#cb594-1"></a><span class="co"># Load the packages</span></span>
<span id="cb594-2"><a href="answers-to-practical-exercises.html#cb594-2"></a><span class="kw">library</span>(ape)</span>
<span id="cb594-3"><a href="answers-to-practical-exercises.html#cb594-3"></a><span class="kw">library</span>(BAMMtools)</span>
<span id="cb594-4"><a href="answers-to-practical-exercises.html#cb594-4"></a><span class="kw">library</span>(coda)</span>
<span id="cb594-5"><a href="answers-to-practical-exercises.html#cb594-5"></a><span class="kw">library</span>(phytools)</span>
<span id="cb594-6"><a href="answers-to-practical-exercises.html#cb594-6"></a><span class="kw">library</span>(viridis)</span>
<span id="cb594-7"><a href="answers-to-practical-exercises.html#cb594-7"></a></span>
<span id="cb594-8"><a href="answers-to-practical-exercises.html#cb594-8"></a><span class="co"># Load whales data from BAMMtools</span></span>
<span id="cb594-9"><a href="answers-to-practical-exercises.html#cb594-9"></a><span class="kw">data</span>(whales)</span>
<span id="cb594-10"><a href="answers-to-practical-exercises.html#cb594-10"></a><span class="kw">data</span>(mcmc.whales)</span>
<span id="cb594-11"><a href="answers-to-practical-exercises.html#cb594-11"></a><span class="kw">data</span>(events.whales)</span>
<span id="cb594-12"><a href="answers-to-practical-exercises.html#cb594-12"></a></span>
<span id="cb594-13"><a href="answers-to-practical-exercises.html#cb594-13"></a><span class="co"># Load fishes data from BAMMtools</span></span>
<span id="cb594-14"><a href="answers-to-practical-exercises.html#cb594-14"></a><span class="kw">data</span>(fishes)</span>
<span id="cb594-15"><a href="answers-to-practical-exercises.html#cb594-15"></a><span class="kw">data</span>(mcmc.fishes)</span></code></pre></div>
<pre><code>## Warning in data(mcmc.fishes): data set &#39;mcmc.fishes&#39; not found</code></pre>
<div class="sourceCode" id="cb596"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb596-1"><a href="answers-to-practical-exercises.html#cb596-1"></a><span class="kw">data</span>(events.fishes)</span>
<span id="cb596-2"><a href="answers-to-practical-exercises.html#cb596-2"></a></span>
<span id="cb596-3"><a href="answers-to-practical-exercises.html#cb596-3"></a><span class="co"># Read in the tree, mcmc, and edata for whales</span></span>
<span id="cb596-4"><a href="answers-to-practical-exercises.html#cb596-4"></a>whales_tree &lt;-<span class="st"> </span>whales</span>
<span id="cb596-5"><a href="answers-to-practical-exercises.html#cb596-5"></a>whales_mcmc &lt;-<span class="st"> </span>mcmc.whales</span>
<span id="cb596-6"><a href="answers-to-practical-exercises.html#cb596-6"></a>whales_edata &lt;-<span class="st"> </span><span class="kw">getEventData</span>(whales, <span class="dt">eventdata =</span> events.whales, <span class="dt">burnin =</span> <span class="fl">0.1</span>)</span></code></pre></div>
<pre><code>## Processing event data from data.frame
## 
## Discarded as burnin: GENERATIONS &lt;  995000
## Analyzing  1801  samples from posterior
## 
## Setting recursive sequence on tree...
## 
## Done with recursive sequence</code></pre>
<div class="sourceCode" id="cb598"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb598-1"><a href="answers-to-practical-exercises.html#cb598-1"></a><span class="co"># Read in the tree, mcmc, and edata for fishes</span></span>
<span id="cb598-2"><a href="answers-to-practical-exercises.html#cb598-2"></a>fishes_tree &lt;-<span class="st"> </span>fishes</span>
<span id="cb598-3"><a href="answers-to-practical-exercises.html#cb598-3"></a>fishes_edata &lt;-<span class="st"> </span><span class="kw">getEventData</span>(fishes, <span class="dt">eventdata =</span> events.fishes, <span class="dt">burnin =</span> <span class="fl">0.1</span>)</span></code></pre></div>
<pre><code>## Processing event data from data.frame
## 
## Discarded as burnin: GENERATIONS &lt;  5
## Analyzing  46  samples from posterior
## 
## Setting recursive sequence on tree...
## 
## Done with recursive sequence</code></pre>
<p>Then answer the following questions:</p>
<ol style="list-style-type: decimal">
<li>How can we tell that the BAMM model for whales has converged? What is the effective sample size for number of shifts?</li>
</ol>
<div class="sourceCode" id="cb600"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb600-1"><a href="answers-to-practical-exercises.html#cb600-1"></a><span class="co"># Take a look at the plot</span></span>
<span id="cb600-2"><a href="answers-to-practical-exercises.html#cb600-2"></a><span class="kw">plot</span>(mcmc<span class="op">$</span>logLik <span class="op">~</span><span class="st"> </span>mcmc<span class="op">$</span>generation)</span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-358-1.png" width="672" /></p>
<div class="sourceCode" id="cb601"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb601-1"><a href="answers-to-practical-exercises.html#cb601-1"></a><span class="co"># What is 10% of rows?</span></span>
<span id="cb601-2"><a href="answers-to-practical-exercises.html#cb601-2"></a>burnstart &lt;-<span class="st"> </span><span class="kw">floor</span>(<span class="fl">0.1</span> <span class="op">*</span><span class="st"> </span><span class="kw">nrow</span>(mcmc))</span>
<span id="cb601-3"><a href="answers-to-practical-exercises.html#cb601-3"></a><span class="co"># Remove the first 10% from the runs </span></span>
<span id="cb601-4"><a href="answers-to-practical-exercises.html#cb601-4"></a>postburn &lt;-<span class="st"> </span>mcmc[burnstart<span class="op">:</span><span class="kw">nrow</span>(mcmc), ]</span>
<span id="cb601-5"><a href="answers-to-practical-exercises.html#cb601-5"></a></span>
<span id="cb601-6"><a href="answers-to-practical-exercises.html#cb601-6"></a><span class="co"># Extract ESS for the number of rate shifts parameter</span></span>
<span id="cb601-7"><a href="answers-to-practical-exercises.html#cb601-7"></a><span class="kw">effectiveSize</span>(postburn<span class="op">$</span>N_shifts)</span></code></pre></div>
<pre><code>##     var1 
## 273.7424</code></pre>
<div class="sourceCode" id="cb603"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb603-1"><a href="answers-to-practical-exercises.html#cb603-1"></a><span class="co"># Extract ESS for the log likelihood</span></span>
<span id="cb603-2"><a href="answers-to-practical-exercises.html#cb603-2"></a><span class="kw">effectiveSize</span>(postburn<span class="op">$</span>logLik)</span></code></pre></div>
<pre><code>##    var1 
## 219.238</code></pre>
<p><em>The plot looks like a furry caterpillar (!), and effective sample sizes are well over 200. ESS = 1225.609</em></p>
<ol start="2" style="list-style-type: decimal">
<li>What is the most common number of rate shifts in the posterior for the whales data?</li>
</ol>
<div class="sourceCode" id="cb605"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb605-1"><a href="answers-to-practical-exercises.html#cb605-1"></a><span class="co"># Look at the posterior for number of shifts</span></span>
<span id="cb605-2"><a href="answers-to-practical-exercises.html#cb605-2"></a><span class="kw">summary</span>(whales_edata)</span></code></pre></div>
<pre><code>## 
## Analyzed 1801 posterior samples
## Shift posterior distribution:
## 
##          0     0.0770
##          1     0.6600
##          2     0.2100
##          3     0.0430
##          4     0.0110
##          5     0.0022
## 
## Compute credible set of shift configurations for more information:
##  See ?credibleShiftSet and ?getBestShiftConfiguration</code></pre>
<p><em>1</em></p>
<ol start="3" style="list-style-type: decimal">
<li>Plot the 95% credible set of rate shift configurations for whales.</li>
</ol>
<div class="sourceCode" id="cb607"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb607-1"><a href="answers-to-practical-exercises.html#cb607-1"></a><span class="co"># Extract 95% credible set of rate shift configurations</span></span>
<span id="cb607-2"><a href="answers-to-practical-exercises.html#cb607-2"></a>whales_css &lt;-<span class="st"> </span><span class="kw">credibleShiftSet</span>(whales_edata, <span class="dt">expectedNumberOfShifts =</span> <span class="dv">1</span>, </span>
<span id="cb607-3"><a href="answers-to-practical-exercises.html#cb607-3"></a>                        <span class="dt">threshold =</span> <span class="dv">5</span>, <span class="dt">set.limit =</span> <span class="fl">0.95</span>)</span>
<span id="cb607-4"><a href="answers-to-practical-exercises.html#cb607-4"></a><span class="co"># Look at the summary</span></span>
<span id="cb607-5"><a href="answers-to-practical-exercises.html#cb607-5"></a><span class="kw">summary</span>(whales_css)</span></code></pre></div>
<pre><code>## 
##  95 % credible set of rate shift configurations sampled with BAMM
## 
## Distinct shift configurations in credible set:  4
## 
## Frequency of 4 shift configurations with highest posterior probability:
## 
## 
##    rank     probability cumulative  Core_shifts
##          1 0.46418656  0.4641866          1
##          2 0.28928373  0.7534703          1
##          3 0.14158801  0.8950583          1
##          4 0.09272626  0.9877846          0</code></pre>
<div class="sourceCode" id="cb609"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb609-1"><a href="answers-to-practical-exercises.html#cb609-1"></a><span class="co"># Plot 95 % credible set of rate shift configurations</span></span>
<span id="cb609-2"><a href="answers-to-practical-exercises.html#cb609-2"></a><span class="kw">plot.credibleshiftset</span>(whales_css)</span></code></pre></div>
<pre><code>## Omitted 0 plots</code></pre>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-360-1.png" width="672" /></p>
<ol start="4" style="list-style-type: decimal">
<li>Plot the mean phylorate plot for fishes.</li>
</ol>
<div class="sourceCode" id="cb611"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb611-1"><a href="answers-to-practical-exercises.html#cb611-1"></a><span class="co"># Plot mean phylorate plot with legend and slighting larger line width (lwd)</span></span>
<span id="cb611-2"><a href="answers-to-practical-exercises.html#cb611-2"></a><span class="kw">plot.bammdata</span>(fishes_edata, <span class="dt">legend =</span> <span class="ot">TRUE</span>, <span class="dt">lwd =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="pcm-primer_files/figure-html/unnamed-chunk-361-1.png" width="672" /></p>
<ol start="5" style="list-style-type: decimal">
<li>What are the mean speciation rate and the 90% posterior density for mean speciation rate in fishes?</li>
</ol>
<div class="sourceCode" id="cb612"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb612-1"><a href="answers-to-practical-exercises.html#cb612-1"></a><span class="co"># Extract speciation and extinction rates from the whole posterior</span></span>
<span id="cb612-2"><a href="answers-to-practical-exercises.html#cb612-2"></a>allrates &lt;-<span class="st"> </span><span class="kw">getCladeRates</span>(fishes_edata)</span>
<span id="cb612-3"><a href="answers-to-practical-exercises.html#cb612-3"></a><span class="co"># Calculate overall mean speciation rate</span></span>
<span id="cb612-4"><a href="answers-to-practical-exercises.html#cb612-4"></a><span class="kw">mean</span>(allrates<span class="op">$</span>lambda)</span></code></pre></div>
<pre><code>## [1] 0.09470252</code></pre>
<div class="sourceCode" id="cb614"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb614-1"><a href="answers-to-practical-exercises.html#cb614-1"></a><span class="co"># Calculate 90% highest posterior density for speciation rate</span></span>
<span id="cb614-2"><a href="answers-to-practical-exercises.html#cb614-2"></a><span class="kw">quantile</span>(allrates<span class="op">$</span>lambda, <span class="kw">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>))</span></code></pre></div>
<pre><code>##         5%        95% 
## 0.09020489 0.09960825</code></pre>
<p><em>Mean = 0.09470252, 5% = 0.09020489, 95% = 0.09960825</em></p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-arnold201010ktrees">
<p>Arnold, Christian, Luke J Matthews, and Charles L Nunn. 2010. “The 10kTrees Website: A New Online Resource for Primate Phylogeny.” <em>Evolutionary Anthropology: Issues, News, and Reviews</em> 19 (3): 114–18.</p>
</div>
<div id="ref-feng2017phylogenomics">
<p>Feng, Yan-Jie, David C Blackburn, Dan Liang, David M Hillis, David B Wake, David C Cannatella, and Peng Zhang. 2017. “Phylogenomics Reveals Rapid, Simultaneous Diversification of Three Major Clades of Gondwanan Frogs at the Cretaceous–Paleogene Boundary.” <em>Proceedings of the National Academy of Sciences</em> 114 (29): E5864–E5870.</p>
</div>
<div id="ref-jones2009pantheria">
<p>Jones, Kate E, Jon Bielby, Marcel Cardillo, Susanne A Fritz, Justin O’Dell, C David L Orme, Kamran Safi, et al. 2009. “PanTHERIA: A Species-Level Database of Life History, Ecology, and Geography of Extant and Recently Extinct Mammals: Ecological Archives E090-184.” <em>Ecology</em> 90 (9): 2648–8.</p>
</div>
<div id="ref-rabosky2013rates">
<p>Rabosky, Daniel L, Francesco Santini, Jonathan Eastman, Stephen A Smith, Brian Sidlauskas, Jonathan Chang, and Michael E Alfaro. 2013. “Rates of Speciation and Morphological Evolution Are Correlated Across the Largest Vertebrate Radiation.” <em>Nature Communications</em> 4 (1): 1–8.</p>
</div>
<div id="ref-steeman2009radiation">
<p>Steeman, Mette E, Martin B Hebsgaard, R Ewan Fordyce, Simon YW Ho, Daniel L Rabosky, Rasmus Nielsen, Carsten Rahbek, Henrik Glenner, Martin V Sørensen, and Eske Willerslev. 2009. “Radiation of Extant Cetaceans Driven by Restructuring of the Oceans.” <em>Systematic Biology</em> 58 (6): 573–85.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="critical-thinking-about-methods-and-analyses.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/12-Answers.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["pcm-primer.pdf", "pcm-primer.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
