<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Phylogenetic Signal in R | Practical exercises for Methods in Macroecology and Macroevolution module</title>
  <meta name="description" content="Chapter 5 Phylogenetic Signal in R | Practical exercises for Methods in Macroecology and Macroevolution module" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Phylogenetic Signal in R | Practical exercises for Methods in Macroecology and Macroevolution module" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Phylogenetic Signal in R | Practical exercises for Methods in Macroecology and Macroevolution module" />
  
  
  

<meta name="author" content="Natalie Cooper" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="preparing-your-tree-and-data-for-pcms-in-r.html"/>
<link rel="next" href="phylogenetic-generalised-least-squares-pgls-in-r.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">PCM Primer</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#datasets-and-scripts"><i class="fa fa-check"></i>Datasets and scripts</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#how-to-use-these-materials"><i class="fa fa-check"></i>How to use these materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#boxes"><i class="fa fa-check"></i>Boxes</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#example-datasets"><i class="fa fa-check"></i>Example datasets</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#frog-eye-size-evolution"><i class="fa fa-check"></i>Frog eye size evolution</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#natricine-snake-head-shape-evolution"><i class="fa fa-check"></i>Natricine snake head shape evolution</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#diversification-in-dragonflies"><i class="fa fa-check"></i>Diversification in dragonflies</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citing-r-and-r-packages"><i class="fa fa-check"></i>Citing R and R packages</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><i class="fa fa-check"></i><b>1</b> What you need to be able to do in R before you start</a><ul>
<li class="chapter" data-level="1.1" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#installing-r-and-rstudio"><i class="fa fa-check"></i><b>1.1</b> Installing R (and RStudio)</a></li>
<li class="chapter" data-level="1.2" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#downloading-the-data"><i class="fa fa-check"></i><b>1.2</b> Downloading the data</a></li>
<li class="chapter" data-level="1.3" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#r-projects"><i class="fa fa-check"></i><b>1.3</b> R Projects</a></li>
<li class="chapter" data-level="1.4" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#using-a-script"><i class="fa fa-check"></i><b>1.4</b> Using a script</a></li>
<li class="chapter" data-level="1.5" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#installing-and-loading-extra-packages-in-r"><i class="fa fa-check"></i><b>1.5</b> Installing and loading extra packages in R</a><ul>
<li class="chapter" data-level="1.5.1" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#loading-packages-into-r"><i class="fa fa-check"></i><b>1.5.1</b> Loading packages into R</a></li>
<li class="chapter" data-level="1.5.2" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#ggtree-a-special-case-for-package-installation"><i class="fa fa-check"></i><b>1.5.2</b> <code>ggtree</code>, a special case for package installation</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#loading-and-viewing-your-data-in-r"><i class="fa fa-check"></i><b>1.6</b> Loading and viewing your data in R</a></li>
<li class="chapter" data-level="1.7" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#errors-and-warnings"><i class="fa fa-check"></i><b>1.7</b> Errors and warnings</a><ul>
<li class="chapter" data-level="1.7.1" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#removing-the"><i class="fa fa-check"></i><b>1.7.1</b> Removing the <code>+</code></a></li>
</ul></li>
<li class="chapter" data-level="1.8" data-path="what-you-need-to-be-able-to-do-in-r-before-you-start.html"><a href="what-you-need-to-be-able-to-do-in-r-before-you-start.html#summary"><i class="fa fa-check"></i><b>1.8</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html"><i class="fa fa-check"></i><b>2</b> Diversity Indices in R</a><ul>
<li class="chapter" data-level="2.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#before-you-start"><i class="fa fa-check"></i><b>2.1</b> Before you start</a></li>
<li class="chapter" data-level="2.2" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#preparation"><i class="fa fa-check"></i><b>2.2</b> Preparation</a></li>
<li class="chapter" data-level="2.3" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#what-are-diversity-indices"><i class="fa fa-check"></i><b>2.3</b> What are diversity indices?</a><ul>
<li class="chapter" data-level="2.3.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#alpha-beta-and-gamma-diversity"><i class="fa fa-check"></i><b>2.3.1</b> <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span> and <span class="math inline">\(\gamma\)</span> diversity</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#practical-example-using-pokemon"><i class="fa fa-check"></i><b>2.4</b> Practical example using Pokemon…</a></li>
<li class="chapter" data-level="2.5" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#species-diversity-indices"><i class="fa fa-check"></i><b>2.5</b> Species diversity indices</a><ul>
<li class="chapter" data-level="2.5.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#alpha-diversity"><i class="fa fa-check"></i><b>2.5.1</b> <span class="math inline">\(\alpha\)</span> diversity</a></li>
<li class="chapter" data-level="2.5.2" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#beta-diversity"><i class="fa fa-check"></i><b>2.5.2</b> <span class="math inline">\(\beta\)</span> diversity</a></li>
<li class="chapter" data-level="2.5.3" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#gamma-diversity"><i class="fa fa-check"></i><b>2.5.3</b> <span class="math inline">\(\gamma\)</span> diversity</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#species-accumulation-curves-colwell-coddington-1994"><i class="fa fa-check"></i><b>2.6</b> Species accumulation curves (Colwell &amp; Coddington 1994)</a><ul>
<li class="chapter" data-level="2.6.1" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#pokemon-species-accumulation-curves"><i class="fa fa-check"></i><b>2.6.1</b> Pokemon species accumulation curves</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="diversity-indices-in-r.html"><a href="diversity-indices-in-r.html#practical-exercise"><i class="fa fa-check"></i><b>2.7</b> Practical exercise</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html"><i class="fa fa-check"></i><b>3</b> Understanding, plotting and manipulating phylogenies in R</a><ul>
<li class="chapter" data-level="3.1" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#before-you-start-1"><i class="fa fa-check"></i><b>3.1</b> Before you start</a></li>
<li class="chapter" data-level="3.2" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#what-do-phylogeny-files-look-like"><i class="fa fa-check"></i><b>3.2</b> What do phylogeny files look like?</a><ul>
<li class="chapter" data-level="3.2.1" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#newick-trees"><i class="fa fa-check"></i><b>3.2.1</b> Newick trees</a></li>
<li class="chapter" data-level="3.2.2" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#nexus-trees"><i class="fa fa-check"></i><b>3.2.2</b> NEXUS trees</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#how-are-phylogenies-formatted-in-r"><i class="fa fa-check"></i><b>3.3</b> How are phylogenies formatted in R?</a></li>
<li class="chapter" data-level="3.4" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#plotting-phylogenies"><i class="fa fa-check"></i><b>3.4</b> Plotting phylogenies</a><ul>
<li class="chapter" data-level="3.4.1" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#basic-phylogeny-plotting-with-ape"><i class="fa fa-check"></i><b>3.4.1</b> Basic phylogeny plotting with APE</a></li>
<li class="chapter" data-level="3.4.2" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#basic-phylogeny-plotting-with-ggtree"><i class="fa fa-check"></i><b>3.4.2</b> Basic phylogeny plotting with <code>ggtree</code></a></li>
<li class="chapter" data-level="3.4.3" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#back-to-ggtree"><i class="fa fa-check"></i><b>3.4.3</b> Back to <code>ggtree</code></a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#phylogeny-manipulation"><i class="fa fa-check"></i><b>3.5</b> Phylogeny manipulation</a><ul>
<li class="chapter" data-level="3.5.1" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#removing-polytomies"><i class="fa fa-check"></i><b>3.5.1</b> Removing polytomies</a></li>
<li class="chapter" data-level="3.5.2" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#rooting-your-phylogeny"><i class="fa fa-check"></i><b>3.5.2</b> Rooting your phylogeny</a></li>
<li class="chapter" data-level="3.5.3" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#manipulating-the-species-in-your-phlyogeny"><i class="fa fa-check"></i><b>3.5.3</b> Manipulating the species in your phlyogeny</a></li>
<li class="chapter" data-level="3.5.4" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#writing-phylogenies-to-file"><i class="fa fa-check"></i><b>3.5.4</b> Writing phylogenies to file</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#summary-1"><i class="fa fa-check"></i><b>3.6</b> Summary</a></li>
<li class="chapter" data-level="3.7" data-path="understanding-plotting-and-manipulating-phylogenies-in-r.html"><a href="understanding-plotting-and-manipulating-phylogenies-in-r.html#practical-exercise-1"><i class="fa fa-check"></i><b>3.7</b> Practical exercise</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html"><i class="fa fa-check"></i><b>4</b> Preparing your tree and data for PCMs in R</a><ul>
<li class="chapter" data-level="4.1" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#before-you-start-2"><i class="fa fa-check"></i><b>4.1</b> Before you start</a></li>
<li class="chapter" data-level="4.2" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#preparation-1"><i class="fa fa-check"></i><b>4.2</b> Preparation</a></li>
<li class="chapter" data-level="4.3" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#reading-and-checking-your-phylogeny-in-r"><i class="fa fa-check"></i><b>4.3</b> Reading and checking your phylogeny in R</a></li>
<li class="chapter" data-level="4.4" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#reading-the-data-into-r"><i class="fa fa-check"></i><b>4.4</b> Reading the data into R</a></li>
<li class="chapter" data-level="4.5" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#matching-your-data-to-your-phylogeny"><i class="fa fa-check"></i><b>4.5</b> Matching your data to your phylogeny</a><ul>
<li class="chapter" data-level="4.5.1" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#species-names-with-spaces"><i class="fa fa-check"></i><b>4.5.1</b> Species names with spaces</a></li>
<li class="chapter" data-level="4.5.2" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#mismatches-between-species-in-your-data-and-phylogeny"><i class="fa fa-check"></i><b>4.5.2</b> Mismatches between species in your data and phylogeny</a></li>
<li class="chapter" data-level="4.5.3" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#matching-the-tree-and-the-data"><i class="fa fa-check"></i><b>4.5.3</b> Matching the tree and the data</a></li>
<li class="chapter" data-level="4.5.4" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#subsetting-your-tree-and-data"><i class="fa fa-check"></i><b>4.5.4</b> Subsetting your tree and data</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#quick-template-code"><i class="fa fa-check"></i><b>4.6</b> Quick template code</a></li>
<li class="chapter" data-level="4.7" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#summary-2"><i class="fa fa-check"></i><b>4.7</b> Summary</a></li>
<li class="chapter" data-level="4.8" data-path="preparing-your-tree-and-data-for-pcms-in-r.html"><a href="preparing-your-tree-and-data-for-pcms-in-r.html#practical-exercise-2"><i class="fa fa-check"></i><b>4.8</b> Practical exercise</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Signal in R</a><ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#before-you-start-3"><i class="fa fa-check"></i><b>5.1</b> Before you start</a></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#preparation-2"><i class="fa fa-check"></i><b>5.2</b> Preparation</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#estimating-phylogenetic-signal-for-continuous-variables"><i class="fa fa-check"></i><b>5.3</b> Estimating phylogenetic signal for continuous variables</a><ul>
<li class="chapter" data-level="5.3.1" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#pagels-lambda"><i class="fa fa-check"></i><b>5.3.1</b> Pagel’s <span class="math inline">\(\lambda\)</span></a></li>
<li class="chapter" data-level="5.3.2" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#blombergs-k-blomberg-et-al-2003"><i class="fa fa-check"></i><b>5.3.2</b> Blomberg’s <em>K</em> (Blomberg et al 2003)</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#estimating-phylogenetic-signal-for-non-continuous-variables"><i class="fa fa-check"></i><b>5.4</b> Estimating phylogenetic signal for non-continuous variables</a><ul>
<li class="chapter" data-level="5.4.1" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#d"><i class="fa fa-check"></i><b>5.4.1</b> <em>D</em></a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#summary-3"><i class="fa fa-check"></i><b>5.5</b> Summary</a></li>
<li class="chapter" data-level="5.6" data-path="phylogenetic-signal-in-r.html"><a href="phylogenetic-signal-in-r.html#practical-exercises"><i class="fa fa-check"></i><b>5.6</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html"><i class="fa fa-check"></i><b>6</b> Phylogenetic Generalised Least Squares (PGLS) in R</a><ul>
<li class="chapter" data-level="6.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#before-you-start-4"><i class="fa fa-check"></i><b>6.1</b> Before you start</a></li>
<li class="chapter" data-level="6.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#preparation-3"><i class="fa fa-check"></i><b>6.2</b> Preparation</a></li>
<li class="chapter" data-level="6.3" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#phylogenetic-generalized-least-squares-models-pgls"><i class="fa fa-check"></i><b>6.3</b> Phylogenetic generalized least squares models (PGLS)</a><ul>
<li class="chapter" data-level="6.3.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#fitting-pgls-models-in-r"><i class="fa fa-check"></i><b>6.3.1</b> Fitting PGLS models in R</a></li>
<li class="chapter" data-level="6.3.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#model-diagnostics-for-pgls-models"><i class="fa fa-check"></i><b>6.3.2</b> Model diagnostics for PGLS models</a></li>
<li class="chapter" data-level="6.3.3" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#likelihood-profiles-for-lambda-in-pgls-models"><i class="fa fa-check"></i><b>6.3.3</b> Likelihood profiles for <span class="math inline">\(\lambda\)</span> in PGLS models</a></li>
<li class="chapter" data-level="6.3.4" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#a-pgls-example-with-more-than-one-predictor-variable"><i class="fa fa-check"></i><b>6.3.4</b> A PGLS example with more than one predictor variable</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#common-issues-when-running-pgls-models-in-r"><i class="fa fa-check"></i><b>6.4</b> Common issues when running PGLS models in R</a><ul>
<li class="chapter" data-level="6.4.1" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#optim-errors"><i class="fa fa-check"></i><b>6.4.1</b> Optim errors</a></li>
<li class="chapter" data-level="6.4.2" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#big-problems-with-small-datasets"><i class="fa fa-check"></i><b>6.4.2</b> Big problems with small datasets</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#summary-4"><i class="fa fa-check"></i><b>6.5</b> Summary</a></li>
<li class="chapter" data-level="6.6" data-path="phylogenetic-generalised-least-squares-pgls-in-r.html"><a href="phylogenetic-generalised-least-squares-pgls-in-r.html#practical-exercises-1"><i class="fa fa-check"></i><b>6.6</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html"><i class="fa fa-check"></i><b>7</b> Macroevolutionary models for continuous traits</a><ul>
<li class="chapter" data-level="7.1" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#before-you-start-5"><i class="fa fa-check"></i><b>7.1</b> Before you start</a></li>
<li class="chapter" data-level="7.2" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#preparation-4"><i class="fa fa-check"></i><b>7.2</b> Preparation</a></li>
<li class="chapter" data-level="7.3" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#models-of-evolution-for-continuous-traits"><i class="fa fa-check"></i><b>7.3</b> Models of evolution for continuous traits</a><ul>
<li class="chapter" data-level="7.3.1" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#loook-at-the-data"><i class="fa fa-check"></i><b>7.3.1</b> loook at the data????</a></li>
<li class="chapter" data-level="7.3.2" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#fitting-the-brownian-motion-and-ou-models-of-evolution-using-fitcontinuous"><i class="fa fa-check"></i><b>7.3.2</b> Fitting the Brownian motion and OU models of evolution using <code>fitContinuous</code></a></li>
<li class="chapter" data-level="7.3.3" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#comparing-models-of-evolution-using-aic"><i class="fa fa-check"></i><b>7.3.3</b> Comparing models of evolution using AIC</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#summary-5"><i class="fa fa-check"></i><b>7.4</b> Summary</a></li>
<li class="chapter" data-level="7.5" data-path="macroevolutionary-models-for-continuous-traits.html"><a href="macroevolutionary-models-for-continuous-traits.html#practical-exercises-2"><i class="fa fa-check"></i><b>7.5</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html"><i class="fa fa-check"></i><b>8</b> Macroevolutionary models for discrete traits</a><ul>
<li class="chapter" data-level="8.1" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#before-you-start-6"><i class="fa fa-check"></i><b>8.1</b> Before you start</a></li>
<li class="chapter" data-level="8.2" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#preparation-5"><i class="fa fa-check"></i><b>8.2</b> Preparation</a></li>
<li class="chapter" data-level="8.3" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#models-of-evolution-for-discrete-traits"><i class="fa fa-check"></i><b>8.3</b> Models of evolution for discrete traits</a><ul>
<li class="chapter" data-level="8.3.1" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#fitting-the-er-sym-and-ard-models-of-evolution-using-fitdiscrete"><i class="fa fa-check"></i><b>8.3.1</b> Fitting the ER, SYM and ARD models of evolution using <code>fitDiscrete</code></a></li>
<li class="chapter" data-level="8.3.2" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#comparing-models-of-evolution-using-aic-1"><i class="fa fa-check"></i><b>8.3.2</b> Comparing models of evolution using AIC</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#ancestral-state-estimations"><i class="fa fa-check"></i><b>8.4</b> Ancestral state estimations</a></li>
<li class="chapter" data-level="8.5" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#summary-6"><i class="fa fa-check"></i><b>8.5</b> Summary</a></li>
<li class="chapter" data-level="8.6" data-path="macroevolutionary-models-for-discrete-traits.html"><a href="macroevolutionary-models-for-discrete-traits.html#practical-exercises-3"><i class="fa fa-check"></i><b>8.6</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><i class="fa fa-check"></i><b>9</b> Beyond simple macroevolutionary models of trait evolution</a><ul>
<li class="chapter" data-level="9.1" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#before-you-start-7"><i class="fa fa-check"></i><b>9.1</b> Before you start</a></li>
<li class="chapter" data-level="9.2" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#preparation-6"><i class="fa fa-check"></i><b>9.2</b> Preparation</a></li>
<li class="chapter" data-level="9.3" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#more-complex-models-of-trait-evolution"><i class="fa fa-check"></i><b>9.3</b> More complex models of trait evolution</a><ul>
<li class="chapter" data-level="9.3.1" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#preparing-the-tree-and-data-for-ouwie"><i class="fa fa-check"></i><b>9.3.1</b> Preparing the tree and data for OUwie</a></li>
<li class="chapter" data-level="9.3.2" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#running-models-using-ouwie"><i class="fa fa-check"></i><b>9.3.2</b> Running models using OUwie</a></li>
<li class="chapter" data-level="9.3.3" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#single-rate-bm-models"><i class="fa fa-check"></i><b>9.3.3</b> Single rate BM models</a></li>
<li class="chapter" data-level="9.3.4" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#single-optimum-ou-models"><i class="fa fa-check"></i><b>9.3.4</b> Single optimum OU models</a></li>
<li class="chapter" data-level="9.3.5" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#multi-rate-bm-bms-models"><i class="fa fa-check"></i><b>9.3.5</b> Multi-rate BM (BMS) models</a></li>
<li class="chapter" data-level="9.3.6" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#multi-optima-ou-oum-models"><i class="fa fa-check"></i><b>9.3.6</b> Multi-optima OU (OUM) models</a></li>
<li class="chapter" data-level="9.3.7" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#multi-rate-multi-optima-ou-motion-models-oumv"><i class="fa fa-check"></i><b>9.3.7</b> Multi-rate multi-optima OU motion models (OUMV)</a></li>
<li class="chapter" data-level="9.3.8" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#multi-alpha-multi-optima-ou-motion-models-ouma"><i class="fa fa-check"></i><b>9.3.8</b> Multi-alpha multi-optima OU motion models (OUMA)</a></li>
<li class="chapter" data-level="9.3.9" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#multi-rate-multi-optima-ou-motion-models-oumva"><i class="fa fa-check"></i><b>9.3.9</b> Multi-rate multi-optima OU motion models (OUMVA)</a></li>
<li class="chapter" data-level="9.3.10" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#comparing-models-of-evolution-using-aic-2"><i class="fa fa-check"></i><b>9.3.10</b> Comparing models of evolution using AIC</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#summary-7"><i class="fa fa-check"></i><b>9.4</b> Summary</a></li>
<li class="chapter" data-level="9.5" data-path="beyond-simple-macroevolutionary-models-of-trait-evolution.html"><a href="beyond-simple-macroevolutionary-models-of-trait-evolution.html#practical-exercises-4"><i class="fa fa-check"></i><b>9.5</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html"><i class="fa fa-check"></i><b>10</b> Diversification rate analyses</a><ul>
<li class="chapter" data-level="10.1" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#before-you-start-8"><i class="fa fa-check"></i><b>10.1</b> Before you start</a></li>
<li class="chapter" data-level="10.2" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#preparation-7"><i class="fa fa-check"></i><b>10.2</b> Preparation</a></li>
<li class="chapter" data-level="10.3" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#lineage-through-time-plots-and-the-gamma-statistic"><i class="fa fa-check"></i><b>10.3</b> Lineage through time plots and the <span class="math inline">\(\gamma\)</span> statistic</a></li>
<li class="chapter" data-level="10.4" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#bamm-bayesian-analysis-of-macroevolutionary-mixtures"><i class="fa fa-check"></i><b>10.4</b> BAMM: Bayesian Analysis of Macroevolutionary Mixtures</a></li>
<li class="chapter" data-level="10.5" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#running-bamm"><i class="fa fa-check"></i><b>10.5</b> Running BAMM</a></li>
<li class="chapter" data-level="10.6" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#looking-at-the-output-of-bamm-using-bammtools-in-r"><i class="fa fa-check"></i><b>10.6</b> Looking at the output of BAMM using BAMMtools in R</a><ul>
<li class="chapter" data-level="10.6.1" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#assessing-mcmc-convergence"><i class="fa fa-check"></i><b>10.6.1</b> Assessing MCMC convergence</a></li>
<li class="chapter" data-level="10.6.2" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#analysis-of-rate-shifts-in-the-dragonfly-phylogeny"><i class="fa fa-check"></i><b>10.6.2</b> Analysis of rate shifts in the dragonfly phylogeny</a></li>
<li class="chapter" data-level="10.6.3" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#mean-phylorate-plot"><i class="fa fa-check"></i><b>10.6.3</b> Mean phylorate plot</a></li>
<li class="chapter" data-level="10.6.4" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#best-single-distinct-shift-configuration"><i class="fa fa-check"></i><b>10.6.4</b> Best single distinct shift configuration</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#credible-sets-of-shifts"><i class="fa fa-check"></i><b>10.7</b> Credible sets of shifts</a></li>
<li class="chapter" data-level="10.8" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#clade-specific-evolutionary-rates"><i class="fa fa-check"></i><b>10.8</b> Clade specific evolutionary rates</a></li>
<li class="chapter" data-level="10.9" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#rate-through-time-analysis"><i class="fa fa-check"></i><b>10.9</b> Rate-through-time analysis</a></li>
<li class="chapter" data-level="10.10" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#macroevolutionary-cohort-analysis"><i class="fa fa-check"></i><b>10.10</b> Macroevolutionary cohort analysis</a></li>
<li class="chapter" data-level="10.11" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#assumptions-and-issues-with-bamm"><i class="fa fa-check"></i><b>10.11</b> Assumptions and issues with BAMM</a></li>
<li class="chapter" data-level="10.12" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#summary-8"><i class="fa fa-check"></i><b>10.12</b> Summary</a></li>
<li class="chapter" data-level="10.13" data-path="diversification-rate-analyses.html"><a href="diversification-rate-analyses.html#practical-exercises-5"><i class="fa fa-check"></i><b>10.13</b> Practical exercises</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html"><i class="fa fa-check"></i><b>11</b> Critical thinking about methods and analyses</a><ul>
<li class="chapter" data-level="11.1" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#how-to-critically-evaluate-a-paper-that-uses-phylogenetic-comparative-methods"><i class="fa fa-check"></i><b>11.1</b> How to critically evaluate a paper that uses phylogenetic comparative methods</a><ul>
<li class="chapter" data-level="" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#logicinterpretation"><i class="fa fa-check"></i>Logic/interpretation</a></li>
<li class="chapter" data-level="" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#data"><i class="fa fa-check"></i>Data</a></li>
<li class="chapter" data-level="" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#methods"><i class="fa fa-check"></i>Methods</a></li>
<li class="chapter" data-level="" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#moving-forwards"><i class="fa fa-check"></i>Moving forwards</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="critical-thinking-about-methods-and-analyses.html"><a href="critical-thinking-about-methods-and-analyses.html#further-reading"><i class="fa fa-check"></i><b>11.2</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html"><i class="fa fa-check"></i><b>12</b> Answers to practical exercises</a><ul>
<li class="chapter" data-level="12.1" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#diversity-indices"><i class="fa fa-check"></i><b>12.1</b> Diversity indices</a></li>
<li class="chapter" data-level="12.2" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#phylogenies"><i class="fa fa-check"></i><b>12.2</b> Phylogenies</a></li>
<li class="chapter" data-level="12.3" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#preparation-8"><i class="fa fa-check"></i><b>12.3</b> Preparation</a></li>
<li class="chapter" data-level="12.4" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#phylogenetic-signal"><i class="fa fa-check"></i><b>12.4</b> Phylogenetic signal</a></li>
<li class="chapter" data-level="12.5" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#pgls"><i class="fa fa-check"></i><b>12.5</b> PGLS</a></li>
<li class="chapter" data-level="12.6" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#models-of-evolution-with-continuous-traits"><i class="fa fa-check"></i><b>12.6</b> Models of evolution with continuous traits</a></li>
<li class="chapter" data-level="12.7" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#models-of-evolution-with-discrete-traits"><i class="fa fa-check"></i><b>12.7</b> Models of evolution with discrete traits</a></li>
<li class="chapter" data-level="12.8" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#beyond-simple-macroevolutionary-models-of-trait-evolution-1"><i class="fa fa-check"></i><b>12.8</b> Beyond simple macroevolutionary models of trait evolution</a></li>
<li class="chapter" data-level="12.9" data-path="answers-to-practical-exercises.html"><a href="answers-to-practical-exercises.html#diversification"><i class="fa fa-check"></i><b>12.9</b> Diversification</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Practical exercises for Methods in Macroecology and Macroevolution module</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="phylogenetic-signal-in-r" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Phylogenetic Signal in R</h1>
<p>Phylogenetic signal is the pattern where close relatives have more similar trait values than more distant relatives. The aims of this exercise are to learn how to use R to estimate phylogenetic signal using Pagel’s <span class="math inline">\(\lambda\)</span> <span class="citation">(Pagel <a href="#ref-pagel1997inferring" role="doc-biblioref">1997</a>, @pagel1999inferring)</span> and Blomberg’s <em>K</em> <span class="citation">(Blomberg, Garland Jr, and Ives <a href="#ref-blomberg2003testing" role="doc-biblioref">2003</a>)</span>.</p>
<div class="info">
<p>
We will be using the evolution of eye size in frogs as an example. The data and modified tree come from <span class="citation"><span class="citation">Thomas et al. (<a href="#ref-thomas2020eye" role="doc-biblioref">2020</a>)</span></span>, and the original tree comes from <span class="citation"><span class="citation">Feng et al. (<a href="#ref-feng2017phylogenomics" role="doc-biblioref">2017</a>)</span></span>. I’ve removed a few species and a few variables to make things a bit more straightforward. If you want to see the full results check out <span class="citation"><span class="citation">Thomas et al. (<a href="#ref-thomas2020eye" role="doc-biblioref">2020</a>)</span></span>!
</p>
</div>
<div id="before-you-start-3" class="section level2">
<h2><span class="header-section-number">5.1</span> Before you start</h2>
<ul>
<li>Open the <code>05-PhyloSignal.RProj</code> file in the <code>05-PhyloSignal</code> folder to open your R Project for this exercise.</li>
<li>Make yourself a new R script for your code.</li>
</ul>
<p>You will also need to install the following packages:</p>
<ul>
<li><code>tidyverse</code></li>
<li><code>ape</code></li>
<li><code>geiger</code></li>
<li><code>treeplyr</code></li>
<li><code>phytools</code></li>
<li><code>caper</code></li>
</ul>
</div>
<div id="preparation-2" class="section level2">
<h2><span class="header-section-number">5.2</span> Preparation</h2>
<p>To begin we need to load the packages for this practical.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="phylogenetic-signal-in-r.html#cb141-1"></a><span class="co"># Load the packages</span></span>
<span id="cb141-2"><a href="phylogenetic-signal-in-r.html#cb141-2"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb141-3"><a href="phylogenetic-signal-in-r.html#cb141-3"></a><span class="kw">library</span>(ape)</span>
<span id="cb141-4"><a href="phylogenetic-signal-in-r.html#cb141-4"></a><span class="kw">library</span>(geiger)</span>
<span id="cb141-5"><a href="phylogenetic-signal-in-r.html#cb141-5"></a><span class="kw">library</span>(treeplyr)</span>
<span id="cb141-6"><a href="phylogenetic-signal-in-r.html#cb141-6"></a><span class="kw">library</span>(phytools)</span>
<span id="cb141-7"><a href="phylogenetic-signal-in-r.html#cb141-7"></a><span class="kw">library</span>(caper)</span></code></pre></div>
<p>Next we need to prepare the tree and data for the analyses. In the <code>04-Preparation</code> exercise we read in our tree and data, checked them, and matched them so only species in both were retained. Please refer to that exercise for more details on how and why we do these things, or run through it now if you haven’t previously.</p>
<p>It is important to do these things before beginning a phylogenetic comparative analysis, so let’s run through that code again here.</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="phylogenetic-signal-in-r.html#cb142-1"></a><span class="co"># Read in the data</span></span>
<span id="cb142-2"><a href="phylogenetic-signal-in-r.html#cb142-2"></a>frogdata &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="st">&quot;data/frog-eyes.csv&quot;</span>)</span>
<span id="cb142-3"><a href="phylogenetic-signal-in-r.html#cb142-3"></a><span class="co"># Check everything loaded corrected</span></span>
<span id="cb142-4"><a href="phylogenetic-signal-in-r.html#cb142-4"></a><span class="kw">glimpse</span>(frogdata)</span></code></pre></div>
<pre><code>## Rows: 215
## Columns: 11
## $ Binomial         &lt;chr&gt; &quot;Allophryne_ruthveni&quot;, &quot;Eupsophus_roseus&quot;, &quot;Aly…
## $ Family           &lt;chr&gt; &quot;Allophrynidae&quot;, &quot;Alsodidae&quot;, &quot;Alytidae&quot;, &quot;Alyt…
## $ Genus            &lt;chr&gt; &quot;Allophryne&quot;, &quot;Eupsophus&quot;, &quot;Alytes&quot;, &quot;Discoglos…
## $ tiplabel         &lt;chr&gt; &quot;Allophryne_ruthveni_Allophrynidae&quot;, &quot;Eupsophus…
## $ Adult_habitat    &lt;chr&gt; &quot;Scansorial&quot;, &quot;Ground-dwelling&quot;, &quot;Ground-dwelli…
## $ Life_history     &lt;chr&gt; &quot;Free-living larvae&quot;, &quot;Free-living larvae&quot;, &quot;Fr…
## $ Sex_dichromatism &lt;chr&gt; &quot;Absent&quot;, &quot;Absent&quot;, &quot;Absent&quot;, &quot;Absent&quot;, &quot;Absent…
## $ SVL              &lt;dbl&gt; 23.76667, 38.37500, 37.46667, 62.64000, 25.6000…
## $ mass             &lt;dbl&gt; 1.0000000, 7.5000000, 6.6666667, 24.4000000, 2.…
## $ rootmass         &lt;dbl&gt; 0.9917748, 1.9273451, 1.8690601, 2.8885465, 1.2…
## $ eyesize          &lt;dbl&gt; 3.200000, 5.362500, 6.366667, 7.550000, 4.11666…</code></pre>
<p>To load the tree we will use <code>read.nexus</code>.</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="phylogenetic-signal-in-r.html#cb144-1"></a><span class="co"># Read in the tree</span></span>
<span id="cb144-2"><a href="phylogenetic-signal-in-r.html#cb144-2"></a>frogtree &lt;-<span class="st"> </span><span class="kw">read.nexus</span>(<span class="st">&quot;data/frog-tree.nex&quot;</span>)</span>
<span id="cb144-3"><a href="phylogenetic-signal-in-r.html#cb144-3"></a><span class="co"># Check it loaded correctly</span></span>
<span id="cb144-4"><a href="phylogenetic-signal-in-r.html#cb144-4"></a><span class="kw">str</span>(frogtree)</span></code></pre></div>
<pre><code>## List of 4
##  $ edge       : int [1:426, 1:2] 215 216 217 218 219 220 221 222 223 224 ...
##  $ edge.length: num [1:426] 0.166 0.114 0.102 0.4 0.133 ...
##  $ Nnode      : int 213
##  $ tip.label  : chr [1:214] &quot;Ascaphus_truei_Ascaphidae&quot; &quot;Leiopelma_hochstetteri_Leiopelmatidae&quot; &quot;Alytes_obstetricans_Alytidae&quot; &quot;Discoglossus_pictus_Alytidae&quot; ...
##  - attr(*, &quot;class&quot;)= chr &quot;phylo&quot;
##  - attr(*, &quot;order&quot;)= chr &quot;cladewise&quot;</code></pre>
<p>Remember to check the tree is dichotomous, i.e. has no polytomies, rooted, and ultrametric.</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="phylogenetic-signal-in-r.html#cb146-1"></a><span class="co"># Check whether the tree is binary</span></span>
<span id="cb146-2"><a href="phylogenetic-signal-in-r.html#cb146-2"></a><span class="co"># We want this to be TRUE</span></span>
<span id="cb146-3"><a href="phylogenetic-signal-in-r.html#cb146-3"></a><span class="kw">is.binary.tree</span>(frogtree)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="phylogenetic-signal-in-r.html#cb148-1"></a><span class="co"># Check whether the tree is rooted </span></span>
<span id="cb148-2"><a href="phylogenetic-signal-in-r.html#cb148-2"></a><span class="co"># We want this to be TRUE </span></span>
<span id="cb148-3"><a href="phylogenetic-signal-in-r.html#cb148-3"></a><span class="kw">is.rooted</span>(frogtree)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="phylogenetic-signal-in-r.html#cb150-1"></a><span class="co"># Check whether the tree is ultrametric</span></span>
<span id="cb150-2"><a href="phylogenetic-signal-in-r.html#cb150-2"></a><span class="co"># We want this to be TRUE </span></span>
<span id="cb150-3"><a href="phylogenetic-signal-in-r.html#cb150-3"></a><span class="kw">is.ultrametric</span>(frogtree)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>Next check that the species names match up in the tree and the data. This should reveal any typos and/or taxonomic differences that need to be fixed before going any further.</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="phylogenetic-signal-in-r.html#cb152-1"></a><span class="co"># Check whether the names match in the data and the tree</span></span>
<span id="cb152-2"><a href="phylogenetic-signal-in-r.html#cb152-2"></a>check &lt;-<span class="st"> </span><span class="kw">name.check</span>(<span class="dt">phy =</span> frogtree, <span class="dt">data =</span> frogdata, </span>
<span id="cb152-3"><a href="phylogenetic-signal-in-r.html#cb152-3"></a>                    <span class="dt">data.names =</span> frogdata<span class="op">$</span>tiplabel)</span>
<span id="cb152-4"><a href="phylogenetic-signal-in-r.html#cb152-4"></a><span class="co"># Look at check</span></span>
<span id="cb152-5"><a href="phylogenetic-signal-in-r.html#cb152-5"></a>check</span></code></pre></div>
<pre><code>## $tree_not_data
## [1] &quot;Incilius_nebulifer_Bufonidae&quot;           
## [2] &quot;Leptobrachella_bidoupensis_Megophryidae&quot;
## [3] &quot;Microhyla_fissipes_Microhylidae&quot;        
## [4] &quot;Microhyla_marmorata_Microhylidae&quot;       
## 
## $data_not_tree
## [1] &quot;Gastrophryne_carolinensis_Microhylidae&quot;
## [2] &quot;Leptobrachella_dringi_Megophryidae&quot;    
## [3] &quot;Megophrys_gerti_Megophryidae&quot;          
## [4] &quot;Microhyla_pulverata_Microhylidae&quot;      
## [5] &quot;Oreobates_quixensis_Strabomantidae&quot;</code></pre>
<p>Here all the excluded species are excluded because they are genuinely missing, not because of any typos, so we can move on.</p>
<p>Next we combine the tree and data to exclude species that are not in both, using the <code>treeplyr</code> function <code>make.treedata</code>.</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="phylogenetic-signal-in-r.html#cb154-1"></a><span class="co"># Combine and match the tree and data</span></span>
<span id="cb154-2"><a href="phylogenetic-signal-in-r.html#cb154-2"></a>frogstuff &lt;-<span class="st"> </span><span class="kw">make.treedata</span>(<span class="dt">tree =</span> frogtree,  <span class="dt">data =</span> frogdata, </span>
<span id="cb154-3"><a href="phylogenetic-signal-in-r.html#cb154-3"></a>                              <span class="dt">name_column =</span> <span class="st">&quot;tiplabel&quot;</span>)</span>
<span id="cb154-4"><a href="phylogenetic-signal-in-r.html#cb154-4"></a></span>
<span id="cb154-5"><a href="phylogenetic-signal-in-r.html#cb154-5"></a><span class="co"># Look at the tree</span></span>
<span id="cb154-6"><a href="phylogenetic-signal-in-r.html#cb154-6"></a>frogstuff<span class="op">$</span>phy</span></code></pre></div>
<pre><code>## 
## Phylogenetic tree with 210 tips and 209 internal nodes.
## 
## Tip labels:
##   Ascaphus_truei_Ascaphidae, Leiopelma_hochstetteri_Leiopelmatidae, Alytes_obstetricans_Alytidae, Discoglossus_pictus_Alytidae, Barbourula_busuangensis_Bombinatoridae, Bombina_orientalis_Bombinatoridae, ...
## 
## Rooted; includes branch lengths.</code></pre>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="phylogenetic-signal-in-r.html#cb156-1"></a><span class="co"># Look at the data</span></span>
<span id="cb156-2"><a href="phylogenetic-signal-in-r.html#cb156-2"></a><span class="kw">glimpse</span>(frogstuff<span class="op">$</span>dat)</span></code></pre></div>
<pre><code>## Rows: 210
## Columns: 10
## $ Binomial         &lt;fct&gt; Ascaphus_truei, Leiopelma_hochstetteri, Alytes_…
## $ Family           &lt;fct&gt; Ascaphidae, Leiopelmatidae, Alytidae, Alytidae,…
## $ Genus            &lt;fct&gt; Ascaphus, Leiopelma, Alytes, Discoglossus, Barb…
## $ Adult_habitat    &lt;fct&gt; Semiaquatic, Semiaquatic, Ground-dwelling, Grou…
## $ Life_history     &lt;fct&gt; Free-living larvae, Free-living larvae, Free-li…
## $ Sex_dichromatism &lt;fct&gt; Absent, Absent, Absent, Absent, Absent, Absent,…
## $ SVL              &lt;dbl&gt; 38.95000, 38.96667, 37.46667, 62.64000, 59.4750…
## $ mass             &lt;dbl&gt; 6.000000, 5.333333, 6.666667, 24.400000, 24.250…
## $ rootmass         &lt;dbl&gt; 1.809268, 1.740547, 1.869060, 2.888547, 2.73310…
## $ eyesize          &lt;dbl&gt; 5.587500, 6.283333, 6.366667, 7.550000, 8.23750…</code></pre>
<p>Then we replace the removed species names column, and save the data as a dataframe for later analyses.</p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="phylogenetic-signal-in-r.html#cb158-1"></a><span class="co"># Make a new column called tiplabel with the tip labels in it</span></span>
<span id="cb158-2"><a href="phylogenetic-signal-in-r.html#cb158-2"></a>frogstuff<span class="op">$</span>dat<span class="op">$</span>tiplabel &lt;-<span class="st"> </span>frogstuff<span class="op">$</span>phy<span class="op">$</span>tip.label</span>
<span id="cb158-3"><a href="phylogenetic-signal-in-r.html#cb158-3"></a><span class="co"># Force mydata to be a data frame</span></span>
<span id="cb158-4"><a href="phylogenetic-signal-in-r.html#cb158-4"></a>mydata &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(frogstuff<span class="op">$</span>dat)</span></code></pre></div>
<p>Finally we save the tree:</p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="phylogenetic-signal-in-r.html#cb159-1"></a><span class="co"># Save tree as mytree</span></span>
<span id="cb159-2"><a href="phylogenetic-signal-in-r.html#cb159-2"></a>mytree &lt;-<span class="st"> </span>frogstuff<span class="op">$</span>phy</span></code></pre></div>
<p>Note that is is possible to use the <code>make.treedata</code> object directly to run some analyses, but we have saved these separately here as many PCMs cannot work with this.</p>
<p>Now we’re ready to run our analyses!</p>
</div>
<div id="estimating-phylogenetic-signal-for-continuous-variables" class="section level2">
<h2><span class="header-section-number">5.3</span> Estimating phylogenetic signal for continuous variables</h2>
<p>As is common in R, there are a number of ways to estimate Pagel’s <span class="math inline">\(\lambda\)</span> and Blomberg’s <em>K</em>. I’ve chosen to show you the way implemented in the package <code>phytools</code> because it allows you to use the same function for both.</p>
<div class="info">
<p>
Why do we look at two different measures of phylogenetic signal? We don’t have to, you could choose one and stick to it. However, I guarantee if you choose one then a reviewer/your supervisor/boss will ask for the other, so why not do both and pop one in the appendix?
</p>
</div>
<p>Let’s estimate <span class="math inline">\(\lambda\)</span> for log eye size.</p>
<p>The first thing we need to do is to create an object in R that only contains the variable required, and the species names (so we can match it up to the tree).</p>
<p>We can use the function <code>pull</code> to extract just the eye size values, and we can log transform all these numbers using <code>log</code> if we want to work with log eye size values.</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="phylogenetic-signal-in-r.html#cb160-1"></a><span class="co"># Create logEye containing just log eye size length values</span></span>
<span id="cb160-2"><a href="phylogenetic-signal-in-r.html#cb160-2"></a>logEye &lt;-<span class="st"> </span><span class="kw">log</span>(<span class="kw">pull</span>(mydata, eyesize))</span>
<span id="cb160-3"><a href="phylogenetic-signal-in-r.html#cb160-3"></a><span class="co"># Look at the first few rows</span></span>
<span id="cb160-4"><a href="phylogenetic-signal-in-r.html#cb160-4"></a><span class="kw">head</span>(logEye)</span></code></pre></div>
<pre><code>## [1] 1.720532 1.837901 1.851076 2.021548 2.108697 1.740466</code></pre>
<p>Notice that this is currently just a long list of numbers. We can then name these values with the species names from <code>mydata</code> using the function <code>names</code>. Note that this requires the trait data is in the <em>same order</em> as the tree tip labels, but luckily <code>make.treedata</code> does this automatically.</p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="phylogenetic-signal-in-r.html#cb162-1"></a><span class="co"># Give log Eye names = species names at the tips of the phylogeny</span></span>
<span id="cb162-2"><a href="phylogenetic-signal-in-r.html#cb162-2"></a><span class="kw">names</span>(logEye) &lt;-<span class="st"> </span>mydata<span class="op">$</span>tiplabel</span>
<span id="cb162-3"><a href="phylogenetic-signal-in-r.html#cb162-3"></a><span class="co"># Look at the first few rows</span></span>
<span id="cb162-4"><a href="phylogenetic-signal-in-r.html#cb162-4"></a><span class="kw">head</span>(logEye)</span></code></pre></div>
<pre><code>##              Ascaphus_truei_Ascaphidae 
##                               1.720532 
##  Leiopelma_hochstetteri_Leiopelmatidae 
##                               1.837901 
##           Alytes_obstetricans_Alytidae 
##                               1.851076 
##           Discoglossus_pictus_Alytidae 
##                               2.021548 
## Barbourula_busuangensis_Bombinatoridae 
##                               2.108697 
##      Bombina_orientalis_Bombinatoridae 
##                               1.740466</code></pre>
<p>Now we have a list of values with associated species names.</p>
<div id="pagels-lambda" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Pagel’s <span class="math inline">\(\lambda\)</span></h3>
<p>We can now estimate <span class="math inline">\(\lambda\)</span> using the function <code>phylosig</code>:</p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="phylogenetic-signal-in-r.html#cb164-1"></a><span class="co"># Estimate lambda</span></span>
<span id="cb164-2"><a href="phylogenetic-signal-in-r.html#cb164-2"></a>lambdaEye &lt;-<span class="st"> </span><span class="kw">phylosig</span>(mytree, logEye, <span class="dt">method =</span> <span class="st">&quot;lambda&quot;</span>, <span class="dt">test =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><code>test = TRUE</code> specifies that we want to run a likelihood ratio test to determine if <span class="math inline">\(\lambda\)</span> is significantly different from 0.</p>
<p>To look at the output we just type in the name of the model; <code>lambdaEye</code> in this case</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="phylogenetic-signal-in-r.html#cb165-1"></a>lambdaEye</span></code></pre></div>
<pre><code>## 
## Phylogenetic signal lambda : 0.802864 
## logL(lambda) : -112.137 
## LR(lambda=0) : 62.803 
## P-value (based on LR test) : 2.28447e-15</code></pre>
<p>The <span class="math inline">\(\lambda\)</span> estimate for log eye size is around 0.803. <code>logL</code> is the log-likelihood, <code>LR(lambda=0)</code> is the log-likelihood for <span class="math inline">\(\lambda\)</span> of 0, and <code>P-value</code> is the p value from a likelihood ratio test testing whether <span class="math inline">\(\lambda\)</span> is significantly different from 0 (no phylogenetic signal).</p>
<p>Here P &lt; 0.001. We interpret this as <span class="math inline">\(\lambda\)</span> being significantly different from 0, i.e. there is significant phylogenetic signal in log eye size.</p>
</div>
<div id="blombergs-k-blomberg-et-al-2003" class="section level3">
<h3><span class="header-section-number">5.3.2</span> Blomberg’s <em>K</em> (Blomberg et al 2003)</h3>
<p>To estimate Blomberg’s <em>K</em> we also use <code>phylosig</code> but with <code>method = K</code>.</p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="phylogenetic-signal-in-r.html#cb167-1"></a><span class="co"># Estimate Blomberg’s *K*</span></span>
<span id="cb167-2"><a href="phylogenetic-signal-in-r.html#cb167-2"></a>KEye &lt;-<span class="st"> </span><span class="kw">phylosig</span>(mytree, logEye, <span class="dt">method =</span> <span class="st">&quot;K&quot;</span>, <span class="dt">test =</span> <span class="ot">TRUE</span>, <span class="dt">nsim =</span> <span class="dv">1000</span>)</span></code></pre></div>
<p>Additionally we add the argument <code>nsim = 1000</code>. This is because we need to use a randomisation test to determine whether <em>K</em> is significantly different from 0. <code>phylosig</code> randomly assigns the trait values to the species and then calculates <em>K</em> as many times as we ask it to in <code>nsim</code> (number of simulations). Here we asked for 1000. After the random simulations are run, the observed value of K is then compared to the randomized values. The p value tells us how many times out of 1000, a randomised value of <em>K</em> is more extreme than the observed value. If this number is low, the p value is low (e.g. if 5 out of 1000 randomised values of <em>K</em> are more extreme than the observed value p = 5/1000 = 0.005).</p>
<p>As above, to look at the output we just type in the name of the model; <code>KEye</code> in this case</p>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="phylogenetic-signal-in-r.html#cb168-1"></a>KEye</span></code></pre></div>
<pre><code>## 
## Phylogenetic signal K : 0.283009 
## P-value (based on 1000 randomizations) : 0.001</code></pre>
<p><em>K</em> for log eye size is 0.283. The p value tells us how many times out of 1000, a randomised value of <em>K</em> is more extreme than the observed value. If this number is low, the p value is low (e.g. if 5 out of 1000 randomised values of <em>K</em> are more extreme than the observed value p = 5/1000 = 0.005). Here p = 0.001, suggesting that only 1 randomised value of <em>K</em> was more extreme than the observed value.</p>
<p>We interpret this as <em>K</em> being significantly different from 0, i.e. there is significant phylogenetic signal in log eye size.</p>
<div class="warning">
<p>
Remember that when fitting models to account for phylogenetic non-independence, it is <em>not</em> phylogenetic signal in the individual variables that is important. It is phylogenetic signal in the <strong>residuals</strong> of the model that matters. Evidence of phylogenetic signal in variable X (or variable Y) does not necessarily mean that there will be phylogenetic signal in the residuals of a model correlating variable X with variable Y. Conversely, lack of evidence of phylogenetic signal in variable X (or variable Y) does not necessarily mean that there will be no phylogenetic signal in the residuals of a model correlating variable X with variable Y.
</p>
</div>
</div>
</div>
<div id="estimating-phylogenetic-signal-for-non-continuous-variables" class="section level2">
<h2><span class="header-section-number">5.4</span> Estimating phylogenetic signal for non-continuous variables</h2>
<p>Not all variables are continuous, some are categorical, some are binary. For example in the frog data <code>Adult_habitat</code>, <code>Life_history</code> and <code>Sex_dichromatism</code> are categorical variables. We can also code <code>Sex_dichromatism</code> as a binary variable if we code Absent as 0 and Present as 1.</p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="phylogenetic-signal-in-r.html#cb170-1"></a><span class="kw">glimpse</span>(mydata)</span></code></pre></div>
<pre><code>## Rows: 210
## Columns: 11
## $ Binomial         &lt;fct&gt; Ascaphus_truei, Leiopelma_hochstetteri, Alytes_…
## $ Family           &lt;fct&gt; Ascaphidae, Leiopelmatidae, Alytidae, Alytidae,…
## $ Genus            &lt;fct&gt; Ascaphus, Leiopelma, Alytes, Discoglossus, Barb…
## $ Adult_habitat    &lt;fct&gt; Semiaquatic, Semiaquatic, Ground-dwelling, Grou…
## $ Life_history     &lt;fct&gt; Free-living larvae, Free-living larvae, Free-li…
## $ Sex_dichromatism &lt;fct&gt; Absent, Absent, Absent, Absent, Absent, Absent,…
## $ SVL              &lt;dbl&gt; 38.95000, 38.96667, 37.46667, 62.64000, 59.4750…
## $ mass             &lt;dbl&gt; 6.000000, 5.333333, 6.666667, 24.400000, 24.250…
## $ rootmass         &lt;dbl&gt; 1.809268, 1.740547, 1.869060, 2.888547, 2.73310…
## $ eyesize          &lt;dbl&gt; 5.587500, 6.283333, 6.366667, 7.550000, 8.23750…
## $ tiplabel         &lt;chr&gt; &quot;Ascaphus_truei_Ascaphidae&quot;, &quot;Leiopelma_hochste…</code></pre>
<p>Estimating phylogenetic signal in categorical variables is tricky. Let’s take <code>Adult_habitat</code> as an example. In this variable we have the following categories:</p>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="phylogenetic-signal-in-r.html#cb172-1"></a>mydata <span class="op">%&gt;%</span></span>
<span id="cb172-2"><a href="phylogenetic-signal-in-r.html#cb172-2"></a><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(Adult_habitat) <span class="op">%&gt;%</span></span>
<span id="cb172-3"><a href="phylogenetic-signal-in-r.html#cb172-3"></a><span class="st"> </span><span class="kw">distinct</span>()</span></code></pre></div>
<pre><code>##     Adult_habitat
## 1     Semiaquatic
## 2 Ground-dwelling
## 3         Aquatic
## 4       Fossorial
## 5    Subfossorial
## 6      Scansorial</code></pre>
<p>Remember with phylogenetic signal we are looking for the pattern where close relatives are more similar to one another than more distant relatives. For these categories, it would be sensible to say we had high phylogenetic signal if, for example, all toads are Semiaquatic, and all tree frogs are Scansorial (climbers). But what if some toads are Semiaquatic but others are Ground-dwelling? If we don’t know how species transition from state to state, it’s hard to know what we might expect to see as evolutionary distance between species increases. We might sensibly assume that species easily evolve from Semiaquatic to Aquatic, or from Subfossorial to Fossorial (burrowing), but what about changes from Aquatic to Fossorial? All of this makes phylogenetic signal for categorical variables a bit of a mess.</p>
<p>There are a couple of, more or less satisfying, solutions…</p>
<ol style="list-style-type: decimal">
<li><p>Do we actually <em>need</em> to know the phylogenetic signal for these variables? For the models we are interested in phylogenetic signal in the <em>residuals</em>, so maybe we don’t care about phylogenetic signal in the variables? Unless there is a real need, don’t bother! You could visualise what is going on instead by adding colours to the tips of the phylogeny to represent the different categories. This should give you an idea about whether categories cluster in different clades or not.</p></li>
<li><p>We could code our categories numerically then use <span class="math inline">\(\lambda\)</span> and <em>K</em> as usual. <em>This is only suitable if the categories are ordered, and if the difference between each pair of categories can be considered equal</em>. For example, a variable that has low, medium and high values could be coded as low = 1, medium = 2, and high = 3. Again this is not ideal, but will give you an answer. I’d avoid this if possible.</p></li>
<li><p>We could recode these as binary variables and use D <span class="citation">(Fritz and Purvis <a href="#ref-fritz2010selectivity" role="doc-biblioref">2010</a>)</span>. For example, creating a new variable called Aquatic, and coding each species as 0 = not aquatic; 1 = aquatic. This is probably the best solution.</p></li>
</ol>
<div id="d" class="section level3">
<h3><span class="header-section-number">5.4.1</span> <em>D</em></h3>
<p>We can estimate D using the function <code>phylo.d</code> in the <code>caper</code> package.</p>
<p>First we’d need to set up a binary (0,1) variable. Let’s look at <code>Sex_dichromatism</code> as this only has two categories already (Absent and Present).</p>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="phylogenetic-signal-in-r.html#cb174-1"></a>mydata <span class="op">%&gt;%</span></span>
<span id="cb174-2"><a href="phylogenetic-signal-in-r.html#cb174-2"></a><span class="st">  </span><span class="kw">group_by</span>(Sex_dichromatism) <span class="op">%&gt;%</span></span>
<span id="cb174-3"><a href="phylogenetic-signal-in-r.html#cb174-3"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="kw">n</span>())</span></code></pre></div>
<pre><code>## `summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre><code>## # A tibble: 3 x 2
##   Sex_dichromatism `n()`
##   &lt;fct&gt;            &lt;int&gt;
## 1 Absent             150
## 2 Present             24
## 3 &lt;NA&gt;                36</code></pre>
<p>There are 36 species without data for this variable. If we were doing something like PGLS (see next exercise), <code>caper</code> would be fine with this, but for <code>phylo.d</code> we can only work with data without NAs. To fix this, we can use <code>filter</code> to exclude the species with NAs, but we also want to remove these species from the tree. It’s easiest to go back to our <code>treeplyr</code> functions to do this.</p>
<p>Let’s first go back to <code>frogstuff</code>, the combined object with our data and our tree in it, so we can exclude the species without <code>Sex_dichromatism</code> data, making <code>frogstuff2</code>.</p>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="phylogenetic-signal-in-r.html#cb177-1"></a><span class="co"># Filter out species with no Sex_dichromatism data</span></span>
<span id="cb177-2"><a href="phylogenetic-signal-in-r.html#cb177-2"></a>frogstuff2 &lt;-</span>
<span id="cb177-3"><a href="phylogenetic-signal-in-r.html#cb177-3"></a><span class="st">  </span>frogstuff <span class="op">%&gt;%</span></span>
<span id="cb177-4"><a href="phylogenetic-signal-in-r.html#cb177-4"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(Sex_dichromatism))</span></code></pre></div>
<p>Now we can go back to our usual workflow, but using <code>frogstuff2</code>.</p>
<p>Replace the removed species names column, save the data as a dataframe for later analyses, and save the tree.</p>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="phylogenetic-signal-in-r.html#cb178-1"></a><span class="co"># Make a new column called tiplabel with the tip labels in it</span></span>
<span id="cb178-2"><a href="phylogenetic-signal-in-r.html#cb178-2"></a>frogstuff2<span class="op">$</span>dat<span class="op">$</span>tiplabel &lt;-<span class="st"> </span>frogstuff2<span class="op">$</span>phy<span class="op">$</span>tip.label</span>
<span id="cb178-3"><a href="phylogenetic-signal-in-r.html#cb178-3"></a><span class="co"># Force mydata to be a data frame</span></span>
<span id="cb178-4"><a href="phylogenetic-signal-in-r.html#cb178-4"></a>mydata &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(frogstuff2<span class="op">$</span>dat)</span>
<span id="cb178-5"><a href="phylogenetic-signal-in-r.html#cb178-5"></a><span class="co"># Save tree as mytree</span></span>
<span id="cb178-6"><a href="phylogenetic-signal-in-r.html#cb178-6"></a>mytree &lt;-<span class="st"> </span>frogstuff2<span class="op">$</span>phy</span></code></pre></div>
<p>We also need the variable Sex_dichromatism to be either 0 or 1. We can do this fairly easily using mutate to create a new variable called <code>sex_di_binary</code>.</p>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="phylogenetic-signal-in-r.html#cb179-1"></a>mydata &lt;-<span class="st"> </span></span>
<span id="cb179-2"><a href="phylogenetic-signal-in-r.html#cb179-2"></a><span class="st">  </span>mydata <span class="op">%&gt;%</span></span>
<span id="cb179-3"><a href="phylogenetic-signal-in-r.html#cb179-3"></a><span class="st">  </span><span class="co"># Make a new variable called sex_di_binary</span></span>
<span id="cb179-4"><a href="phylogenetic-signal-in-r.html#cb179-4"></a><span class="st">  </span><span class="co"># which is Sex_dichromatism expressed as numbers</span></span>
<span id="cb179-5"><a href="phylogenetic-signal-in-r.html#cb179-5"></a><span class="st">  </span><span class="co"># Absent = 1 and Present = 2. To make these 0 and 1 instead we just use -1</span></span>
<span id="cb179-6"><a href="phylogenetic-signal-in-r.html#cb179-6"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sex_di_binary =</span> <span class="kw">as.numeric</span>(Sex_dichromatism) <span class="op">-</span><span class="st"> </span><span class="dv">1</span>)</span></code></pre></div>
<p>To use <code>phylo.d</code> we need the <code>caper</code> package. <code>caper</code> requires you to first combine the phylogeny and data into one object using the function <code>comparative.data</code>. This is similar to what we did with <code>make.treedata</code>, but it does some stuff that is particular to how <code>caper</code> works so we still need to do this here.</p>
<p>Note that <code>vcv = TRUE</code> stores a variance covariance matrix of your tree (you will need this for the <code>pgls</code> function in the next exercise). <code>na.omit = FALSE</code> stops the function from removing species without data for all variables. <code>warn.dropped = TRUE</code> will tell you if any species are not in both the tree and the data and are therefore dropped from the comparative data object.</p>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="phylogenetic-signal-in-r.html#cb180-1"></a>frog &lt;-<span class="st"> </span><span class="kw">comparative.data</span>(<span class="dt">phy =</span> mytree, <span class="dt">data =</span> mydata, </span>
<span id="cb180-2"><a href="phylogenetic-signal-in-r.html#cb180-2"></a>                            <span class="dt">names.col =</span> tiplabel, <span class="dt">vcv =</span> <span class="ot">TRUE</span>, </span>
<span id="cb180-3"><a href="phylogenetic-signal-in-r.html#cb180-3"></a>                            <span class="dt">na.omit =</span> <span class="ot">FALSE</span>, <span class="dt">warn.dropped =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p>This function will give a warning telling you that some species have been dropped. <strong>Always</strong> make sure you check the list of dropped species is what you expected, as it often reveals typos in your species names, or mismatches in taxonomies used etc.</p>
<p>You can view the dropped species using:</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="phylogenetic-signal-in-r.html#cb181-1"></a>frog<span class="op">$</span>dropped<span class="op">$</span>unmatched.rows</span></code></pre></div>
<pre><code>## character(0)</code></pre>
<p>This shows nothing because we already fixed the data so that all the species in the data are in the tree.</p>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="phylogenetic-signal-in-r.html#cb183-1"></a>frog<span class="op">$</span>dropped<span class="op">$</span>tips</span></code></pre></div>
<pre><code>## character(0)</code></pre>
<p>This also shows nothing because we already fixed the tree so that all the species in the tree are in the data.</p>
<p>One last bit of prep required is that <em>D</em> cannot be estimated where trees have zero length branches. In <code>04-Preparation</code> you may recall that we dealt with polytomies by replacing them with zero length branches using the <code>ape</code> function <code>mutli2di</code>. To remove the zero length branches we are going to use the opposite function: <code>di2multi</code>. This deletes all zero length branches and collapses them back into polytomies.</p>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="phylogenetic-signal-in-r.html#cb185-1"></a>frog<span class="op">$</span>phy &lt;-<span class="st"> </span><span class="kw">di2multi</span>(frog<span class="op">$</span>phy)</span></code></pre></div>
<p>Now we are ready to to estimate <em>D</em>.</p>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="phylogenetic-signal-in-r.html#cb186-1"></a><span class="co"># Estimate D</span></span>
<span id="cb186-2"><a href="phylogenetic-signal-in-r.html#cb186-2"></a>Dsexdi &lt;-<span class="st"> </span><span class="kw">phylo.d</span>(<span class="dt">data =</span> frog, <span class="dt">names.col =</span> tiplabel, <span class="dt">binvar =</span> sex_di_binary, </span>
<span id="cb186-3"><a href="phylogenetic-signal-in-r.html#cb186-3"></a>                  <span class="dt">permut =</span> <span class="dv">1000</span>)</span>
<span id="cb186-4"><a href="phylogenetic-signal-in-r.html#cb186-4"></a></span>
<span id="cb186-5"><a href="phylogenetic-signal-in-r.html#cb186-5"></a><span class="co"># Look at the output</span></span>
<span id="cb186-6"><a href="phylogenetic-signal-in-r.html#cb186-6"></a>Dsexdi</span></code></pre></div>
<pre><code>## 
## Calculation of D statistic for the phylogenetic structure of a binary variable
## 
##   Data :  mydata
##   Binary variable :  sex_di_binary
##   Counts of states:  0 = 150
##                      1 = 24
##   Phylogeny :  mytree
##   Number of permutations :  1000
## 
## Estimated D :  0.8709497
## Probability of E(D) resulting from no (random) phylogenetic structure :  0.2
## Probability of E(D) resulting from Brownian phylogenetic structure    :  0.001</code></pre>
<p><code>phylo.d</code> estimates the D value, then tests the estimated D value for significant departure from random association and a Brownian evolution threshold model (see <em>Primer</em> and <span class="citation">Fritz and Purvis (<a href="#ref-fritz2010selectivity" role="doc-biblioref">2010</a>)</span> for more details).</p>
<p>Here <em>D</em> for <code>Sex_dichromatism</code> is 0.871. This is significantly different from the expectation under a Brownian threshold model (p = 0.001) but not significantly different from the trait being randomly assorted on the phylogeny (p = 0.2). So here we can say there is no significant phylogenetic signal in the trait.</p>
<p>We can make a really quick plot to check we agree with that statement (never take the results of any analysis at face value). It doesn’t seem like sexual dichromatism (red text) is a trait that is particularly clustered within certain taxonomic groups, so seems that <em>D</em> is giving us a sensible answer.</p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="phylogenetic-signal-in-r.html#cb188-1"></a>mycolours &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>)</span>
<span id="cb188-2"><a href="phylogenetic-signal-in-r.html#cb188-2"></a></span>
<span id="cb188-3"><a href="phylogenetic-signal-in-r.html#cb188-3"></a><span class="kw">plot</span>(frog<span class="op">$</span>phy, <span class="dt">show.tip.label =</span> <span class="ot">TRUE</span>, <span class="dt">tip.color =</span>     </span>
<span id="cb188-4"><a href="phylogenetic-signal-in-r.html#cb188-4"></a>       mycolours[<span class="kw">as.numeric</span>(mydata<span class="op">$</span>sex_di_binary)<span class="op">+</span><span class="dv">1</span>], </span>
<span id="cb188-5"><a href="phylogenetic-signal-in-r.html#cb188-5"></a>       <span class="dt">no.margin =</span> <span class="ot">TRUE</span>, <span class="dt">cex =</span> <span class="fl">0.4</span>)</span>
<span id="cb188-6"><a href="phylogenetic-signal-in-r.html#cb188-6"></a><span class="co"># Add a legend</span></span>
<span id="cb188-7"><a href="phylogenetic-signal-in-r.html#cb188-7"></a><span class="kw">legend</span>(<span class="st">&quot;bottomleft&quot;</span>, <span class="dt">fill =</span> mycolours, </span>
<span id="cb188-8"><a href="phylogenetic-signal-in-r.html#cb188-8"></a>       <span class="dt">legend =</span> <span class="kw">c</span>(<span class="st">&quot;Absent&quot;</span>, <span class="st">&quot;Present&quot;</span>), </span>
<span id="cb188-9"><a href="phylogenetic-signal-in-r.html#cb188-9"></a>       <span class="dt">bty =</span> <span class="st">&quot;n&quot;</span>)</span></code></pre></div>
<img src="pcm-primer_files/figure-html/unnamed-chunk-146-1.png" width="672" />
<div class="warning">
<p>
Always carefully consider what variation in values of <span class="math inline"><span class="math inline">\(\lambda\)</span></span> and <em>K</em> and <em>D</em> across traits and groups really means. It may not tell you as much about your system as you think it does. Phylogenetic signal is only a pattern, not a process!
</p>
</div>
</div>
</div>
<div id="summary-3" class="section level2">
<h2><span class="header-section-number">5.5</span> Summary</h2>
<p>You should now know how to estimate the phylogenetic signal of your variables using R.</p>
</div>
<div id="practical-exercises" class="section level2">
<h2><span class="header-section-number">5.6</span> Practical exercises</h2>
<p>In the data folder there is another tree (<code>primate-tree.nex</code>) and dataset (<code>primate-data.csv</code>) for investigating the evolution of primate life-history variables. These data come from the PanTHERIA database <span class="citation">(Jones et al. <a href="#ref-jones2009pantheria" role="doc-biblioref">2009</a>)</span> and 10kTrees <span class="citation">(Arnold, Matthews, and Nunn <a href="#ref-arnold201010ktrees" role="doc-biblioref">2010</a>)</span>.</p>
<p>Read in the tree and data, then prepare them for a PCM analysis (you may have already done this in the previous exercise which should save you some time). Then answer the following questions:</p>
<ol style="list-style-type: decimal">
<li>What is <span class="math inline">\(\lambda\)</span> for log gestation length? (<code>GestationLen_d</code>)</li>
<li>What is <em>K</em> for log gestation length? (<code>GestationLen_d</code>)</li>
<li>What is D for social status? (<code>SocialStatus</code>)</li>
</ol>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-arnold201010ktrees">
<p>Arnold, Christian, Luke J Matthews, and Charles L Nunn. 2010. “The 10kTrees Website: A New Online Resource for Primate Phylogeny.” <em>Evolutionary Anthropology: Issues, News, and Reviews</em> 19 (3): 114–18.</p>
</div>
<div id="ref-blomberg2003testing">
<p>Blomberg, Simon P, Theodore Garland Jr, and Anthony R Ives. 2003. “Testing for Phylogenetic Signal in Comparative Data: Behavioral Traits Are More Labile.” <em>Evolution</em> 57 (4): 717–45.</p>
</div>
<div id="ref-feng2017phylogenomics">
<p>Feng, Yan-Jie, David C Blackburn, Dan Liang, David M Hillis, David B Wake, David C Cannatella, and Peng Zhang. 2017. “Phylogenomics Reveals Rapid, Simultaneous Diversification of Three Major Clades of Gondwanan Frogs at the Cretaceous–Paleogene Boundary.” <em>Proceedings of the National Academy of Sciences</em> 114 (29): E5864–E5870.</p>
</div>
<div id="ref-fritz2010selectivity">
<p>Fritz, Susanne A, and Andy Purvis. 2010. “Selectivity in Mammalian Extinction Risk and Threat Types: A New Measure of Phylogenetic Signal Strength in Binary Traits.” <em>Conservation Biology</em> 24 (4): 1042–51.</p>
</div>
<div id="ref-jones2009pantheria">
<p>Jones, Kate E, Jon Bielby, Marcel Cardillo, Susanne A Fritz, Justin O’Dell, C David L Orme, Kamran Safi, et al. 2009. “PanTHERIA: A Species-Level Database of Life History, Ecology, and Geography of Extant and Recently Extinct Mammals: Ecological Archives E090-184.” <em>Ecology</em> 90 (9): 2648–8.</p>
</div>
<div id="ref-pagel1997inferring">
<p>Pagel, Mark. 1997. “Inferring Evolutionary Processes from Phylogenies.” <em>Zoologica Scripta</em> 26 (4): 331–48.</p>
</div>
<div id="ref-thomas2020eye">
<p>Thomas, Kate N, David J Gower, Rayna C Bell, Matthew K Fujita, Ryan K Schott, and Jeffrey W Streicher. 2020. “Eye Size and Investment in Frogs and Toads Correlate with Adult Habitat, Activity Pattern and Breeding Ecology.” <em>Proceedings of the Royal Society B</em> 287 (1935): 20201393.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="preparing-your-tree-and-data-for-pcms-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="phylogenetic-generalised-least-squares-pgls-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05-PhyloSignal.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["pcm-primer.pdf", "pcm-primer.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
